<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>BST280</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="bst280_files/libs/clipboard/clipboard.min.js"></script>
<script src="bst280_files/libs/quarto-html/quarto.js"></script>
<script src="bst280_files/libs/quarto-html/popper.min.js"></script>
<script src="bst280_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="bst280_files/libs/quarto-html/anchor.min.js"></script>
<link href="bst280_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="bst280_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="bst280_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="bst280_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="bst280_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="bst280_files/libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="bst280_files/libs/vis-9.1.0/vis-network.min.css" rel="stylesheet">
<script src="bst280_files/libs/vis-9.1.0/vis-network.min.js"></script>
<script src="bst280_files/libs/visNetwork-binding-2.1.2/visNetwork.js"></script>


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">BST280</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biobase)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,
    as.data.frame, basename, cbind, colnames, dirname, do.call,
    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,
    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, rank, rbind, rownames, sapply, saveRDS, setdiff, table,
    tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'limma'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    plotMA</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggfortify)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: ggplot2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gplots)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'gplots'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:stats':

    lowess</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'matrixStats'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:Biobase':

    anyMissing, rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:Biobase':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:gplots':

    space</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    I, expand.grid, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'ggplot2'</span>)      <span class="co"># For plotting</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'reshape2'</span>)     <span class="co"># For data processing</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'visNetwork'</span>)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">'fgsea'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>##Load Data and Preprocess</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># downloading data</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>gtex <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"gtex_pancreas.rds"</span>)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>tcga <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"tcga_paad.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="co"># structure of both datasets</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>gtex</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: RangedSummarizedExperiment 
dim: 31530 349 
metadata(8): time_created recount3_version ... annotation recount3_url
assays(2): raw_counts logtpm
rownames(31530): ENSG00000278267.1 ENSG00000227232.5 ...
  ENSG00000234399.1 ENSG00000236379.2
rowData names(10): source type ... havana_gene tag
colnames(349): GTEX-13FLV-0626-SM-5IFEY.1 GTEX-15ER7-1326-SM-6LPK9.1
  ... GTEX-1M5QR-0926-SM-DHXK8.1 GTEX-1PPH8-2326-SM-EAZ48.1
colData names(198): rail_id external_id ... recount_seq_qc.errq
  BigWigURL</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>tcga</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: RangedSummarizedExperiment 
dim: 34032 179 
metadata(8): time_created recount3_version ... annotation recount3_url
assays(2): raw_counts logtpm
rownames(34032): ENSG00000223972.5 ENSG00000278267.1 ...
  ENSG00000224240.1 ENSG00000237917.1
rowData names(10): source type ... havana_gene tag
colnames(179): 4a8f7d08-654e-4200-9b80-975d2ed0b205
  f7b2e263-f0ae-440e-96c5-139233079a06 ...
  47df123d-bb72-43bc-b7d6-caca55af62f7
  8956a38e-5c9a-4077-ba5f-1472b2b86afb
colData names(937): rail_id external_id ... recount_seq_qc.errq
  BigWigURL</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># checking sequencing platform</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(<span class="fu">colData</span>(tcga)<span class="sc">$</span>tcga.gdc_platform)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Illumina HiSeq"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(<span class="fu">colData</span>(tcga))[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "rail_id"           "external_id"       "study"            
[4] "tcga.tcga_barcode" "tcga.gdc_file_id" </code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># RangedSummarizedExperiment object consists of 3 main sections:</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Assay (expression data), </span></span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co"># ColData (an optional DataFrame describing the samples, row names, if present, </span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="co"># become the column names of the RangedSummarizedExperiment), </span></span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a><span class="co"># RowData (describing the rows, the number of rows of the RowData DataFrame must </span></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a><span class="co"># equal the number of rows of the matrices in Assay. </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(<span class="fu">assays</span>(tcga))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Formal class 'SimpleList' [package "S4Vectors"] with 4 slots
  ..@ listData       :List of 2
  .. ..$ raw_counts: num [1:34032, 1:179] 496 63 39846 212 0 ...
  .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. ..$ : chr [1:34032] "ENSG00000223972.5" "ENSG00000278267.1" "ENSG00000227232.5" "ENSG00000243485.5" ...
  .. .. .. ..$ : chr [1:179] "4a8f7d08-654e-4200-9b80-975d2ed0b205" "f7b2e263-f0ae-440e-96c5-139233079a06" "9709f7c5-4a89-49e5-9983-8d94eefec50d" "2e8f3cef-6852-4118-a6df-513c72488013" ...
  .. ..$ logtpm    : num [1:34032, 1:179] 0.1081 0.4589 2.5887 0.0747 0 ...
  .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. ..$ : chr [1:34032] "ENSG00000223972.5" "ENSG00000278267.1" "ENSG00000227232.5" "ENSG00000243485.5" ...
  .. .. .. ..$ : chr [1:179] "4a8f7d08-654e-4200-9b80-975d2ed0b205" "f7b2e263-f0ae-440e-96c5-139233079a06" "9709f7c5-4a89-49e5-9983-8d94eefec50d" "2e8f3cef-6852-4118-a6df-513c72488013" ...
  ..@ elementType    : chr "ANY"
  ..@ elementMetadata: NULL
  ..@ metadata       : list()</code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">assay</span>(tcga, <span class="st">"logtpm"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 34032   179</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(<span class="fu">colData</span>(tcga))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Formal class 'DFrame' [package "S4Vectors"] with 6 slots
  ..@ rownames       : chr [1:179] "4a8f7d08-654e-4200-9b80-975d2ed0b205" "f7b2e263-f0ae-440e-96c5-139233079a06" "9709f7c5-4a89-49e5-9983-8d94eefec50d" "2e8f3cef-6852-4118-a6df-513c72488013" ...
  ..@ nrows          : int 179
  ..@ elementType    : chr "ANY"
  ..@ elementMetadata: NULL
  ..@ metadata       : list()
  ..@ listData       :List of 937
  .. ..$ rail_id                                                                                           : int [1:179] 106668 106676 106789 107166 107207 107700 108164 108175 108309 108741 ...
  .. ..$ external_id                                                                                       : chr [1:179] "4a8f7d08-654e-4200-9b80-975d2ed0b205" "f7b2e263-f0ae-440e-96c5-139233079a06" "9709f7c5-4a89-49e5-9983-8d94eefec50d" "2e8f3cef-6852-4118-a6df-513c72488013" ...
  .. ..$ study                                                                                             : chr [1:179] "PAAD" "PAAD" "PAAD" "PAAD" ...
  .. ..$ tcga.tcga_barcode                                                                                 : chr [1:179] "TCGA-HV-A7OP-01A-11R-A33R-07" "TCGA-IB-A5ST-01A-11R-A32O-07" "TCGA-FB-AAPS-01A-12R-A39D-07" "TCGA-XN-A8T3-01A-11R-A36G-07" ...
  .. ..$ tcga.gdc_file_id                                                                                  : chr [1:179] "4a8f7d08-654e-4200-9b80-975d2ed0b205" "f7b2e263-f0ae-440e-96c5-139233079a06" "9709f7c5-4a89-49e5-9983-8d94eefec50d" "2e8f3cef-6852-4118-a6df-513c72488013" ...
  .. ..$ tcga.gdc_cases.project.name                                                                       : chr [1:179] "Pancreatic Adenocarcinoma" "Pancreatic Adenocarcinoma" "Pancreatic Adenocarcinoma" "Pancreatic Adenocarcinoma" ...
  .. ..$ tcga.gdc_cases.project.released                                                                   : logi [1:179] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ tcga.gdc_cases.project.state                                                                      : chr [1:179] "legacy" "legacy" "legacy" "legacy" ...
  .. ..$ tcga.gdc_cases.project.primary_site                                                               : chr [1:179] "Pancreas" "Pancreas" "Pancreas" "Pancreas" ...
  .. ..$ tcga.gdc_cases.project.project_id                                                                 : chr [1:179] "TCGA-PAAD" "TCGA-PAAD" "TCGA-PAAD" "TCGA-PAAD" ...
  .. ..$ tcga.cgc_sample_sample_type                                                                       : chr [1:179] "Primary Tumor" "Primary Tumor" "Primary Tumor" "Primary Tumor" ...
  .. ..$ tcga.gdc_cases.case_id                                                                            : chr [1:179] "5c51fbfe-153c-4cd2-97bd-e444daa0b20f" "6c0ea771-ea86-4db6-994a-2372b2b8f536" "2048d0f9-ed9f-46df-bcd4-3dfd9137670f" "dd7bcdd8-1660-4dab-9860-9d136c4cee77" ...
  .. ..$ tcga.gdc_updated_datetime                                                                         : chr [1:179] "2016-08-26T01:45:12.068480-05:00" "2016-08-27T04:55:42.941893-05:00" "2016-08-09T14:40:08.877139-05:00" "2016-09-29T18:45:35.086804-05:00" ...
  .. ..$ tcga.gdc_file_name                                                                                : chr [1:179] "UNCID_2363472.caee83ab-8805-45e2-817c-0cef5f71c292.131217_UNC15-SN850_0343_AC330UACXX_1_GCCAAT.tar.gz" "UNCID_2328645.2e691339-21b8-452d-a399-41247febd91f.131008_UNC11-SN627_0326_AC2KLPACXX_2_CAGATC.tar.gz" "UNCID_2648455.8077c0b3-255d-437e-969d-c1b588edc28d.140617_UNC12-SN629_0374_AC4KA5ACXX_2_CAGATC.tar.gz" "UNCID_2479110.b2323204-612b-44a0-a268-ac8f07f01462.140404_UNC11-SN627_0350_BC3K66ACXX_5_TTAGGC.tar.gz" ...
  .. ..$ tcga.gdc_submitter_id                                                                             : chr [1:179] "72d057ef-6e32-4cf2-b820-5f5310d9c3e0" "3ebfaadc-6ecc-4ad3-86c1-5861ef822d98" "0083370e-c824-4d35-bac2-95ece7b63008" "ce094ced-7a77-4a8c-939b-a5714e92ad4f" ...
  .. ..$ tcga.gdc_file_size                                                                                : num [1:179] 4.48e+09 5.92e+09 4.39e+09 5.21e+09 7.65e+09 ...
  .. ..$ tcga.gdc_access                                                                                   : chr [1:179] "controlled" "controlled" "controlled" "controlled" ...
  .. ..$ tcga.gdc_platform                                                                                 : chr [1:179] "Illumina HiSeq" "Illumina HiSeq" "Illumina HiSeq" "Illumina HiSeq" ...
  .. ..$ tcga.gdc_state                                                                                    : chr [1:179] "live" "live" "live" "live" ...
  .. ..$ tcga.gdc_type                                                                                     : chr [1:179] "file" "file" "file" "file" ...
  .. ..$ tcga.gdc_file_state                                                                               : logi [1:179] NA NA NA NA NA NA ...
  .. ..$ tcga.gdc_experimental_strategy                                                                    : chr [1:179] "RNA-Seq" "RNA-Seq" "RNA-Seq" "RNA-Seq" ...
  .. ..$ tcga.gdc_data_type                                                                                : chr [1:179] "Unaligned reads" "Unaligned reads" "Unaligned reads" "Unaligned reads" ...
  .. ..$ tcga.gdc_md5sum                                                                                   : chr [1:179] "6686fc510790f119404b78161cd0e634" "998a543698403b7a5564b6ef12082bd4" "a854ab949186f8158aaa3e8392256d3b" "2cb650af99d85c4aa9c857c34b502f5e" ...
  .. ..$ tcga.gdc_data_format                                                                              : chr [1:179] "TARGZ" "TARGZ" "TARGZ" "TARGZ" ...
  .. ..$ tcga.gdc_data_category                                                                            : chr [1:179] "Raw sequencing data" "Raw sequencing data" "Raw sequencing data" "Raw sequencing data" ...
  .. ..$ tcga.gdc_center.code                                                                              : int [1:179] 7 7 7 7 7 7 7 7 7 7 ...
  .. ..$ tcga.gdc_center.name                                                                              : chr [1:179] "University of North Carolina" "University of North Carolina" "University of North Carolina" "University of North Carolina" ...
  .. ..$ tcga.gdc_center.short_name                                                                        : chr [1:179] "UNC" "UNC" "UNC" "UNC" ...
  .. ..$ tcga.gdc_center.center_id                                                                         : chr [1:179] "ee7a85b3-8177-5d60-a10c-51180eb9009c" "ee7a85b3-8177-5d60-a10c-51180eb9009c" "ee7a85b3-8177-5d60-a10c-51180eb9009c" "ee7a85b3-8177-5d60-a10c-51180eb9009c" ...
  .. ..$ tcga.gdc_center.namespace                                                                         : chr [1:179] "unc.edu" "unc.edu" "unc.edu" "unc.edu" ...
  .. ..$ tcga.gdc_center.center_type                                                                       : chr [1:179] "CGCC" "CGCC" "CGCC" "CGCC" ...
  .. ..$ tcga.gdc_metadata_files.updated_datetime.analysis                                                 : chr [1:179] "2016-09-29T23:19:33.237312-05:00" "2016-09-29T21:09:16.486588-05:00" "2016-09-29T18:46:06.225737-05:00" "2016-09-29T23:51:12.187655-05:00" ...
  .. ..$ tcga.gdc_metadata_files.created_datetime.analysis                                                 : chr [1:179] "2016-05-03T11:38:01.261401-05:00" "2016-05-03T11:35:41.350369-05:00" "2016-05-03T12:06:38.887369-05:00" "2016-05-03T11:48:06.252263-05:00" ...
  .. ..$ tcga.gdc_metadata_files.file_name.analysis                                                        : chr [1:179] "72d057ef-6e32-4cf2-b820-5f5310d9c3e0_analysis.xml" "3ebfaadc-6ecc-4ad3-86c1-5861ef822d98_analysis.xml" "0083370e-c824-4d35-bac2-95ece7b63008_analysis.xml" "ce094ced-7a77-4a8c-939b-a5714e92ad4f_analysis.xml" ...
  .. ..$ tcga.gdc_metadata_files.md5sum.analysis                                                           : chr [1:179] "b0dc079c4dd8ac45f6be894098e508ca" "8ee778bcc01028bd1cd8eb1eeec38409" "190241055fd5c3d90dec19ad211737c6" "b0fce2ae45bb93c80ce22bd6ac2914cf" ...
  .. ..$ tcga.gdc_metadata_files.data_format.analysis                                                      : chr [1:179] "SRA XML" "SRA XML" "SRA XML" "SRA XML" ...
  .. ..$ tcga.gdc_metadata_files.access.analysis                                                           : chr [1:179] "open" "open" "open" "open" ...
  .. ..$ tcga.gdc_metadata_files.file_id.analysis                                                          : chr [1:179] "d85c55e3-cd76-48d0-b412-6f8deae53324" "87dd881a-5ea3-4822-8a39-4088913bed7e" "2ee9c45e-cfcb-458e-ab6e-dc94c53ce439" "ec33f3f9-498f-4e99-9af5-65be5467eecb" ...
  .. ..$ tcga.gdc_metadata_files.data_category.analysis                                                    : chr [1:179] "Sequencing Data" "Sequencing Data" "Sequencing Data" "Sequencing Data" ...
  .. ..$ tcga.gdc_metadata_files.file_size.analysis                                                        : int [1:179] 2124 2124 2124 2124 2124 2123 2124 2124 2124 2124 ...
  .. ..$ tcga.gdc_metadata_files.file_state.analysis                                                       : chr [1:179] "submitted" "submitted" "submitted" "submitted" ...
  .. ..$ tcga.gdc_metadata_files.updated_datetime.experiment                                               : chr [1:179] "2016-10-07T16:10:54.772464-05:00" "2016-09-29T18:20:11.226709-05:00" "2016-09-29T20:32:23.047570-05:00" "2016-09-29T23:41:11.970429-05:00" ...
  .. ..$ tcga.gdc_metadata_files.file_name.experiment                                                      : chr [1:179] "72d057ef-6e32-4cf2-b820-5f5310d9c3e0_experiment.xml" "3ebfaadc-6ecc-4ad3-86c1-5861ef822d98_experiment.xml" "0083370e-c824-4d35-bac2-95ece7b63008_experiment.xml" "ce094ced-7a77-4a8c-939b-a5714e92ad4f_experiment.xml" ...
  .. ..$ tcga.gdc_metadata_files.md5sum.experiment                                                         : chr [1:179] "6fe2c4fdcc94f87ff1fa5d8242eb5779" "f6c73c7d7481a88ff3cd724a3bbefe3e" "1d2839df97345ee2cb7bc2ecc83f3fdb" "6787fe845b130d8d535892ee2ee56a91" ...
  .. ..$ tcga.gdc_metadata_files.file_id.experiment                                                        : chr [1:179] "d426a37f-a7d9-4bae-bc1a-fef4644a34ad" "1e6dfb10-d0d5-46c6-bd73-21522a7cb22d" "70cab6ec-5d45-4e6c-9ccc-5e5282ed12ea" "e5f42c1d-7eae-4043-8d42-f2b46e6922ba" ...
  .. ..$ tcga.gdc_metadata_files.file_size.experiment                                                      : int [1:179] 2628 2628 2628 2628 2431 2431 2628 2431 2431 2628 ...
  .. ..$ tcga.gdc_metadata_files.updated_datetime.run                                                      : chr [1:179] "2016-09-29T19:29:44.062629-05:00" "2016-09-29T22:21:22.215824-05:00" "2016-09-29T21:12:47.746716-05:00" "2016-09-29T21:48:11.981553-05:00" ...
  .. ..$ tcga.gdc_metadata_files.file_name.run                                                             : chr [1:179] "72d057ef-6e32-4cf2-b820-5f5310d9c3e0_run.xml" "3ebfaadc-6ecc-4ad3-86c1-5861ef822d98_run.xml" "0083370e-c824-4d35-bac2-95ece7b63008_run.xml" "ce094ced-7a77-4a8c-939b-a5714e92ad4f_run.xml" ...
  .. ..$ tcga.gdc_metadata_files.md5sum.run                                                                : chr [1:179] "e561cc398f1a0beaee858e6c6bc82d53" "76b77773ee8ec7d424f05f65067d0ad0" "7294adacb12a7a8ceaae0497abc0c76e" "14c6acff90bb0117040305059a6d0828" ...
  .. ..$ tcga.gdc_metadata_files.file_id.run                                                               : chr [1:179] "4a6c209e-5027-4ceb-bdba-8ce93c99845e" "b4516681-4089-4f3e-be68-e8a9688980b6" "8a133520-09d5-494c-a41c-b0a911960515" "9fe8b58e-9323-4664-a228-a3b45ebc021a" ...
  .. ..$ tcga.gdc_metadata_files.file_size.run                                                             : int [1:179] 354 354 354 354 354 354 354 354 354 354 ...
  .. ..$ tcga.gdc_cases.updated_datetime                                                                   : chr [1:179] "2016-09-08T10:38:21.691030-05:00" "2016-09-08T10:25:38.866496-05:00" "2016-09-08T10:53:55.250065-05:00" "2016-09-08T12:08:39.174411-05:00" ...
  .. ..$ tcga.gdc_cases.submitter_id                                                                       : chr [1:179] "TCGA-HV-A7OP" "TCGA-IB-A5ST" "TCGA-FB-AAPS" "TCGA-XN-A8T3" ...
  .. ..$ tcga.gdc_cases.demographic.updated_datetime                                                       : chr [1:179] "2016-09-02T18:52:01.501356-05:00" "2016-09-02T18:53:52.626620-05:00" "2016-09-02T18:52:37.976843-05:00" "2016-09-02T18:52:11.282687-05:00" ...
  .. ..$ tcga.gdc_cases.demographic.gender                                                                 : chr [1:179] "male" "female" "female" "male" ...
  .. ..$ tcga.gdc_cases.demographic.submitter_id                                                           : chr [1:179] "TCGA-HV-A7OP_demographic" "TCGA-IB-A5ST_demographic" "TCGA-FB-AAPS_demographic" "TCGA-XN-A8T3_demographic" ...
  .. ..$ tcga.gdc_cases.demographic.year_of_birth                                                          : int [1:179] 1940 1948 1946 1945 1932 1942 1963 1932 1955 1939 ...
  .. ..$ tcga.gdc_cases.demographic.race                                                                   : chr [1:179] "asian" "white" "white" "white" ...
  .. ..$ tcga.gdc_cases.demographic.demographic_id                                                         : chr [1:179] "89dc66e6-0a32-532d-bb93-4e2ae08c5139" "bcccc40c-988b-5c4d-a181-6ae33a324eff" "91880be0-92d1-5f19-9d6e-14cfe7f13315" "a92a0953-9ffb-5874-9f95-cb446f8247bd" ...
  .. ..$ tcga.gdc_cases.demographic.ethnicity                                                              : chr [1:179] "not reported" "not hispanic or latino" "not hispanic or latino" "hispanic or latino" ...
  .. ..$ tcga.gdc_cases.demographic.year_of_death                                                          : int [1:179] NA NA NA NA NA NA 2013 NA 2009 NA ...
  .. ..$ tcga.gdc_cases.project.program.dbgap_accession_number                                             : chr [1:179] "phs000178" "phs000178" "phs000178" "phs000178" ...
  .. ..$ tcga.gdc_cases.project.program.name                                                               : chr [1:179] "TCGA" "TCGA" "TCGA" "TCGA" ...
  .. ..$ tcga.gdc_cases.project.program.program_id                                                         : chr [1:179] "b80aa962-9650-5110-b3eb-bd087da808db" "b80aa962-9650-5110-b3eb-bd087da808db" "b80aa962-9650-5110-b3eb-bd087da808db" "b80aa962-9650-5110-b3eb-bd087da808db" ...
  .. ..$ tcga.gdc_cases.tissue_source_site.project                                                         : chr [1:179] "Pancreatic adenocarcinoma" "Pancreatic adenocarcinoma" "Pancreatic adenocarcinoma" "Pancreatic adenocarcinoma" ...
  .. ..$ tcga.gdc_cases.tissue_source_site.bcr_id                                                          : chr [1:179] "IGC" "IGC" "IGC" "NCH" ...
  .. ..$ tcga.gdc_cases.tissue_source_site.code                                                            : chr [1:179] "HV" "IB" "FB" "XN" ...
  .. ..$ tcga.gdc_cases.tissue_source_site.tissue_source_site_id                                           : chr [1:179] "3cdff411-38c0-56d2-b98d-bd27c5461050" "e84a845d-1e5d-5f50-8177-c71af2a77ebc" "8cc41622-7c32-5f35-9ba6-dbc913c8da6d" "70722577-0ac4-53a6-b0f3-a52156981c44" ...
  .. ..$ tcga.gdc_cases.tissue_source_site.name                                                            : chr [1:179] "National Cancer Center Korea" "Alberta Health Services" "Asterand" "University of Sao Paulo" ...
  .. ..$ tcga.gdc_cases.diagnoses.classification_of_tumor                                                  : chr [1:179] "not reported" "not reported" "not reported" "not reported" ...
  .. ..$ tcga.gdc_cases.diagnoses.primary_diagnosis                                                        : chr [1:179] "c25.0" "c25.0" "c25.0" "c25.0" ...
  .. ..$ tcga.gdc_cases.diagnoses.submitter_id                                                             : chr [1:179] "TCGA-HV-A7OP_diagnosis" "TCGA-IB-A5ST_diagnosis" "TCGA-FB-AAPS_diagnosis" "TCGA-XN-A8T3_diagnosis" ...
  .. ..$ tcga.gdc_cases.diagnoses.tumor_stage                                                              : chr [1:179] "stage iib" "stage iib" "stage iib" "stage ib" ...
  .. ..$ tcga.gdc_cases.diagnoses.age_at_diagnosis                                                         : int [1:179] 26322 23672 22981 24810 28554 25227 18315 29213 19718 26943 ...
  .. ..$ tcga.gdc_cases.diagnoses.vital_status                                                             : chr [1:179] "alive" "alive" "alive" "alive" ...
  .. ..$ tcga.gdc_cases.diagnoses.morphology                                                               : chr [1:179] "8500/3" "8500/3" "8500/3" "8500/3" ...
  .. ..$ tcga.gdc_cases.diagnoses.days_to_death                                                            : int [1:179] NA NA NA NA NA NA 183 NA 486 NA ...
  .. ..$ tcga.gdc_cases.diagnoses.days_to_last_follow_up                                                   : int [1:179] 978 635 228 951 1257 706 NA 462 NA 676 ...
  .. ..$ tcga.gdc_cases.diagnoses.diagnosis_id                                                             : chr [1:179] "622433cf-f4c5-5c27-800b-34ad5d73b365" "7477b48a-b339-56c7-bb57-46b90004d8e1" "7ccc058f-a131-5802-8228-cf571ab031db" "17c0b5e9-5ddb-5dc1-a550-38eddb8ee5b9" ...
  .. ..$ tcga.gdc_cases.diagnoses.tissue_or_organ_of_origin                                                : chr [1:179] "c25.0" "c25.0" "c25.0" "c25.0" ...
  .. ..$ tcga.gdc_cases.diagnoses.days_to_birth                                                            : int [1:179] -26322 -23672 -22981 -24810 -28554 -25227 -18315 -29213 -19718 -26943 ...
  .. ..$ tcga.gdc_cases.diagnoses.treatments.submitter_id                                                  : chr [1:179] "TCGA-HV-A7OP_treatment" "TCGA-IB-A5ST_treatment" "TCGA-FB-AAPS_treatment" "TCGA-XN-A8T3_treatment" ...
  .. ..$ tcga.gdc_cases.diagnoses.treatments.treatment_id                                                  : chr [1:179] "310c608e-c1b0-5b5c-9c5d-ba66be5387ac" "13be0b4a-da65-5570-90d8-3c61f70856be" "72539f7c-cec6-5992-adfe-ea87858e1d25" "569fa435-3d5f-5ea9-b8be-2f7f52a6dff4" ...
  .. ..$ tcga.gdc_cases.samples.sample_type_id                                                             : int [1:179] 1 1 1 1 1 1 1 1 1 1 ...
  .. ..$ tcga.gdc_cases.samples.oct_embedded                                                               : chr [1:179] "false" "true" "true" "true" ...
  .. ..$ tcga.gdc_cases.samples.submitter_id                                                               : chr [1:179] "TCGA-HV-A7OP-01A" "TCGA-IB-A5ST-01A" "TCGA-FB-AAPS-01A" "TCGA-XN-A8T3-01A" ...
  .. ..$ tcga.gdc_cases.samples.intermediate_dimension                                                     : num [1:179] NA NA NA NA 1 0.9 NA 0.5 0.4 NA ...
  .. ..$ tcga.gdc_cases.samples.sample_id                                                                  : chr [1:179] "4a004132-335f-4299-88cd-95f473523ed3" "08fc3507-8072-4fbb-a39e-46b0d51ba8d0" "0dd80160-fc7e-4162-b744-32474bb7223e" "0f850dd7-7a15-434d-acf6-2398f53cf6ad" ...
  .. ..$ tcga.gdc_cases.samples.is_ffpe                                                                    : logi [1:179] FALSE FALSE FALSE FALSE FALSE FALSE ...
  .. ..$ tcga.gdc_cases.samples.pathology_report_uuid                                                      : chr [1:179] "9CF238DA-57D1-4DDF-9ECB-23BC9F887056" "DB6AF92C-6929-41F6-A40F-4CA1C40F2A30" "0369BABC-3B58-4386-9CF6-71889B1F651B" "4B77F18C-2A61-42E5-BF41-05FB9959EB3F" ...
  .. ..$ tcga.gdc_cases.samples.sample_type                                                                : chr [1:179] "Primary Tumor" "Primary Tumor" "Primary Tumor" "Primary Tumor" ...
  .. ..$ tcga.gdc_cases.samples.shortest_dimension                                                         : num [1:179] NA NA NA NA 0.2 0.5 NA 0.2 0.3 NA ...
  .. ..$ tcga.gdc_cases.samples.days_to_collection                                                         : int [1:179] 453 103 2035 628 NA NA 79 NA NA 515 ...
  .. ..$ tcga.gdc_cases.samples.initial_weight                                                             : int [1:179] 80 790 90 80 NA NA 240 NA NA 80 ...
  .. ..$ tcga.gdc_cases.samples.longest_dimension                                                          : num [1:179] NA NA NA NA 2 0.9 NA 1.5 1 NA ...
  .. ..$ tcga.gdc_cases.samples.portions.creation_datetime                                                 : int [1:179] 1379462400 1361491200 1395792000 1387238400 1325116800 1328140800 1399852800 1332460800 1325116800 1397520000 ...
  .. ..$ tcga.gdc_cases.samples.portions.portion_number                                                    : int [1:179] 11 11 12 11 11 11 22 11 21 12 ...
  .. ..$ tcga.gdc_cases.samples.portions.weight                                                            : int [1:179] 30 30 30 30 27 36 20 35 40 30 ...
  .. .. [list output truncated]</code></pre>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">colData</span>(tcga))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 179 937</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a>raw_counts <span class="ot">&lt;-</span> <span class="fu">assay</span>(tcga, <span class="st">"raw_counts"</span>)</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a>gene_lengths <span class="ot">&lt;-</span> <span class="fu">rowData</span>(tcga)<span class="sc">$</span>bp_length<span class="sc">/</span><span class="dv">1000</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>rpk <span class="ot">&lt;-</span> raw_counts<span class="sc">/</span>gene_lengths</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>scaling_factors <span class="ot">&lt;-</span> <span class="fu">colSums</span>(rpk)</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>tpm <span class="ot">&lt;-</span> <span class="fu">sweep</span>(rpk, <span class="dv">2</span>, scaling_factors, <span class="at">FUN =</span> <span class="st">"/"</span>) <span class="sc">*</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>tpm[<span class="dv">1</span>, <span class="dv">0</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>4a8f7d08-654e-4200-9b80-975d2ed0b205 f7b2e263-f0ae-440e-96c5-139233079a06 
                          0.11945405                           0.25148839 
9709f7c5-4a89-49e5-9983-8d94eefec50d 2e8f3cef-6852-4118-a6df-513c72488013 
                          0.16608711                           0.07032308 
8b23ab23-8931-42ca-972d-1528d44bf218 
                          0.17929883 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># which.max(rowData(tcga)$bp_length)</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="co"># rowData(tcga)[7664,]</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="co"># head(tpm)[1,0:5]</span></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="co"># x &lt;- assay(tcga, "logtpm")[1,0:5]</span></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a><span class="co"># ((exp(x) - 1) * (scaling_factors[0:5]) / 10^6) * gene_lengths[1] # unclear which method was used to calculate logtpm - ??? </span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="co"># assay(tcga, "logtpm")[1,0:5]</span></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="co"># filtering TCGA data to only include first and second grade tumors</span></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>keep_stages <span class="ot">&lt;-</span> tcga<span class="sc">@</span>colData<span class="sc">@</span>listData<span class="sc">$</span>tcga.gdc_cases.diagnoses.tumor_stage <span class="sc">%in%</span> </span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"stage i"</span>, <span class="st">"stage ia"</span>, <span class="st">"stage ib"</span>, <span class="st">"stage iia"</span>, <span class="st">"stage iib"</span>)</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tcga)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 34032   179</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>tcga <span class="ot">&lt;-</span> tcga[, keep_stages]</span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tcga)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 34032   168</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="co"># defining a column that outlines the source: GTEX or TCGA</span></span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(gtex)<span class="sc">$</span>source <span class="ot">&lt;-</span> <span class="st">"GTEX"</span></span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tcga)<span class="sc">$</span>source <span class="ot">&lt;-</span> <span class="st">"TCGA"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol type="1">
<li>Analysis for Normal vs Tumour</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co">#intersection of columns (features of sample data)</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>common_cols <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">names</span>(<span class="fu">colData</span>(gtex)), <span class="fu">names</span>(<span class="fu">colData</span>(tcga)))</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a><span class="co"># identification of common genes for both GTEX and TCGA</span></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>common_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">rownames</span>(<span class="fu">assay</span>(gtex, <span class="st">"logtpm"</span>)), </span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">rownames</span>(<span class="fu">assay</span>(tcga, <span class="st">"logtpm"</span>)))</span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a><span class="co"># updating datasets by applying common genes and common columns filters</span></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>gtex_upd <span class="ot">&lt;-</span> gtex[common_genes, ]</span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>tcga_upd <span class="ot">&lt;-</span> tcga[common_genes, ]</span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(gtex_upd) <span class="ot">&lt;-</span> <span class="fu">colData</span>(gtex_upd)[, common_cols, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(tcga_upd) <span class="ot">&lt;-</span> <span class="fu">colData</span>(tcga_upd)[, common_cols, drop <span class="ot">=</span> <span class="cn">FALSE</span>]</span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a><span class="co"># merging</span></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>merged_logtpm <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">assay</span>(gtex_upd, <span class="st">"logtpm"</span>), <span class="fu">assay</span>(tcga_upd, <span class="st">"logtpm"</span>))</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(merged_logtpm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 29373   517</code></pre>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>merged_colData <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">rbind</span>(<span class="fu">colData</span>(gtex_upd), <span class="fu">colData</span>(tcga_upd)))</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(merged_colData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 517 101</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>PCA <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(merged_logtpm))</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(PCA, <span class="at">data =</span> merged_colData, <span class="at">colour=</span><span class="st">"source"</span>)</span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># genes will be filtered if they have very low counts (tpm &lt; 1) in 25% of the samples</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>merged_exprs <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">assay</span>(gtex_upd, <span class="st">"raw_counts"</span>), <span class="fu">assay</span>(tcga_upd, <span class="st">"raw_counts"</span>))</span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>merged_gene_lengths <span class="ot">&lt;-</span> <span class="fu">rowData</span>(gtex_upd)<span class="sc">$</span>bp_length<span class="sc">/</span><span class="dv">1000</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>merged_rpk <span class="ot">&lt;-</span> merged_exprs<span class="sc">/</span>merged_gene_lengths</span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>merged_scaling_factors <span class="ot">&lt;-</span> <span class="fu">colSums</span>(merged_rpk)</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>merged_tpm <span class="ot">&lt;-</span> <span class="fu">sweep</span>(merged_rpk, <span class="dv">2</span>, merged_scaling_factors, <span class="at">FUN =</span> <span class="st">"/"</span>) <span class="sc">*</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span> </span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(merged_tpm <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">&gt;=</span> <span class="fu">ncol</span>(merged_tpm)<span class="sc">/</span><span class="dv">4</span></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a>gtex_upd <span class="ot">&lt;-</span> gtex_upd[keep, ]</span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>tcga_upd <span class="ot">&lt;-</span> tcga_upd[keep, ]</span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(gtex_upd, <span class="fu">file</span>(<span class="st">"gtex.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in base::saveRDS(object, file = file, ascii = ascii, version = version,
: 'compress' is ignored unless 'file' is a file name</code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>merged_exprs <span class="ot">&lt;-</span> merged_exprs[keep, ]</span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(keep)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 21205</code></pre>
</div>
</div>
<section id="differential-gene-analysis" class="level3">
<h3 class="anchored" data-anchor-id="differential-gene-analysis">Differential Gene analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>condition <span class="ot">&lt;-</span> <span class="fu">factor</span>(merged_colData<span class="sc">$</span>source, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"GTEX"</span>, <span class="st">"TCGA"</span>))</span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> condition)</span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>voomOutput <span class="ot">&lt;-</span> <span class="fu">voom</span>(merged_exprs, design, <span class="at">plot=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(voomOutput, design)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit)</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSA</span>(fit, <span class="at">main=</span><span class="st">"Final model mean-variance trend"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit, <span class="at">coef=</span><span class="st">"conditionTCGA"</span>, <span class="at">number=</span><span class="cn">Inf</span>)</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     logFC    AveExpr        t       P.Value     adj.P.Val
ENSG00000256338.2 3.960009  3.9783624 83.24877 1.448398e-302 3.071329e-298
ENSG00000243517.1 3.296634  2.9468791 79.31877 1.887918e-292 2.001665e-288
ENSG00000258477.1 4.232352  1.6270415 76.34234 1.674308e-284 1.183457e-280
ENSG00000220793.5 4.685057 -0.7850547 75.41663 5.606926e-282 2.972371e-278
ENSG00000249264.1 3.504937  4.5060521 74.92418 1.266051e-280 5.369324e-277
ENSG00000219133.2 3.926446  1.3846414 74.60696 9.514242e-280 3.362492e-276
                         B
ENSG00000256338.2 683.3623
ENSG00000243517.1 660.0728
ENSG00000258477.1 641.6990
ENSG00000220793.5 635.2131
ENSG00000249264.1 632.9555
ENSG00000219133.2 630.7518</code></pre>
</div>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(res<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4042</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> res<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3199</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> res<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 843</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">=</span> <span class="fu">ifelse</span>(res<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(res<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">2</span>, <span class="at">yes=</span><span class="st">"red"</span>,<span class="at">no=</span><span class="st">"black"</span>)</span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">x=</span>res<span class="sc">$</span>logFC, <span class="at">y=</span><span class="sc">-</span><span class="fu">log</span>(res<span class="sc">$</span>adj.P.Val), <span class="at">xlab=</span><span class="st">"log-fold-change"</span>,</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab=</span><span class="st">"-log(FDR)"</span>,<span class="at">pch=</span><span class="dv">21</span>,<span class="at">col=</span>sig,<span class="at">bty=</span><span class="st">"l"</span>, <span class="at">main=</span><span class="st">"Volcano plot"</span>, <span class="at">cex=</span><span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">=</span> <span class="fu">ifelse</span>(res<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(res<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">2</span>, <span class="at">yes=</span><span class="st">"red"</span>,<span class="at">no=</span><span class="st">"black"</span>)</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">x=</span>res<span class="sc">$</span>AveExpr, <span class="at">y=</span>res<span class="sc">$</span>logFC, <span class="at">xlab=</span><span class="st">"Average log-expression"</span>, <span class="at">ylab=</span><span class="st">"log-fold-change"</span>,</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch=</span><span class="dv">21</span>,<span class="at">col=</span>sig,<span class="at">bty=</span><span class="st">"l"</span>, <span class="at">main=</span><span class="st">"MA plot"</span>, <span class="at">cex=</span><span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-14-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="gene-set-enrichment-analysis" class="level3">
<h3 class="anchored" data-anchor-id="gene-set-enrichment-analysis">Gene Set Enrichment Analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">"ensembl"</span>,<span class="at">dataset=</span><span class="st">"hsapiens_gene_ensembl"</span>,<span class="at">host =</span> <span class="st">"https://www.ensembl.org"</span>)</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>ens_genes <span class="ot">&lt;-</span> <span class="fu">row.names</span>(tcga_upd)</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the version of the Ensembl ID (the number after the .)</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>ens_genes <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, ens_genes)</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes=</span><span class="fu">c</span>(<span class="st">'ensembl_gene_id'</span>, <span class="st">'entrezgene_id'</span>, <span class="st">'hgnc_symbol'</span>), </span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">filters =</span> <span class="st">'ensembl_gene_id'</span>, </span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> ens_genes, </span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">mart =</span> ensembl)</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  ensembl_gene_id entrezgene_id hgnc_symbol
1 ENSG00000000457         57147       SCYL3
2 ENSG00000000460         55732       FIRRM
3 ENSG00000000938          2268         FGR
4 ENSG00000000971          3075         CFH
5 ENSG00000001460         90529       STPG1
6 ENSG00000001461         57185      NIPAL3</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>up_tcga <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res)[<span class="fu">which</span>(res<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">2</span> <span class="sc">&amp;</span> res<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)]</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>up_tcga <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, up_tcga) <span class="co"># This will remove the version in the ensemble ID</span></span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>up_tcga_ID <span class="ot">&lt;-</span> annot[<span class="fu">match</span>(up_tcga, annot<span class="sc">$</span>ensembl_gene_id), <span class="st">"entrezgene_id"</span>]</span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>up_gtex <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res)[<span class="fu">which</span>(res<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">2</span> <span class="sc">&amp;</span> res<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)]</span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>up_gtex <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, up_gtex)</span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>up_gtex_ID <span class="ot">&lt;-</span> annot[<span class="fu">match</span>(up_gtex, annot<span class="sc">$</span>ensembl_gene_id), <span class="st">"entrezgene_id"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform over-representation analyses for Gene Ontology terms using the limma package</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>go_tcga <span class="ot">&lt;-</span> <span class="fu">goana</span>(up_tcga_ID, <span class="at">species=</span><span class="st">"Hs"</span>)</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the list of top GO terms enriched for genes highly expressed in tcga</span></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a><span class="fu">topGO</span>(go_tcga, <span class="at">ontology =</span> <span class="st">"BP"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                     Term Ont    N   DE
GO:0007155                                  cell adhesion  BP 1553  427
GO:0002376                          immune system process  BP 2967  638
GO:0009605                  response to external stimulus  BP 2515  560
GO:0002682            regulation of immune system process  BP 1658  411
GO:0030155                    regulation of cell adhesion  BP  821  265
GO:0050896                           response to stimulus  BP 9302 1345
GO:0001775                                cell activation  BP 1146  321
GO:0051239 regulation of multicellular organismal process  BP 3118  617
GO:0032501               multicellular organismal process  BP 7568 1153
GO:0006952                               defense response  BP 1949  446
GO:0048518      positive regulation of biological process  BP 6485 1026
GO:0048522        positive regulation of cellular process  BP 5814  945
GO:0048583             regulation of response to stimulus  BP 4227  749
GO:0016477                                 cell migration  BP 1651  393
GO:0045321                           leukocyte activation  BP  993  281
GO:0006955                                immune response  BP 2162  465
GO:0098609                             cell-cell adhesion  BP  986  279
GO:0045785           positive regulation of cell adhesion  BP  499  180
GO:0048584    positive regulation of response to stimulus  BP 2421  491
GO:0002684   positive regulation of immune system process  BP 1140  296
                   P.DE
GO:0007155 4.964830e-88
GO:0002376 6.446211e-85
GO:0009605 7.265547e-79
GO:0002682 9.327482e-70
GO:0030155 7.549828e-69
GO:0050896 5.194363e-67
GO:0001775 5.733146e-67
GO:0051239 1.988488e-66
GO:0032501 6.485388e-65
GO:0006952 9.379418e-65
GO:0048518 3.317694e-63
GO:0048522 3.416285e-62
GO:0048583 2.779648e-61
GO:0016477 2.884409e-61
GO:0045321 2.666978e-59
GO:0006955 5.191643e-59
GO:0098609 7.420349e-59
GO:0045785 1.677063e-54
GO:0048584 2.423424e-54
GO:0002684 9.771403e-54</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>go_gtex <span class="ot">&lt;-</span> <span class="fu">goana</span>(up_gtex_ID, <span class="at">species=</span><span class="st">"Hs"</span>)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the list of top GO terms enriched for genes highly expressed in gtex</span></span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a><span class="fu">topGO</span>(go_gtex, <span class="at">ontology =</span> <span class="st">"BP"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                              Term Ont    N DE
GO:0007586                                               digestion  BP  144 21
GO:0003013                              circulatory system process  BP  619 32
GO:1901606                      alpha-amino acid catabolic process  BP   98 12
GO:0046395                       carboxylic acid catabolic process  BP  249 18
GO:0016054                          organic acid catabolic process  BP  249 18
GO:0170040              proteinogenic amino acid catabolic process  BP   71 10
GO:0009063                            amino acid catabolic process  BP  111 12
GO:0008015                                       blood circulation  BP  535 27
GO:0170035                          L-amino acid catabolic process  BP   75 10
GO:0044282                        small molecule catabolic process  BP  375 21
GO:0098739                           import across plasma membrane  BP  206 15
GO:1901605                      alpha-amino acid metabolic process  BP  209 15
GO:0006575          cellular modified amino acid metabolic process  BP  183 14
GO:0048878                                    chemical homeostasis  BP 1064 39
GO:0010749 regulation of nitric oxide mediated signal transduction  BP    9  4
GO:0006520                            amino acid metabolic process  BP  289 17
GO:0043436                               oxoacid metabolic process  BP  931 35
GO:0006082                          organic acid metabolic process  BP  937 35
GO:0030001                                     metal ion transport  BP  901 34
GO:0042219          cellular modified amino acid catabolic process  BP   32  6
                   P.DE
GO:0007586 5.115160e-14
GO:0003013 2.661489e-08
GO:1901606 1.058518e-07
GO:0046395 2.922311e-07
GO:0016054 2.922311e-07
GO:0170040 3.264762e-07
GO:0009063 4.217241e-07
GO:0008015 5.444568e-07
GO:0170035 5.518616e-07
GO:0044282 2.035340e-06
GO:0098739 2.623432e-06
GO:1901605 3.136507e-06
GO:0006575 3.179693e-06
GO:0048878 5.595581e-06
GO:0010749 9.669739e-06
GO:0006520 1.018893e-05
GO:0043436 1.020596e-05
GO:0006082 1.169903e-05
GO:0030001 1.271249e-05
GO:0042219 1.442587e-05</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform over-representation analyses for KEGG pathways</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>kegg_tcga <span class="ot">&lt;-</span> <span class="fu">kegga</span>(up_tcga_ID, <span class="at">species=</span><span class="st">"Hs"</span>)</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the list of top KEGG pathwayss enriched for genes highly expressed in tcga</span></span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a><span class="fu">topKEGG</span>(kegg_tcga)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                               Pathway   N  DE
hsa04145                                                     Phagosome 159  66
hsa04514                                       Cell adhesion molecules 158  63
hsa04820                                  Cytoskeleton in muscle cells 232  75
hsa04061 Viral protein interaction with cytokine and cytokine receptor 100  44
hsa04060                        Cytokine-cytokine receptor interaction 298  86
hsa04640                                    Hematopoietic cell lineage 100  43
hsa05416                                             Viral myocarditis  70  34
hsa05323                                          Rheumatoid arthritis  95  40
hsa04512                                      ECM-receptor interaction  89  38
hsa05150                               Staphylococcus aureus infection 102  41
hsa05140                                                 Leishmaniasis  79  34
hsa04659                                     Th17 cell differentiation 109  41
hsa05332                                     Graft-versus-host disease  45  24
hsa05330                                           Allograft rejection  39  22
hsa05200                                            Pathways in cancer 533 118
hsa05166                       Human T-cell leukemia virus 1 infection 224  63
hsa04940                                      Type I diabetes mellitus  44  23
hsa04610                           Complement and coagulation cascades  88  34
hsa04510                                                Focal adhesion 203  58
hsa05152                                                  Tuberculosis 182  53
                 P.DE
hsa04145 1.448157e-20
hsa04514 1.302066e-18
hsa04820 7.195970e-16
hsa04061 3.083983e-15
hsa04060 1.024378e-14
hsa04640 1.746575e-14
hsa05416 1.276480e-13
hsa05323 3.360380e-13
hsa04512 7.776478e-13
hsa05150 1.085616e-12
hsa05140 9.630267e-12
hsa04659 1.390457e-11
hsa05332 4.151450e-11
hsa05330 6.218226e-11
hsa05200 7.456089e-11
hsa05166 1.424277e-10
hsa04940 1.819356e-10
hsa04610 3.362602e-10
hsa04510 3.995235e-10
hsa05152 1.089030e-09</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform over-representation analyses for KEGG pathways</span></span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>kegg_gtex <span class="ot">&lt;-</span> <span class="fu">kegga</span>(up_gtex_ID, <span class="at">species=</span><span class="st">"Hs"</span>)</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the list of top KEGG pathwayss enriched for genes highly expressed in gtex</span></span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a><span class="fu">topKEGG</span>(kegg_gtex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                          Pathway    N DE         P.DE
hsa04972                     Pancreatic secretion  106 26 7.229222e-22
hsa04974         Protein digestion and absorption  105 20 9.763117e-15
hsa00260 Glycine, serine and threonine metabolism   41  8 1.065267e-06
hsa00670                One carbon pool by folate   39  7 9.262276e-06
hsa04975             Fat digestion and absorption   43  7 1.815797e-05
hsa04080  Neuroactive ligand-receptor interaction  370 20 3.826473e-05
hsa00500            Starch and sucrose metabolism   40  6 1.184599e-04
hsa04950     Maturity onset diabetes of the young   26  5 1.321104e-04
hsa04973    Carbohydrate digestion and absorption   52  6 5.167809e-04
hsa04970                       Salivary secretion   97  8 6.310693e-04
hsa01230              Biosynthesis of amino acids   75  7 6.587305e-04
hsa01100                       Metabolic pathways 1564 48 7.548163e-04
hsa00330          Arginine and proline metabolism   50  5 2.906340e-03
hsa00270       Cysteine and methionine metabolism   52  5 3.454184e-03
hsa04971                   Gastric acid secretion   76  6 3.762657e-03
hsa00750                    Vitamin B6 metabolism    6  2 5.500857e-03
hsa04978                       Mineral absorption   61  5 6.861599e-03
hsa00561                  Glycerolipid metabolism   65  5 8.946712e-03
hsa04964  Proximal tubule bicarbonate reclamation   23  3 9.976761e-03
hsa04020                Calcium signaling pathway  254 11 1.169612e-02</code></pre>
</div>
</div>
</section>
<section id="gene-regulatory-network-analysis-with-panda" class="level3">
<h3 class="anchored" data-anchor-id="gene-regulatory-network-analysis-with-panda">Gene Regulatory Network Analysis with PANDA</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Filtering based on PANDA requirements</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>zero_na_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assay</span>(gtex_upd, <span class="st">"logtpm"</span>), <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="fu">length</span>(x[(<span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x<span class="sc">!=</span><span class="dv">0</span>) ]))</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(gtex_upd)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 21205   349</code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a>expr_gtex <span class="ot">&lt;-</span> <span class="fu">assay</span>(gtex_upd, <span class="st">"logtpm"</span>)[zero_na_counts <span class="sc">&gt;</span> <span class="dv">20</span>,]</span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr_gtex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 21204   349</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>zero_na_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assay</span>(tcga_upd, <span class="st">"logtpm"</span>), <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="fu">length</span>(x[(<span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x<span class="sc">!=</span><span class="dv">0</span>) ]))</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a>expr_tcga <span class="ot">&lt;-</span> <span class="fu">assay</span>(tcga_upd, <span class="st">"logtpm"</span>)[zero_na_counts <span class="sc">&gt;</span> <span class="dv">20</span>,]</span>
<span id="cb94-3"><a href="#cb94-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(expr_tcga)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 21204   168</code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(expr_tcga) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">row.names</span>(expr_tcga))</span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(expr_gtex) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">row.names</span>(expr_gtex))</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(expr_tcga, file = "tcga.txt", sep = "\t", quote = FALSE, row.names = TRUE)</span></span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(expr_gtex, file = "gtex.txt", sep = "\t", quote = FALSE, row.names = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Load Panda output</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>tcga_network_full <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"output_panda_lcl_tcga.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>tcga_network <span class="ot">&lt;-</span> tcga_network_full[<span class="fu">order</span>(tcga_network_full<span class="sc">$</span>force, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>tcga_network <span class="ot">&lt;-</span> tcga_network[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>,]</span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>tcga_network <span class="ot">&lt;-</span> <span class="fu">subset</span>(tcga_network, <span class="at">select=</span><span class="sc">-</span>motif)</span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>tcga_network<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(tcga_network<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,tcga_network<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>hgnc_symbol[<span class="fu">match</span>(tcga_network<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>tcga_network</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              tf            gene    force
19118282  ZNF341 ENSG00000268836 20.65815
20187077 ZNF385D ENSG00000277368 17.04131
19050497 ZNF385D ENSG00000268129 15.36688
15491207 ZNF385D      ZNF687-AS1 15.36162
9286876  ZNF385D          SPACA9 14.99284
13452342 ZNF385D         TMEM240 14.84384
16968986   CEBPB     ALDH1L1-AS1 14.42439
2588033  ZNF385D           RPS16 14.23371
8493264  ZNF385D           SH2B2 14.19156
16968989   CEBPG     ALDH1L1-AS1 14.14611
1003800  ZNF385D           FAM3A 14.14011
5315825  ZNF385D           KCNC3 14.13916
7961863  ZNF385D        SLC25A28 14.10392
10790352 ZNF385D           DDIT3 13.97960
4084530  ZNF385D         EEF2KMT 13.95834
20822533   GATA2         MIR6789 13.92271
16968987   CEBPD     ALDH1L1-AS1 13.88717
7433453  ZNF385D            MTG1 13.86176
9701628  ZNF385D          ZNF598 13.72654
2236092  ZNF385D           UBL4A 13.70463
15180510   GATA2       LINC01315 13.70072
8378609  ZNF385D        FAM86C2P 13.62263
19847919   TIGD2 ENSG00000274015 13.61669
5503261  ZNF385D           ZBED3 13.60035
9285879  ZNF385D             AK8 13.52205
6950905  ZNF385D            GUK1 13.46038
5611934  ZNF385D           UNC79 13.45794
15952818 ZNF385D ENSG00000236528 13.44359
13032427   TIGD2          RPL10A 13.34460
1117458  ZNF385D          VPS9D1 13.19492
13857676    MTF2          FAM24B 13.16946
9627220    GATA2        CDC42EP5 13.15499
19058840   GATA2 ENSG00000268199 13.13959
18631127   GATA2 ENSG00000264701 13.13150
20186898   TIGD1 ENSG00000277368 13.03930
11215074 ZNF385D            TAF7 13.03298
3702049    GATA2           REEP6 13.00445
13470655   GATA2           SAP25 12.99614
15328517   TIGD1       LINC00852 12.98910
727998     GATA2           RPL18 12.93356
16968985   CEBPA     ALDH1L1-AS1 12.91718
15491056   ZBED1      ZNF687-AS1 12.88998
18911763   ZBED1 ENSG00000267328 12.87413
5234296    CEBPB         ATP5IF1 12.86722
10051800   CEBPB           PUSL1 12.83327
20636094   GATA2 ENSG00000282413 12.76965
11328732 ZNF385D           PUF60 12.69753
10933920 ZNF385D           KCMF1 12.68566
3274966  ZNF385D          SCNN1A 12.65498
13922926 ZNF385D           SIPA1 12.64944
4933974  ZNF385D            RNF6 12.63686
11667082   GATA2          GDPGP1 12.62206
5234299    CEBPG         ATP5IF1 12.61854
5177242  ZNF385D          ZSWIM6 12.58080
19331876   CEBPB ENSG00000270640 12.57787
8260963  ZNF385D          ZNF276 12.51335
7788610    CEBPB            CLGN 12.51092
4483558    CEBPG            EBPL 12.50283
5472354  ZNF385D            PCNA 12.49854
11751460 ZNF385D            LIN9 12.49541
10051803   CEBPG           PUSL1 12.48460
12227029 ZNF385D            PTMA 12.45710
10998095   GATA2            DPP7 12.45680
8491637    GATA2          ALKBH4 12.44868
19959761 ZNF385D            ZNF2 12.44015
17805244 ZNF385D       ITFG2-AS1 12.43319
19636253  LIN28B ENSG00000272836 12.43152
6861175  ZNF385D        METTL25B 12.42937
1820343  ZNF385D            CRKL 12.41647
17579922 ZNF385D            &lt;NA&gt; 12.40734
5234297    CEBPD         ATP5IF1 12.37413
18111144   TIGD1       MBTPS1-DT 12.37238
19331879   CEBPG ENSG00000270640 12.36974
14104202   TIGD2           AS3MT 12.34942
6984803  ZNF385D          ZNF514 12.29842
14702311   SMAD5            &lt;NA&gt; 12.28943
8643039    CEBPB          DIRAS3 12.28670
12128551   CEBPB          CLCNKA 12.28447
10051801   CEBPD           PUSL1 12.25941
15134666   GMEB1     PRKAR1B-AS2 12.24505
17841503   GATA2      RHOXF1-AS1 12.24308
20837488   GATA2         MIR5090 12.23889
5716986    GATA2          CYP2J2 12.22191
14861756    PHF1         RPS28P7 12.20143
19731183  SOHLH2 ENSG00000273192 12.18642
2343993    CEBPB             MPG 12.17170
20213366   GATA2 ENSG00000277654 12.16542
2542538    GATA2         PLEKHJ1 12.16478
2564472    GATA2        C19orf53 12.15414
7788613    CEBPG            CLGN 12.14899
17526451   GATA2 ENSG00000255663 12.07788
6230299    CEBPB          SPTBN5 12.04378
12357987   FOXN1            HBA2 12.02327
13549402   FOXN1            HBA1 12.01821
4817325  ZNF385D          CAPNS1 12.01302
8643042    CEBPG          DIRAS3 12.01061
13569810   TIGD2         RNVU1-7 12.00985
3703046    GATA2           PCSK4 11.99914
19331877   CEBPD ENSG00000270640 11.99585
12932905 ZNF385D            PEG3 11.98890
18073734     ERG ENSG00000259827 11.97640
19887347   GATA2 ENSG00000274422 11.95671
14808629   GATA2         IGBP1P1 11.95559
17456669    GCM2 ENSG00000255173 11.95321
12207386     ERG            MT1X 11.93829
7788611    CEBPD            CLGN 11.93047
17457152  ZBTB37 ENSG00000255173 11.92837
12399359   ZBED1           RPL14 11.92082
5709640  ZNF385D          PIWIL4 11.92059
1929241    CEBPB          SYNGR1 11.89917
17682980   GATA2 ENSG00000257199 11.89709
11346678 ZNF385D           EXOC3 11.88569
2343996    CEBPG             MPG 11.87168
17141063   TIGD1       RNA5SP392 11.87076
19240746   TIGD2 ENSG00000269918 11.84267
5174251  ZNF385D         NDUFA10 11.83623
5678958    CEBPB             FST 11.81863
19196905   ZBED1 ENSG00000269547 11.79752
5296113    CEBPG            PRR7 11.79578
8220453    GATA2            H4C8 11.79408
6230302    CEBPG          SPTBN5 11.79024
17576752   TIGD1 ENSG00000256116 11.78085
6850057    ZBED1         NECTIN4 11.75535
14702310   SMAD3            &lt;NA&gt; 11.74582
17263101   CEBPB       RBPMS-AS1 11.74484
9147145    ZBED1            FMC1 11.73065
19741715     ERG       TRABD-AS1 11.71930
8643040    CEBPD          DIRAS3 11.71570
12476279 ZNF385D         BLOC1S2 11.70584
957938   ZNF385D           DGCR2 11.70232
3255026  ZNF385D           ITFG2 11.69861
13675025   FOXN2           MT-TD 11.69569
8498616    GATA2           LRWD1 11.69191
9319599    TIGD2           ISCA2 11.68549
4900076  ZNF385D           AUNIP 11.67179
12128552   CEBPD          CLCNKA 11.64899
1929244    CEBPG          SYNGR1 11.64673
2499300  ZNF385D        LEPROTL1 11.64299
984857   ZNF385D          MGAT4A 11.64181
13943233   GATA2        GTF3C6P2 11.63268
2343994    CEBPD             MPG 11.61855
8745733    CEBPG           SGPP2 11.61385
16347452   TIGD2 ENSG00000241255 11.61170
13010026   FOXN2          MT-CO2 11.61148
17456668    GCM1 ENSG00000255173 11.60465
4002776  ZNF385D          MAN1C1 11.60179
6864166  ZNF385D         ISG20L2 11.58022
19777607     ERG ENSG00000273394 11.57679
6230300    CEBPD          SPTBN5 11.57332
9869826     RFX7          RNF181 11.57274
17263104   CEBPG       RBPMS-AS1 11.53601
6278315    GMEB1           CH25H 11.51049
19741935  LIN28B       TRABD-AS1 11.50819
13890844   TIGD2        RPL7AP34 11.49935
13357994   GATA2            LST1 11.49630
1525456    CEBPB         SIGLEC1 11.48619
3920727     RFX7          HPCAL4 11.48067
20893655    RFX7         MIR6733 11.47975
9523165  ZNF385D            MBD6 11.47798
15131642   FOXN2        MTCO2P12 11.46474
1902097  ZNF385D            SBF1 11.46102
4459402  ZNF385D           RBM19 11.45858
19663207    MTF2 ENSG00000272927 11.43298
9177431    CEBPB           NFIL3 11.43191
13674028   FOXN2          MT-TS1 11.42960
2904735     PHF1           PALD1 11.42717
8515900     RFX7          LRRC56 11.42395
727001     GATA2           SPHK2 11.41886
11768713    ETS1            COA3 11.40601
3100313    TIGD2           RPL34 11.40582
9107641    CEBPB             SHH 11.40410
16009303    PHF1         RPS28P4 11.40296
1929242    CEBPD          SYNGR1 11.39887
2546526    GATA2           SF3A2 11.39704
5678961    CEBPG             FST 11.39536
5234295    CEBPA         ATP5IF1 11.39311
16712829     ERG ENSG00000245261 11.39153
16751120    RFX7            H2AJ 11.38534
15539332    E2F2           PHGR1 11.37953
8108939   LIN28B         TMEM171 11.37900
6890088  ZNF385D          ANP32E 11.36745
11094734     ERG           SOX12 11.36237
10051799   CEBPA           PUSL1 11.31466
8787676      ERG            PMVK 11.31461
5847593    GATA2            LTV1 11.31323
1554963    TIGD2           RPLP0 11.31246
6712327     RFX7            PNMT 11.31099
19970116   GMEB1 ENSG00000275180 11.30341
6754317    TIGD1          GEMIN7 11.30121
9377603  ZNF385D            CCT2 11.29931
12422808   GATA2          NDUFA4 11.29752
1703694  ZNF385D          SEC22C 11.27121
12509180 ZNF385D          SUPT5H 11.26880
11419684   CEBPB            ST20 11.26837
5678959    CEBPD             FST 11.26198
15781563   CENPB ENSG00000235078 11.26106
17263102   CEBPD       RBPMS-AS1 11.25144
1821340  ZNF385D           LZTR1 11.22596
8360663  ZNF385D        ARHGAP35 11.21951
1525459    CEBPG         SIGLEC1 11.21918</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This is the TCGA network</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> tcga_network</span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(edges) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"from"</span>, <span class="st">"to"</span>, <span class="st">"value"</span>)</span>
<span id="cb99-4"><a href="#cb99-4" aria-hidden="true" tabindex="-1"></a>nodes       <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))), </span>
<span id="cb99-5"><a href="#cb99-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">label=</span><span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))))</span>
<span id="cb99-6"><a href="#cb99-6" aria-hidden="true" tabindex="-1"></a>nodes<span class="sc">$</span>group <span class="ot">=</span> <span class="fu">ifelse</span>(nodes<span class="sc">$</span>id <span class="sc">%in%</span> edges<span class="sc">$</span>from, <span class="st">"TF"</span>, <span class="st">"gene"</span>)</span>
<span id="cb99-7"><a href="#cb99-7" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visNetwork</span>(nodes, edges, <span class="at">width =</span> <span class="st">"100%"</span>)</span>
<span id="cb99-8"><a href="#cb99-8" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"TF"</span>, <span class="at">shape =</span> <span class="st">"triangle"</span>,</span>
<span id="cb99-9"><a href="#cb99-9" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"purple"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb99-10"><a href="#cb99-10" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"gene"</span>, <span class="at">shape =</span> <span class="st">"dot"</span>,       </span>
<span id="cb99-11"><a href="#cb99-11" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"teal"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb99-12"><a href="#cb99-12" aria-hidden="true" tabindex="-1"></a><span class="fu">visLegend</span>(net, <span class="at">main=</span><span class="st">"Legend"</span>, <span class="at">position=</span><span class="st">"right"</span>, <span class="at">ncol=</span><span class="dv">1</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="htmlwidget-b1c8e18f91a9a16a1fed" style="width:100%;height:480px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-b1c8e18f91a9a16a1fed">{"x":{"nodes":{"id":["ZNF341","ZNF385D","CEBPB","CEBPG","GATA2","CEBPD","TIGD2","MTF2","TIGD1","CEBPA","ZBED1","LIN28B","SMAD5","GMEB1","PHF1","SOHLH2","FOXN1","ERG","GCM2","ZBTB37","SMAD3","FOXN2","GCM1","RFX7","ETS1","E2F2","CENPB","ENSG00000268836","ENSG00000277368","ENSG00000268129","ZNF687-AS1","SPACA9","TMEM240","ALDH1L1-AS1","RPS16","SH2B2","FAM3A","KCNC3","SLC25A28","DDIT3","EEF2KMT","MIR6789","MTG1","ZNF598","UBL4A","LINC01315","FAM86C2P","ENSG00000274015","ZBED3","AK8","GUK1","UNC79","ENSG00000236528","RPL10A","VPS9D1","FAM24B","CDC42EP5","ENSG00000268199","ENSG00000264701","TAF7","REEP6","SAP25","LINC00852","RPL18","ENSG00000267328","ATP5IF1","PUSL1","ENSG00000282413","PUF60","KCMF1","SCNN1A","SIPA1","RNF6","GDPGP1","ZSWIM6","ENSG00000270640","ZNF276","CLGN","EBPL","PCNA","LIN9","PTMA","DPP7","ALKBH4","ZNF2","ITFG2-AS1","ENSG00000272836","METTL25B","CRKL",null,"MBTPS1-DT","AS3MT","ZNF514","DIRAS3","CLCNKA","PRKAR1B-AS2","RHOXF1-AS1","MIR5090","CYP2J2","RPS28P7","ENSG00000273192","MPG","ENSG00000277654","PLEKHJ1","C19orf53","ENSG00000255663","SPTBN5","HBA2","HBA1","CAPNS1","RNVU1-7","PCSK4","PEG3","ENSG00000259827","ENSG00000274422","IGBP1P1","ENSG00000255173","MT1X","RPL14","PIWIL4","SYNGR1","ENSG00000257199","EXOC3","RNA5SP392","ENSG00000269918","NDUFA10","FST","ENSG00000269547","PRR7","H4C8","ENSG00000256116","NECTIN4","RBPMS-AS1","FMC1","TRABD-AS1","BLOC1S2","DGCR2","ITFG2","MT-TD","LRWD1","ISCA2","AUNIP","LEPROTL1","MGAT4A","GTF3C6P2","SGPP2","ENSG00000241255","MT-CO2","MAN1C1","ISG20L2","ENSG00000273394","RNF181","CH25H","RPL7AP34","LST1","SIGLEC1","HPCAL4","MIR6733","MBD6","MTCO2P12","SBF1","RBM19","ENSG00000272927","NFIL3","MT-TS1","PALD1","LRRC56","SPHK2","COA3","RPL34","SHH","RPS28P4","SF3A2","ENSG00000245261","H2AJ","PHGR1","TMEM171","ANP32E","SOX12","PMVK","LTV1","RPLP0","PNMT","ENSG00000275180","GEMIN7","CCT2","NDUFA4","SEC22C","SUPT5H","ST20","ENSG00000235078","LZTR1","ARHGAP35"],"label":["ZNF341","ZNF385D","CEBPB","CEBPG","GATA2","CEBPD","TIGD2","MTF2","TIGD1","CEBPA","ZBED1","LIN28B","SMAD5","GMEB1","PHF1","SOHLH2","FOXN1","ERG","GCM2","ZBTB37","SMAD3","FOXN2","GCM1","RFX7","ETS1","E2F2","CENPB","ENSG00000268836","ENSG00000277368","ENSG00000268129","ZNF687-AS1","SPACA9","TMEM240","ALDH1L1-AS1","RPS16","SH2B2","FAM3A","KCNC3","SLC25A28","DDIT3","EEF2KMT","MIR6789","MTG1","ZNF598","UBL4A","LINC01315","FAM86C2P","ENSG00000274015","ZBED3","AK8","GUK1","UNC79","ENSG00000236528","RPL10A","VPS9D1","FAM24B","CDC42EP5","ENSG00000268199","ENSG00000264701","TAF7","REEP6","SAP25","LINC00852","RPL18","ENSG00000267328","ATP5IF1","PUSL1","ENSG00000282413","PUF60","KCMF1","SCNN1A","SIPA1","RNF6","GDPGP1","ZSWIM6","ENSG00000270640","ZNF276","CLGN","EBPL","PCNA","LIN9","PTMA","DPP7","ALKBH4","ZNF2","ITFG2-AS1","ENSG00000272836","METTL25B","CRKL",null,"MBTPS1-DT","AS3MT","ZNF514","DIRAS3","CLCNKA","PRKAR1B-AS2","RHOXF1-AS1","MIR5090","CYP2J2","RPS28P7","ENSG00000273192","MPG","ENSG00000277654","PLEKHJ1","C19orf53","ENSG00000255663","SPTBN5","HBA2","HBA1","CAPNS1","RNVU1-7","PCSK4","PEG3","ENSG00000259827","ENSG00000274422","IGBP1P1","ENSG00000255173","MT1X","RPL14","PIWIL4","SYNGR1","ENSG00000257199","EXOC3","RNA5SP392","ENSG00000269918","NDUFA10","FST","ENSG00000269547","PRR7","H4C8","ENSG00000256116","NECTIN4","RBPMS-AS1","FMC1","TRABD-AS1","BLOC1S2","DGCR2","ITFG2","MT-TD","LRWD1","ISCA2","AUNIP","LEPROTL1","MGAT4A","GTF3C6P2","SGPP2","ENSG00000241255","MT-CO2","MAN1C1","ISG20L2","ENSG00000273394","RNF181","CH25H","RPL7AP34","LST1","SIGLEC1","HPCAL4","MIR6733","MBD6","MTCO2P12","SBF1","RBM19","ENSG00000272927","NFIL3","MT-TS1","PALD1","LRRC56","SPHK2","COA3","RPL34","SHH","RPS28P4","SF3A2","ENSG00000245261","H2AJ","PHGR1","TMEM171","ANP32E","SOX12","PMVK","LTV1","RPLP0","PNMT","ENSG00000275180","GEMIN7","CCT2","NDUFA4","SEC22C","SUPT5H","ST20","ENSG00000235078","LZTR1","ARHGAP35"],"group":["TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene"]},"edges":{"from":["ZNF341","ZNF385D","ZNF385D","ZNF385D","ZNF385D","ZNF385D","CEBPB","ZNF385D","ZNF385D","CEBPG","ZNF385D","ZNF385D","ZNF385D","ZNF385D","ZNF385D","GATA2","CEBPD","ZNF385D","ZNF385D","ZNF385D","GATA2","ZNF385D","TIGD2","ZNF385D","ZNF385D","ZNF385D","ZNF385D","ZNF385D","TIGD2","ZNF385D","MTF2","GATA2","GATA2","GATA2","TIGD1","ZNF385D","GATA2","GATA2","TIGD1","GATA2","CEBPA","ZBED1","ZBED1","CEBPB","CEBPB","GATA2","ZNF385D","ZNF385D","ZNF385D","ZNF385D","ZNF385D","GATA2","CEBPG","ZNF385D","CEBPB","ZNF385D","CEBPB","CEBPG","ZNF385D","ZNF385D","CEBPG","ZNF385D","GATA2","GATA2","ZNF385D","ZNF385D","LIN28B","ZNF385D","ZNF385D","ZNF385D","CEBPD","TIGD1","CEBPG","TIGD2","ZNF385D","SMAD5","CEBPB","CEBPB","CEBPD","GMEB1","GATA2","GATA2","GATA2","PHF1","SOHLH2","CEBPB","GATA2","GATA2","GATA2","CEBPG","GATA2","CEBPB","FOXN1","FOXN1","ZNF385D","CEBPG","TIGD2","GATA2","CEBPD","ZNF385D","ERG","GATA2","GATA2","GCM2","ERG","CEBPD","ZBTB37","ZBED1","ZNF385D","CEBPB","GATA2","ZNF385D","CEBPG","TIGD1","TIGD2","ZNF385D","CEBPB","ZBED1","CEBPG","GATA2","CEBPG","TIGD1","ZBED1","SMAD3","CEBPB","ZBED1","ERG","CEBPD","ZNF385D","ZNF385D","ZNF385D","FOXN2","GATA2","TIGD2","ZNF385D","CEBPD","CEBPG","ZNF385D","ZNF385D","GATA2","CEBPD","CEBPG","TIGD2","FOXN2","GCM1","ZNF385D","ZNF385D","ERG","CEBPD","RFX7","CEBPG","GMEB1","LIN28B","TIGD2","GATA2","CEBPB","RFX7","RFX7","ZNF385D","FOXN2","ZNF385D","ZNF385D","MTF2","CEBPB","FOXN2","PHF1","RFX7","GATA2","ETS1","TIGD2","CEBPB","PHF1","CEBPD","GATA2","CEBPG","CEBPA","ERG","RFX7","E2F2","LIN28B","ZNF385D","ERG","CEBPA","ERG","GATA2","TIGD2","RFX7","GMEB1","TIGD1","ZNF385D","GATA2","ZNF385D","ZNF385D","CEBPB","CEBPD","CENPB","CEBPD","ZNF385D","ZNF385D","CEBPG"],"to":["ENSG00000268836","ENSG00000277368","ENSG00000268129","ZNF687-AS1","SPACA9","TMEM240","ALDH1L1-AS1","RPS16","SH2B2","ALDH1L1-AS1","FAM3A","KCNC3","SLC25A28","DDIT3","EEF2KMT","MIR6789","ALDH1L1-AS1","MTG1","ZNF598","UBL4A","LINC01315","FAM86C2P","ENSG00000274015","ZBED3","AK8","GUK1","UNC79","ENSG00000236528","RPL10A","VPS9D1","FAM24B","CDC42EP5","ENSG00000268199","ENSG00000264701","ENSG00000277368","TAF7","REEP6","SAP25","LINC00852","RPL18","ALDH1L1-AS1","ZNF687-AS1","ENSG00000267328","ATP5IF1","PUSL1","ENSG00000282413","PUF60","KCMF1","SCNN1A","SIPA1","RNF6","GDPGP1","ATP5IF1","ZSWIM6","ENSG00000270640","ZNF276","CLGN","EBPL","PCNA","LIN9","PUSL1","PTMA","DPP7","ALKBH4","ZNF2","ITFG2-AS1","ENSG00000272836","METTL25B","CRKL",null,"ATP5IF1","MBTPS1-DT","ENSG00000270640","AS3MT","ZNF514",null,"DIRAS3","CLCNKA","PUSL1","PRKAR1B-AS2","RHOXF1-AS1","MIR5090","CYP2J2","RPS28P7","ENSG00000273192","MPG","ENSG00000277654","PLEKHJ1","C19orf53","CLGN","ENSG00000255663","SPTBN5","HBA2","HBA1","CAPNS1","DIRAS3","RNVU1-7","PCSK4","ENSG00000270640","PEG3","ENSG00000259827","ENSG00000274422","IGBP1P1","ENSG00000255173","MT1X","CLGN","ENSG00000255173","RPL14","PIWIL4","SYNGR1","ENSG00000257199","EXOC3","MPG","RNA5SP392","ENSG00000269918","NDUFA10","FST","ENSG00000269547","PRR7","H4C8","SPTBN5","ENSG00000256116","NECTIN4",null,"RBPMS-AS1","FMC1","TRABD-AS1","DIRAS3","BLOC1S2","DGCR2","ITFG2","MT-TD","LRWD1","ISCA2","AUNIP","CLCNKA","SYNGR1","LEPROTL1","MGAT4A","GTF3C6P2","MPG","SGPP2","ENSG00000241255","MT-CO2","ENSG00000255173","MAN1C1","ISG20L2","ENSG00000273394","SPTBN5","RNF181","RBPMS-AS1","CH25H","TRABD-AS1","RPL7AP34","LST1","SIGLEC1","HPCAL4","MIR6733","MBD6","MTCO2P12","SBF1","RBM19","ENSG00000272927","NFIL3","MT-TS1","PALD1","LRRC56","SPHK2","COA3","RPL34","SHH","RPS28P4","SYNGR1","SF3A2","FST","ATP5IF1","ENSG00000245261","H2AJ","PHGR1","TMEM171","ANP32E","SOX12","PUSL1","PMVK","LTV1","RPLP0","PNMT","ENSG00000275180","GEMIN7","CCT2","NDUFA4","SEC22C","SUPT5H","ST20","FST","ENSG00000235078","RBPMS-AS1","LZTR1","ARHGAP35","SIGLEC1"],"value":[20.6581542492277,17.04131173576261,15.36687617673946,15.36161997217532,14.99283721727229,14.84383626148754,14.42438964257975,14.23370515577426,14.19156239895525,14.14610565719413,14.14010600610986,14.13916140706564,14.10392376866946,13.97960291199536,13.95833981800742,13.92271427427738,13.88716815736569,13.86175859712736,13.7265362267121,13.70462656929596,13.70072266496681,13.62262821548818,13.61668572290935,13.60034772479087,13.5220505424936,13.46038228734175,13.45794236324873,13.44358955468046,13.34459965525052,13.19491825328692,13.16946346262544,13.15499215751269,13.13958650604603,13.13149854771268,13.03929608242623,13.03297567813825,13.00445074689371,12.99614156050363,12.98909911108901,12.93356233475646,12.91718202149886,12.88997995622607,12.87413392936448,12.86722184148662,12.83327150307271,12.76964985204965,12.69752735326861,12.68565550265487,12.6549812487219,12.64943619573705,12.63686241729486,12.6220596087238,12.61853513341921,12.58080061454164,12.57787141892025,12.51335210781688,12.51091982085104,12.50282733974783,12.4985422444035,12.49540753155505,12.48460051411428,12.45710359443178,12.45679928563019,12.44868320474751,12.44014523074201,12.43318519911346,12.43152499010393,12.42937419544947,12.4164686005241,12.40734397205194,12.37412872106604,12.37238305074737,12.3697415417749,12.34941710811761,12.29842388433598,12.28942643196797,12.28670161345057,12.2844663627099,12.25940604460846,12.24504803400411,12.24307668451121,12.23888958751188,12.22191043342865,12.20142870494964,12.18641555504738,12.17169890903632,12.16542056624493,12.16478461037126,12.15413601233149,12.14899433651012,12.07788378513131,12.04378204609133,12.02326716850278,12.01820920136929,12.01301813726269,12.01060875464362,12.00984985879478,11.99913804552834,11.9958451769014,11.98889615736144,11.97640043729909,11.95671366859689,11.95559019746965,11.9532052835559,11.93828604046391,11.93046952792504,11.92836720395148,11.92082219561498,11.92059239944701,11.89917007848286,11.8970913908635,11.88568996489186,11.8716824957544,11.87075575344981,11.84266741711052,11.83622566115056,11.81862751144447,11.79751965265742,11.795784379568,11.7940790257722,11.79024144424963,11.78085413879781,11.75535365085452,11.74581809364328,11.7448406676957,11.73065470569845,11.71930412537788,11.71569849749884,11.70583875908103,11.70231629255771,11.69860582261457,11.69569433053071,11.69191494456209,11.6854883613566,11.6717895077057,11.64899065702563,11.64672961985112,11.64299114284521,11.64180835041184,11.63267892623685,11.61855107079993,11.61384650274564,11.61169659750114,11.61148229592304,11.60465236253008,11.60178868704396,11.58022063178663,11.576794703286,11.57332373237607,11.57273616606763,11.53601036050324,11.51048622542515,11.50818843213262,11.49934585202164,11.49630016617198,11.4861887635401,11.48066859582056,11.47974927242362,11.477979776906,11.46473658910211,11.46101840085419,11.45857957050278,11.4329789750504,11.4319090275543,11.42959763987391,11.42717134502891,11.42394927268988,11.41886453228562,11.40601396938152,11.405824224377,11.40410206533571,11.4029613327238,11.39887333989461,11.39704167366811,11.39536151848291,11.39311291456493,11.39152701398591,11.38534363633552,11.37953416194062,11.37900440864855,11.36744695127172,11.36236670659284,11.31465574852735,11.31460547422557,11.31323342184269,11.31245869785508,11.3109878074875,11.30340549021044,11.30120588364773,11.29930967475069,11.29751805805476,11.27121373641879,11.26880099146953,11.26836913289084,11.26197903743133,11.26105739046756,11.25143803320732,11.22596053091576,11.21951021123453,11.21917832754919]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false},"groups":{"TF":{"shape":"triangle","color":{"background":"purple","border":"black"}},"useDefaultGroups":true,"gene":{"shape":"dot","color":{"background":"teal","border":"black"}}}},"groups":["TF","gene"],"width":"100%","height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","legend":{"width":0.2,"useGroups":true,"position":"right","ncol":1,"stepX":100,"stepY":100,"zoom":true,"main":{"text":"Legend","style":"font-family:Georgia, Times New Roman, Times, serif;font-weight:bold;font-size:14px;text-align:center;"}}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>gtex_network_full <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"output_panda_lcl_gtex.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a><span class="do">##compute graph difference</span></span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>diffnet <span class="ot">&lt;-</span> tcga_network_full</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>diffnet<span class="sc">$</span>force <span class="ot">&lt;-</span> tcga_network_full<span class="sc">$</span>force<span class="sc">-</span>gtex_network_full<span class="sc">$</span>force</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in tcga_network_full$force - gtex_network_full$force: longer object
length is not a multiple of shorter object length</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a>diffnet <span class="ot">&lt;-</span> <span class="fu">subset</span>(diffnet, <span class="at">select=</span><span class="sc">-</span>motif)</span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>diffnet<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(diffnet<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,diffnet<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>hgnc_symbol[<span class="fu">match</span>(diffnet<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(diffnet,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       tf   gene      force
1    ALX1 TSPAN6 -0.3553039
2    ALX3 TSPAN6  0.3594796
3    ALX4 TSPAN6 -0.3507705
4      AR TSPAN6  0.3151534
5   ARGFX TSPAN6 -0.1770416
6  ARID3A TSPAN6 -0.5754157
7  ARID5B TSPAN6 -0.4177519
8    ARNT TSPAN6 -0.2132221
9   ARNT2 TSPAN6 -0.2162359
10  ARNTL TSPAN6 -0.1640215</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This plots the edges that change the most in cancer vs normal</span></span>
<span id="cb104-2"><a href="#cb104-2" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> diffnet</span>
<span id="cb104-3"><a href="#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(edges) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"from"</span>, <span class="st">"to"</span>, <span class="st">"value"</span>)</span>
<span id="cb104-4"><a href="#cb104-4" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>arrows <span class="ot">=</span> <span class="st">"to"</span>   </span>
<span id="cb104-5"><a href="#cb104-5" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>color  <span class="ot">=</span> <span class="fu">ifelse</span>(edges<span class="sc">$</span>value <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"green"</span>, <span class="st">"red"</span>)</span>
<span id="cb104-6"><a href="#cb104-6" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>value  <span class="ot">=</span> <span class="fu">abs</span>(edges<span class="sc">$</span>value)</span>
<span id="cb104-7"><a href="#cb104-7" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges[<span class="fu">order</span>(edges<span class="sc">$</span>value, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb104-8"><a href="#cb104-8" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>,]</span>
<span id="cb104-9"><a href="#cb104-9" aria-hidden="true" tabindex="-1"></a>nodes       <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))), </span>
<span id="cb104-10"><a href="#cb104-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">label=</span><span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))))</span>
<span id="cb104-11"><a href="#cb104-11" aria-hidden="true" tabindex="-1"></a>nodes<span class="sc">$</span>group <span class="ot">=</span> <span class="fu">ifelse</span>(nodes<span class="sc">$</span>id <span class="sc">%in%</span> edges<span class="sc">$</span>from, <span class="st">"TF"</span>, <span class="st">"gene"</span>)</span>
<span id="cb104-12"><a href="#cb104-12" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visNetwork</span>(nodes, edges, <span class="at">width =</span> <span class="st">"100%"</span>)</span>
<span id="cb104-13"><a href="#cb104-13" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"TF"</span>, <span class="at">shape =</span> <span class="st">"triangle"</span>,</span>
<span id="cb104-14"><a href="#cb104-14" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"purple"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb104-15"><a href="#cb104-15" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"gene"</span>, <span class="at">shape =</span> <span class="st">"dot"</span>,       </span>
<span id="cb104-16"><a href="#cb104-16" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"teal"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb104-17"><a href="#cb104-17" aria-hidden="true" tabindex="-1"></a><span class="fu">visLegend</span>(net, <span class="at">main=</span><span class="st">"Legend"</span>, <span class="at">position=</span><span class="st">"right"</span>, <span class="at">ncol=</span><span class="dv">1</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="htmlwidget-c2e4605cc32d730819c3" style="width:100%;height:480px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-c2e4605cc32d730819c3">{"x":{"nodes":{"id":["ZNF235","ZNF287","PRDM6","ZNF225","ZNF487","ZNF418","GATA2","ZNF879","ZNF98","ZNF334","ZFP28","ZNF182","ZNF570","ZNF613","ZNF492","IRF1","ZNF571","ZNF567","ZNF79","ZNF432","ZNF774","ZNF467","ZNF22","E2F5","SMAD5","ZNF396","ZNF443","FOXR2","ZNF181","ZNF770","RFX7","PAX7","SMAD3","ZNF25","HES7","MIR6513","MIR4775","MIR6789","ENSG00000283977","MIR6852","MIR6820","MIR147B","MIR6743","MIR6808","SCO2","MIR6764","ENSG00000284337","MIR6125","MIR922","BACE1-AS","MIR675","ENSG00000284057","MIR1304","MIR4800","ENSG00000284461","TPTEP2-CSNK1E","ENPP1","MIR939","MIR3655","MIR6845","BACE1","MIR6805","BCRP3","ENSG00000261270","MIR24-1","CNIH3","HERPUD1","MIR4763","MIR7112","SLC41A1","ENSG00000284292","ENSG00000261114","MIR4724","MIR5006","PACSIN2","MIR631","CEP164","MIR5090","MIR936","MIR4657","RNF214","MIR5001","MIR1909","ENSG00000229591","PLCE1","MIR6824","WNT5A-AS1","SH3RF2","MIR4647","MIR7847"],"label":["ZNF235","ZNF287","PRDM6","ZNF225","ZNF487","ZNF418","GATA2","ZNF879","ZNF98","ZNF334","ZFP28","ZNF182","ZNF570","ZNF613","ZNF492","IRF1","ZNF571","ZNF567","ZNF79","ZNF432","ZNF774","ZNF467","ZNF22","E2F5","SMAD5","ZNF396","ZNF443","FOXR2","ZNF181","ZNF770","RFX7","PAX7","SMAD3","ZNF25","HES7","MIR6513","MIR4775","MIR6789","ENSG00000283977","MIR6852","MIR6820","MIR147B","MIR6743","MIR6808","SCO2","MIR6764","ENSG00000284337","MIR6125","MIR922","BACE1-AS","MIR675","ENSG00000284057","MIR1304","MIR4800","ENSG00000284461","TPTEP2-CSNK1E","ENPP1","MIR939","MIR3655","MIR6845","BACE1","MIR6805","BCRP3","ENSG00000261270","MIR24-1","CNIH3","HERPUD1","MIR4763","MIR7112","SLC41A1","ENSG00000284292","ENSG00000261114","MIR4724","MIR5006","PACSIN2","MIR631","CEP164","MIR5090","MIR936","MIR4657","RNF214","MIR5001","MIR1909","ENSG00000229591","PLCE1","MIR6824","WNT5A-AS1","SH3RF2","MIR4647","MIR7847"],"group":["TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene"]},"edges":{"from":["ZNF235","ZNF287","PRDM6","ZNF235","ZNF225","ZNF487","ZNF418","ZNF287","ZNF418","GATA2","ZNF879","ZNF225","PRDM6","ZNF487","ZNF879","ZNF487","ZNF98","ZNF334","ZFP28","ZNF182","ZNF570","ZNF225","ZNF487","ZNF613","ZNF235","ZNF287","ZNF492","ZNF570","IRF1","ZNF571","ZNF567","PRDM6","ZNF235","ZNF182","ZNF225","PRDM6","ZNF487","ZNF567","PRDM6","ZNF182","IRF1","ZNF487","ZNF287","IRF1","PRDM6","ZNF79","ZNF235","ZNF432","ZNF418","ZNF418","ZNF225","ZNF492","ZNF235","ZNF613","ZNF567","ZNF774","ZNF467","ZNF487","ZNF334","ZNF774","PRDM6","ZNF235","ZFP28","ZNF287","ZNF79","ZNF287","ZNF334","ZNF235","ZNF235","ZNF487","ZNF79","ZNF879","ZNF235","ZNF432","ZNF571","ZNF432","ZNF487","ZNF235","ZNF22","ZNF879","ZNF879","ZNF287","ZNF98","ZNF334","ZNF492","ZNF287","ZNF418","ZNF225","ZFP28","E2F5","ZNF287","ZNF235","ZNF225","ZNF235","ZNF182","ZNF467","ZNF287","ZNF467","ZNF571","GATA2","ZNF235","ZNF571","ZNF235","ZNF487","ZNF235","ZNF418","ZNF98","ZNF879","PRDM6","ZNF225","GATA2","ZNF570","ZNF235","ZNF225","ZNF235","ZNF287","ZNF287","ZNF287","ZNF235","PRDM6","ZNF235","ZNF287","ZNF492","ZNF487","ZNF225","ZNF235","ZNF235","ZNF879","ZNF467","ZNF235","ZNF287","ZNF418","PRDM6","SMAD5","ZNF492","PRDM6","ZNF334","ZNF613","ZNF182","ZNF396","ZNF287","ZNF235","GATA2","ZNF287","ZNF287","ZNF235","ZNF235","ZNF287","ZNF418","ZNF443","ZNF235","FOXR2","ZNF287","ZNF487","ZNF235","ZNF22","ZNF181","ZNF287","ZNF287","ZNF225","ZNF418","ZNF770","RFX7","ZNF613","ZFP28","ZNF492","PAX7","ZNF287","ZNF879","ZNF487","ZNF225","ZNF182","PRDM6","ZNF182","ZNF287","ZNF418","SMAD3","ZNF235","ZNF235","PRDM6","ZNF235","IRF1","ZNF235","ZNF25","ZNF287","ZNF487","ZNF567","ZNF225","ZNF225","PRDM6","ZNF287","HES7","ZNF235","ZNF98","ZNF418","ZNF487","PRDM6","ZNF570","ZNF235","PRDM6"],"to":["MIR6513","MIR6513","MIR6513","MIR4775","MIR6513","MIR6513","MIR6513","MIR4775","MIR4775","MIR6789","MIR6513","MIR4775","MIR4775","MIR4775","MIR4775","ENSG00000283977","MIR6513","MIR6852","MIR4775","MIR6513","MIR6513","MIR6820","MIR6820","MIR6513","MIR147B","MIR147B","MIR6513","MIR4775","MIR6513","MIR6743","MIR6513","MIR147B","MIR6808","MIR4775","MIR147B","SCO2","MIR147B","MIR4775","MIR6852","MIR6820","MIR4775","MIR6764","MIR6808","ENSG00000283977","MIR6808","MIR4775","MIR6789","MIR6513","MIR6808","MIR147B","MIR6808","MIR6820","ENSG00000284337","MIR6820","MIR6820","MIR6513","ENSG00000283977","MIR6808","SCO2","MIR6125","MIR6789","MIR922","MIR147B","MIR6789","ENSG00000283977","ENSG00000284337","MIR6125","BACE1-AS","MIR675","ENSG00000284057","MIR6820","MIR6808","MIR1304","MIR4800","ENSG00000284461","TPTEP2-CSNK1E","MIR6789","ENPP1","MIR6513","MIR147B","MIR6789","MIR922","MIR147B","MIR939","MIR3655","MIR675","MIR675","ENSG00000284337","MIR6808","MIR6789","BACE1-AS","MIR6845","MIR675","BACE1","MIR147B","MIR6805","ENPP1","TPTEP2-CSNK1E","MIR6820","BCRP3","ENSG00000261270","MIR24-1","CNIH3","MIR675","HERPUD1","MIR922","MIR6808","MIR675","MIR922","MIR1304","MIR4763","MIR6808","MIR7112","MIR922","SLC41A1","MIR6845","CNIH3","ENSG00000261270","ENSG00000284292","BACE1-AS","ENSG00000261114","BACE1","MIR939","MIR922","MIR6845","MIR4724","MIR5006","MIR922","ENSG00000284461","PACSIN2","ENSG00000284292","MIR7112","MIR6845","MIR631","MIR147B","ENPP1","MIR3655","MIR147B","MIR6808","MIR6789","HERPUD1","CEP164","MIR5090","MIR4724","MIR936","MIR4657","MIR936","PACSIN2","MIR6845","MIR6513","RNF214","MIR4775","SLC41A1","MIR6845","MIR6125","MIR6820","MIR6743","MIR7112","ENSG00000261114","MIR7112","MIR1304","MIR6743","MIR5001","MIR6808","MIR922","MIR6808","MIR1909","MIR5006","MIR7112","BACE1-AS","BACE1-AS","MIR6789","CNIH3","ENSG00000284057","CEP164","MIR5006","MIR6845","ENSG00000229591","PLCE1","BACE1","MIR6824","MIR6808","WNT5A-AS1","MIR4775","SH3RF2","ENSG00000284292","MIR6789","ENSG00000284292","ENPP1","ENSG00000284292","MIR4657","MIR4775","SH3RF2","ENSG00000284337","BACE1-AS","ENPP1","MIR4647","ENSG00000284337","MIR7847","ENSG00000261270"],"value":[16.9778919416758,16.79744288846769,16.27817607408889,16.21818615233156,16.11134746483378,15.98019780214639,15.89982787061004,15.80794641643591,15.57932961432282,15.55496469844439,15.37189083165077,15.32984662628768,15.30841823651836,15.23671489449355,15.14696742569002,15.02231192499639,14.78849104272203,14.64356728249869,14.56551127678195,14.55235413291673,14.51676717724428,14.30698104421723,14.1828308149632,14.10336212306413,14.10112909366048,14.09478637405293,13.95695101984246,13.95361860470256,13.89827577026041,13.76348459176094,13.76338783193157,13.54211239974064,13.52781411289854,13.46009729315497,13.41773211874127,13.34314732688666,13.29330594043235,13.266874283822,13.26666758801993,13.26461055294551,13.21710742906717,13.20529400015537,13.18877912100491,13.16784308514531,13.11445154366005,13.10493390361194,13.04816257654395,13.02944245363626,12.96661017824572,12.96395656958157,12.96303249766668,12.93994909991817,12.9153703860358,12.87379883970169,12.84460759409316,12.80270319038928,12.7669928128055,12.75469332778103,12.71985430098198,12.68677935309676,12.6765544670344,12.64484070063575,12.62685880570366,12.61013517562284,12.6054231858909,12.60239912271808,12.57347050124367,12.57022241844409,12.55863387890949,12.55436799322696,12.55040043640954,12.54252572701133,12.53136534887793,12.53024078509965,12.52074878108458,12.51738274195212,12.48097433102517,12.45600092943321,12.44599094076087,12.43838328110294,12.42569457310532,12.40699448831368,12.39329228451408,12.37097839410622,12.35152244608132,12.34151531160212,12.33082401220947,12.31706346922532,12.30268035369933,12.29264282975326,12.26564357593491,12.25523643655212,12.24488365516036,12.22134847590596,12.2169191322153,12.211616080814,12.14578860518699,12.14120260201828,12.12847663748637,12.12299414444476,12.1153677768621,12.08252829823139,12.08122038332699,12.08043303050278,12.07445022906104,12.07294644119388,12.0652224689245,12.06290125903984,12.05037697350072,12.01468694272143,11.99947292377367,11.99903222316684,11.99501774789363,11.97516800475189,11.96448690608709,11.96310763988236,11.86664216214988,11.861389920293,11.85875803034719,11.85832698140598,11.85805471578736,11.85609884743276,11.84902980700881,11.84386874371908,11.83530989129567,11.79941442060298,11.79594020541526,11.79517167832395,11.79106614070259,11.7907732480385,11.77820574086966,11.77741371139561,11.77486755064892,11.75386093996946,11.73945877849673,11.73639072306738,11.73227116607752,11.72763994115477,11.72614250783121,11.724912599654,11.72170173887294,11.71787975144696,11.71343295711954,11.6961456463427,11.69096800022748,11.68556742354494,11.68466539614278,11.68073172512252,11.6744554132309,11.66669037411567,11.6639278369156,11.65934374071371,11.65216250253285,11.64112487122795,11.63959233174043,11.63452587217425,11.62140823077971,11.61286800280691,11.60138526380285,11.58372212447278,11.57963297599743,11.57893870828016,11.57456743389727,11.57393227652141,11.5697727883844,11.56161811694098,11.55038594020224,11.54761934586169,11.54705080583788,11.53975060088978,11.53498219121992,11.51627944106816,11.51491366700861,11.50346292134893,11.4920943046114,11.47886782769713,11.47426625373102,11.47340770752575,11.46527356511093,11.44094671998423,11.43224699747541,11.43220159568927,11.43103648819721,11.42820874620283,11.42675355310987,11.42563551829092,11.42409730661129,11.41925639043853,11.41779510020395,11.39791479500683,11.39746240064875,11.39696611938501,11.39342504230028,11.39327338361755,11.38910397520189,11.38434737348968,11.38003683490046,11.37143979872232,11.37138311013204,11.36085268110791],"arrows":["to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to"],"color":["green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","red","red","green","green","green","green","green","green","green","green","green","red","green","green","red","green","green","green","red","green","green","red","green","red","green","red","green","red","green","red","red","green","red","red","green","green","red","red","green","red","green","green","red","green","green","green","green","red","green","red","red","green","green","green","green","red","green","green","green","red","green","green","red","green","red","red","green","red","green","green","red","red","green","green","red","green","green","red","red","green","red","green","red","green","green","red","red","green","green","red","red","red","green","green","red","green","green","green","green","green","green","red","green","red","green","green","green","green","green","green","red","red","green","green","green","green","green","red","green","green","green","green","green","green","green","green","green","red","green","green","green","green","red","green","red","green","red","green","red","green","green","red","red","green","red","green","green","red","green","green","red","green","green","green","green","green","green","green","green","red","red","green","green","green","green","red","green","green","green","green","red","green","green","green","green","red","green","green","green"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false},"groups":{"TF":{"shape":"triangle","color":{"background":"purple","border":"black"}},"useDefaultGroups":true,"gene":{"shape":"dot","color":{"background":"teal","border":"black"}}}},"groups":["TF","gene"],"width":"100%","height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","legend":{"width":0.2,"useGroups":true,"position":"right","ncol":1,"stepX":100,"stepY":100,"zoom":true,"main":{"text":"Legend","style":"font-family:Georgia, Times New Roman, Times, serif;font-weight:bold;font-size:14px;text-align:center;"}}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Green is UP regulated in cancer, red is DOWN regulated in cancer</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This computes the difference in "degree" for each gene</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="do">##i,e Suppose normal tissue has 5 arrows to a gene 'A'</span></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="do">## and the same gene 'A' has 20 arrows to it in cancer tissue</span></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Then the difference would be -15</span></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ lubridate 1.9.3     ✔ tibble    3.2.1
✔ purrr     1.0.2     ✔ tidyr     1.3.1
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ lubridate::%within%() masks IRanges::%within%()
✖ ggplot2::Position()   masks BiocGenerics::Position(), base::Position()
✖ dplyr::collapse()     masks IRanges::collapse()
✖ dplyr::combine()      masks Biobase::combine(), BiocGenerics::combine()
✖ dplyr::count()        masks matrixStats::count()
✖ dplyr::desc()         masks IRanges::desc()
✖ tidyr::expand()       masks S4Vectors::expand()
✖ dplyr::filter()       masks stats::filter()
✖ dplyr::first()        masks S4Vectors::first()
✖ dplyr::lag()          masks stats::lag()
✖ purrr::reduce()       masks GenomicRanges::reduce(), IRanges::reduce()
✖ dplyr::rename()       masks S4Vectors::rename()
✖ lubridate::second()   masks S4Vectors::second()
✖ lubridate::second&lt;-() masks S4Vectors::second&lt;-()
✖ dplyr::select()       masks biomaRt::select()
✖ dplyr::slice()        masks IRanges::slice()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>in_degree_gtex<span class="ot">&lt;-</span> gtex_network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(gene) <span class="sc">|&gt;</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">in_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>in_degree_tcga<span class="ot">&lt;-</span> tcga_network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(gene) <span class="sc">|&gt;</span></span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">in_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>in_degree_tcga <span class="ot">&lt;-</span> in_degree_tcga[in_degree_tcga<span class="sc">$</span>gene <span class="sc">%in%</span> in_degree_gtex<span class="sc">$</span>gene,]</span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_degree_gtex</span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a>in_difference<span class="sc">$</span>in_degree <span class="ot">&lt;-</span> in_degree_gtex<span class="sc">$</span>in_degree <span class="sc">-</span> in_degree_tcga<span class="sc">$</span>in_degree</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Using the differences, we compute differentially expressed pathways using kegg</span></span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>pathways <span class="ot">&lt;-</span> <span class="fu">gmtPathways</span>(<span class="st">"c2.cp.kegg_legacy.v2024.1.Hs.entrez.gmt"</span>)</span>
<span id="cb109-3"><a href="#cb109-3" aria-hidden="true" tabindex="-1"></a>in_difference<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(in_difference<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,in_difference<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb109-4"><a href="#cb109-4" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>entrezgene_id[<span class="fu">match</span>(in_difference<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb109-5"><a href="#cb109-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb109-6"><a href="#cb109-6" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_difference[<span class="sc">!</span><span class="fu">is.na</span>(in_difference<span class="sc">$</span>gene),]</span>
<span id="cb109-7"><a href="#cb109-7" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_difference[<span class="sc">!</span><span class="fu">duplicated</span>(in_difference<span class="sc">$</span>gene), ]</span>
<span id="cb109-8"><a href="#cb109-8" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> <span class="fu">setNames</span>(in_difference<span class="sc">$</span>in_degree, in_difference<span class="sc">$</span>gene)</span>
<span id="cb109-9"><a href="#cb109-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb109-10"><a href="#cb109-10" aria-hidden="true" tabindex="-1"></a>fgseaRes <span class="ot">&lt;-</span> <span class="fu">fgsea</span>(pathways, in_difference, <span class="at">minSize=</span><span class="dv">15</span>, <span class="at">maxSize=</span><span class="dv">500</span>,<span class="at">nPermSimple =</span> <span class="dv">10000</span>)</span>
<span id="cb109-11"><a href="#cb109-11" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span> fgseaRes[fgseaRes<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb109-12"><a href="#cb109-12" aria-hidden="true" tabindex="-1"></a>sig[<span class="fu">order</span>(sig<span class="sc">$</span>NES)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],] <span class="co">#These pathways are DOWNregulated in cancer</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                pathway         pval         padj   log2err
                                 &lt;char&gt;        &lt;num&gt;        &lt;num&gt;     &lt;num&gt;
 1:           KEGG_RENAL_CELL_CARCINOMA 5.252594e-06 1.109611e-04 0.6105269
 2:     KEGG_JAK_STAT_SIGNALING_PATHWAY 9.207589e-06 1.414621e-04 0.5933255
 3: KEGG_UBIQUITIN_MEDIATED_PROTEOLYSIS 2.176508e-07 9.195745e-06 0.6901325
 4:        KEGG_CITRATE_CYCLE_TCA_CYCLE 9.182501e-04 7.389727e-03 0.4772708
 5:     KEGG_TGF_BETA_SIGNALING_PATHWAY 3.194079e-05 4.498328e-04 0.5573322
 6: KEGG_NEUROTROPHIN_SIGNALING_PATHWAY 8.531788e-06 1.414621e-04 0.5933255
 7:    KEGG_AMINOACYL_TRNA_BIOSYNTHESIS 6.716154e-04 5.675150e-03 0.4772708
 8:    KEGG_BASAL_TRANSCRIPTION_FACTORS 3.920254e-03 2.208410e-02 0.4070179
 9:       KEGG_CHRONIC_MYELOID_LEUKEMIA 2.833900e-04 2.817230e-03 0.4984931
10:              KEGG_ADHERENS_JUNCTION 5.369882e-04 5.041722e-03 0.4772708
            ES       NES  size  leadingEdge
         &lt;num&gt;     &lt;num&gt; &lt;int&gt;       &lt;list&gt;
 1: -0.4012388 -2.225795    67 2885, 56....
 2: -0.3525271 -2.139517    99 6774, 28....
 3: -0.3301345 -2.095766   130 10075, 8....
 4: -0.4698328 -2.058029    28 5106, 14....
 5: -0.3623853 -2.049323    73 353500, ....
 6: -0.3190618 -1.960748   108 2885, 56....
 7: -0.3948802 -1.926902    41 4141, 85....
 8: -0.4244961 -1.910681    31 2967, 27....
 9: -0.3409736 -1.904928    69 2885, 56....
10: -0.3379586 -1.893207    70 3643, 10....</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>sig[<span class="fu">order</span>(<span class="sc">-</span>sig<span class="sc">$</span>NES)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>],] <span class="co">#These pathways are UPregulated in cancer</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                              pathway         pval         padj
                                               &lt;char&gt;        &lt;num&gt;        &lt;num&gt;
 1:                                     KEGG_RIBOSOME 4.893139e-27 8.269405e-25
 2:                    KEGG_OXIDATIVE_PHOSPHORYLATION 7.799850e-13 6.590874e-11
 3:                           KEGG_PARKINSONS_DISEASE 1.263304e-11 7.116614e-10
 4:                   KEGG_CARDIAC_MUSCLE_CONTRACTION 2.836409e-07 9.587064e-06
 5:                           KEGG_ALZHEIMERS_DISEASE 1.279961e-06 3.605225e-05
 6:                     KEGG_LINOLEIC_ACID_METABOLISM 1.253139e-03 9.207849e-03
 7:         KEGG_MATURITY_ONSET_DIABETES_OF_THE_YOUNG 1.247533e-03 9.207849e-03
 8:                          KEGG_HUNTINGTONS_DISEASE 3.404144e-06 8.218575e-05
 9:                  KEGG_ARACHIDONIC_ACID_METABOLISM 3.916449e-03 2.208410e-02
10: KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450 5.370695e-03 2.750447e-02
11:              KEGG_DRUG_METABOLISM_CYTOCHROME_P450 1.081647e-02 4.458496e-02
12:                                  KEGG_ENDOCYTOSIS 4.400856e-03 2.399177e-02
13:    KEGG_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY 8.677028e-03 3.956630e-02
14:                       KEGG_MAPK_SIGNALING_PATHWAY 2.551636e-04 2.817230e-03
15:                    KEGG_INSULIN_SIGNALING_PATHWAY 3.060606e-03 2.068970e-02
16:                               KEGG_FOCAL_ADHESION 5.760630e-04 5.123928e-03
17:                       KEGG_ERBB_SIGNALING_PATHWAY 8.896565e-03 3.956630e-02
18:                              KEGG_PROSTATE_CANCER 7.602773e-03 3.569080e-02
19:                                    KEGG_APOPTOSIS 3.378859e-03 2.196259e-02
20:             KEGG_REGULATION_OF_ACTIN_CYTOSKELETON 2.761757e-04 2.817230e-03
                                              pathway         pval         padj
      log2err         ES       NES  size  leadingEdge
        &lt;num&gt;      &lt;num&gt;     &lt;num&gt; &lt;int&gt;       &lt;list&gt;
 1: 1.3499257  0.7365364  3.038790    86 6173, 61....
 2: 0.9214260  0.5532066  2.374521   121 1327, 47....
 3: 0.8753251  0.5422312  2.321659   117 1327, 47....
 4: 0.6749629  0.5789312  2.234804    54 1327, 45....
 5: 0.6435518  0.4351414  1.897832   145 572, 132....
 6: 0.4550599  0.6262491  1.892477    17 1559, 10....
 7: 0.4550599  0.5844030  1.880771    22 3110, 32....
 8: 0.6272567  0.4177886  1.835964   160 1327, 27....
 9: 0.2526439  0.4705981  1.735110    42 2879, 40....
10: 0.2133879  0.4574812  1.718544    46 1559, 29....
11: 0.1493082  0.4315524  1.631807    48 1559, 29....
12: 0.4070179 -0.2160730 -1.431039   166 10193, 3....
13: 0.3807304 -0.2541280 -1.495360    88 2215, 28....
14: 0.4984931 -0.2208282 -1.506423   209 1845, 28....
15: 0.4317077 -0.2428449 -1.532169   121 5507, 36....
16: 0.4772708 -0.2324019 -1.558652   175 2885, 70....
17: 0.3807304 -0.2742419 -1.568515    76 2885, 56....
18: 0.4070179 -0.2721160 -1.588962    83 2885, 71....
19: 0.4317077 -0.2844215 -1.626172    77 1676, 33....
20: 0.4984931 -0.2428203 -1.629721   179 10163, 5....
      log2err         ES       NES  size  leadingEdge</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="do">## This plots a bubble plot for the differentially expressed pathways</span></span>
<span id="cb113-2"><a href="#cb113-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-3"><a href="#cb113-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb113-4"><a href="#cb113-4" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(fgseaRes)</span>
<span id="cb113-5"><a href="#cb113-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Settings</span></span>
<span id="cb113-6"><a href="#cb113-6" aria-hidden="true" tabindex="-1"></a>fdrcut <span class="ot">&lt;-</span> <span class="fl">0.05</span> <span class="co"># FDR cut-off to use as output for significant signatures</span></span>
<span id="cb113-7"><a href="#cb113-7" aria-hidden="true" tabindex="-1"></a>dencol_neg <span class="ot">&lt;-</span> <span class="st">"blue"</span> <span class="co"># bubble plot color for negative ES</span></span>
<span id="cb113-8"><a href="#cb113-8" aria-hidden="true" tabindex="-1"></a>dencol_pos <span class="ot">&lt;-</span> <span class="st">"red"</span> <span class="co"># bubble plot color for positive ES</span></span>
<span id="cb113-9"><a href="#cb113-9" aria-hidden="true" tabindex="-1"></a>signnamelength <span class="ot">&lt;-</span> <span class="dv">4</span> <span class="co"># set to remove prefix from signature names (2 for "GO", 4 for "KEGG", 8 for "REACTOME")</span></span>
<span id="cb113-10"><a href="#cb113-10" aria-hidden="true" tabindex="-1"></a>asp <span class="ot">&lt;-</span> <span class="dv">3</span> <span class="co"># aspect ratio of bubble plot</span></span>
<span id="cb113-11"><a href="#cb113-11" aria-hidden="true" tabindex="-1"></a>charcut <span class="ot">&lt;-</span> <span class="dv">100</span> <span class="co"># cut signature name in heatmap to this nr of characters</span></span>
<span id="cb113-12"><a href="#cb113-12" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">as.character</span>(dat<span class="sc">$</span>pathway) <span class="co"># 'a' is a great variable name to substitute row names with something more readable</span></span>
<span id="cb113-13"><a href="#cb113-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(a)){</span>
<span id="cb113-14"><a href="#cb113-14" aria-hidden="true" tabindex="-1"></a>  a[j] <span class="ot">&lt;-</span> <span class="fu">substr</span>(a[j], signnamelength<span class="sc">+</span><span class="dv">2</span>, <span class="fu">nchar</span>(a[j]))</span>
<span id="cb113-15"><a href="#cb113-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb113-16"><a href="#cb113-16" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">tolower</span>(a) <span class="co"># convert to lower case (you may want to comment this out, it really depends on what signatures you are looking at, c6 signatures contain gene names, and converting those to lower case may be confusing)</span></span>
<span id="cb113-17"><a href="#cb113-17" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(a)){</span>
<span id="cb113-18"><a href="#cb113-18" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nchar</span>(a[j])<span class="sc">&gt;</span>charcut) { a[j] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">substr</span>(a[j], <span class="dv">1</span>, charcut), <span class="st">"..."</span>, <span class="at">sep=</span><span class="st">" "</span>)}</span>
<span id="cb113-19"><a href="#cb113-19" aria-hidden="true" tabindex="-1"></a>} <span class="co"># cut signature names that have more characters than charcut, and add "..."</span></span>
<span id="cb113-20"><a href="#cb113-20" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, a)</span>
<span id="cb113-21"><a href="#cb113-21" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>NAME <span class="ot">&lt;-</span> a</span>
<span id="cb113-22"><a href="#cb113-22" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat[dat[,<span class="st">"padj"</span>]<span class="sc">&lt;</span>fdrcut,]</span>
<span id="cb113-23"><a href="#cb113-23" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat2[<span class="fu">order</span>(dat2[,<span class="st">"padj"</span>]),] </span>
<span id="cb113-24"><a href="#cb113-24" aria-hidden="true" tabindex="-1"></a>dat2<span class="sc">$</span>signature <span class="ot">&lt;-</span> <span class="fu">factor</span>(dat2<span class="sc">$</span>NAME, <span class="fu">rev</span>(<span class="fu">as.character</span>(dat2<span class="sc">$</span>NAME)))</span>
<span id="cb113-25"><a href="#cb113-25" aria-hidden="true" tabindex="-1"></a>sign_neg <span class="ot">&lt;-</span> <span class="fu">which</span>(dat2[,<span class="st">"NES"</span>]<span class="sc">&lt;</span><span class="dv">0</span>) <span class="co">#Up regulated in cancer</span></span>
<span id="cb113-26"><a href="#cb113-26" aria-hidden="true" tabindex="-1"></a>sign_pos <span class="ot">&lt;-</span> <span class="fu">which</span>(dat2[,<span class="st">"NES"</span>]<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb113-27"><a href="#cb113-27" aria-hidden="true" tabindex="-1"></a>signcol <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(dat2<span class="sc">$</span>signature))</span>
<span id="cb113-28"><a href="#cb113-28" aria-hidden="true" tabindex="-1"></a>signcol[sign_neg] <span class="ot">&lt;-</span> dencol_neg <span class="co"># text color of negative signatures</span></span>
<span id="cb113-29"><a href="#cb113-29" aria-hidden="true" tabindex="-1"></a>signcol[sign_pos] <span class="ot">&lt;-</span> dencol_pos <span class="co"># text color of positive signatures</span></span>
<span id="cb113-30"><a href="#cb113-30" aria-hidden="true" tabindex="-1"></a>signcol <span class="ot">&lt;-</span> <span class="fu">rev</span>(signcol) <span class="co"># need to revert vector of colors, because ggplot starts plotting these from below</span></span>
<span id="cb113-31"><a href="#cb113-31" aria-hidden="true" tabindex="-1"></a>g<span class="ot">&lt;-</span><span class="fu">ggplot</span>(dat2, <span class="fu">aes</span>(<span class="at">x=</span>padj,<span class="at">y=</span>signature,<span class="at">size=</span>size))</span>
<span id="cb113-32"><a href="#cb113-32" aria-hidden="true" tabindex="-1"></a>g<span class="sc">+</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill=</span>NES), <span class="at">shape=</span><span class="dv">21</span>, <span class="at">colour=</span><span class="st">"white"</span>)<span class="sc">+</span></span>
<span id="cb113-33"><a href="#cb113-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span> <span class="co"># white background, needs to be placed before the "signcol" line</span></span>
<span id="cb113-34"><a href="#cb113-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>,fdrcut)<span class="sc">+</span></span>
<span id="cb113-35"><a href="#cb113-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_area</span>(<span class="at">max_size=</span><span class="dv">10</span>,<span class="at">guide=</span><span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb113-36"><a href="#cb113-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low=</span>dencol_neg, <span class="at">high=</span>dencol_pos)<span class="sc">+</span></span>
<span id="cb113-37"><a href="#cb113-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">colour=</span>signcol))<span class="sc">+</span></span>
<span id="cb113-38"><a href="#cb113-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">aspect.ratio=</span>asp, <span class="at">axis.title.y=</span><span class="fu">element_blank</span>()) <span class="co"># test aspect.ratio</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Vectorized input to `element_text()` is not officially supported.
ℹ Results may be unexpected or may change in future versions of ggplot2.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Blue is UPregulated in cancer, red is DOWNregulated in cancer</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="analysis-by-biological-sex" class="level2">
<h2 class="anchored" data-anchor-id="analysis-by-biological-sex">Analysis by Biological Sex</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>tcga_logtpm <span class="ot">&lt;-</span> <span class="fu">assay</span>(tcga, <span class="st">"logtpm"</span>)</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>tcga_exprs<span class="ot">&lt;-</span> <span class="fu">assay</span>(tcga, <span class="st">"raw_counts"</span>)</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a>tcga_gene_lengths <span class="ot">&lt;-</span> <span class="fu">rowData</span>(tcga)<span class="sc">$</span>bp_length<span class="sc">/</span><span class="dv">1000</span></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>tcga_rpk <span class="ot">&lt;-</span> tcga_exprs<span class="sc">/</span>tcga_gene_lengths</span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>tcga_scaling_factors <span class="ot">&lt;-</span> <span class="fu">colSums</span>(tcga_rpk)</span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>tcga_tpm <span class="ot">&lt;-</span> <span class="fu">sweep</span>(tcga_rpk, <span class="dv">2</span>, tcga_scaling_factors, <span class="at">FUN =</span> <span class="st">"/"</span>) <span class="sc">*</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span> </span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(tcga_tpm <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">&gt;=</span> <span class="fu">ncol</span>(tcga_tpm)<span class="sc">/</span><span class="dv">4</span></span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tcga)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 34032   168</code></pre>
</div>
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a>tcga_filtered <span class="ot">&lt;-</span> tcga[keep, ]</span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(tcga_filtered)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 22942   168</code></pre>
</div>
<div class="sourceCode cell-code" id="cb120"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb120-1"><a href="#cb120-1" aria-hidden="true" tabindex="-1"></a>tcga_exprs_filtered <span class="ot">&lt;-</span> <span class="fu">assay</span>(tcga_filtered, <span class="st">"raw_counts"</span>)</span>
<span id="cb120-2"><a href="#cb120-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb120-3"><a href="#cb120-3" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA - do we work here with results after filtering genes or not?</span></span>
<span id="cb120-4"><a href="#cb120-4" aria-hidden="true" tabindex="-1"></a>PCA <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(tcga_logtpm))</span>
<span id="cb120-5"><a href="#cb120-5" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">autoplot</span>(PCA, <span class="at">data =</span> <span class="fu">as.data.frame</span>(<span class="fu">colData</span>(tcga)), </span>
<span id="cb120-6"><a href="#cb120-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">colour=</span><span class="st">"tcga.gdc_cases.demographic.gender"</span>)</span>
<span id="cb120-7"><a href="#cb120-7" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="co"># voom</span></span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>condition_sex <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">colData</span>(tcga_filtered)<span class="sc">$</span>tcga.gdc_cases.demographic.gender, </span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"male"</span>, <span class="st">"female"</span>))</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>design_sex <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> condition_sex)</span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>voomOutput_sex <span class="ot">&lt;-</span> <span class="fu">voom</span>(tcga_exprs_filtered, design_sex, <span class="at">plot=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a>fit_sex <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(voomOutput_sex, design_sex)</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>fit_sex <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit_sex)</span>
<span id="cb122-3"><a href="#cb122-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSA</span>(fit_sex, <span class="at">main=</span><span class="st">"Final model mean-variance trend"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-31-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb123"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb123-1"><a href="#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="co"># DEGs</span></span>
<span id="cb123-2"><a href="#cb123-2" aria-hidden="true" tabindex="-1"></a>res_sex <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit_sex, <span class="at">coef=</span><span class="st">"condition_sexfemale"</span>, <span class="at">number=</span><span class="cn">Inf</span>)</span>
<span id="cb123-3"><a href="#cb123-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_sex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       logFC   AveExpr         t      P.Value    adj.P.Val
ENSG00000270641.1   11.14252  1.588921  32.06450 8.262851e-74 1.895663e-69
ENSG00000229807.10  11.93958  1.486368  29.91414 1.811801e-69 1.068689e-65
ENSG00000274655.1   12.80854 -5.823626  29.90826 1.863288e-69 1.068689e-65
ENSG00000154620.5  -12.36530 -5.780468 -30.58181 7.718567e-71 8.853968e-67
ENSG00000129824.15 -12.17269  1.337784 -25.29026 2.243555e-59 5.719070e-56
ENSG00000277577.1   12.76855 -6.103659  27.26290 8.180570e-64 2.681123e-60
                          B
ENSG00000270641.1  153.6502
ENSG00000229807.10 143.7728
ENSG00000274655.1  127.6093
ENSG00000154620.5  125.1729
ENSG00000129824.15 119.0083
ENSG00000277577.1  117.9513</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of genes differentially expressed</span></span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_sex<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(res_sex<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 22</code></pre>
</div>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="co"># overexpressed in females</span></span>
<span id="cb127-2"><a href="#cb127-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_sex<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> res_sex<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 6</code></pre>
</div>
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="co"># overexpressed in males</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_sex<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> res_sex<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 16</code></pre>
</div>
</div>
<section id="gene-set-enrichment-analysis-1" class="level3">
<h3 class="anchored" data-anchor-id="gene-set-enrichment-analysis-1">Gene set Enrichment Analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="at">biomart =</span> <span class="st">"genes"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>)</span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a>ensembl_ids <span class="ot">&lt;-</span> <span class="fu">row.names</span>(res_sex[res_sex<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(res_sex<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">1</span>,])</span>
<span id="cb131-3"><a href="#cb131-3" aria-hidden="true" tabindex="-1"></a>ensembl_ids <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, ensembl_ids)</span>
<span id="cb131-4"><a href="#cb131-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-5"><a href="#cb131-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Query BioMart</span></span>
<span id="cb131-6"><a href="#cb131-6" aria-hidden="true" tabindex="-1"></a>gene_info <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb131-7"><a href="#cb131-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">"ensembl_gene_id"</span>, <span class="st">"chromosome_name"</span>),</span>
<span id="cb131-8"><a href="#cb131-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">filters =</span> <span class="st">"ensembl_gene_id"</span>,</span>
<span id="cb131-9"><a href="#cb131-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> ensembl_ids,</span>
<span id="cb131-10"><a href="#cb131-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">mart =</span> ensembl</span>
<span id="cb131-11"><a href="#cb131-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb131-12"><a href="#cb131-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb131-13"><a href="#cb131-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Just prints which chromosome each gene is on</span></span>
<span id="cb131-14"><a href="#cb131-14" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(gene_info)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   ensembl_gene_id chromosome_name
1  ENSG00000012817               Y
2  ENSG00000067048               Y
3  ENSG00000067646               Y
4  ENSG00000099725               Y
5  ENSG00000114374               Y
6  ENSG00000129824               Y
7  ENSG00000131002               Y
8  ENSG00000154620               Y
9  ENSG00000165246               Y
10 ENSG00000176728               Y
11 ENSG00000183878               Y
12 ENSG00000198692               Y
13 ENSG00000205611              20
14 ENSG00000229807               X
15 ENSG00000233070               Y
16 ENSG00000252155               Y
17 ENSG00000270641               X
18 ENSG00000274655               X
19 ENSG00000277577               X
20 ENSG00000278039               X</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">"ensembl"</span>,<span class="at">dataset=</span><span class="st">"hsapiens_gene_ensembl"</span>,<span class="at">host =</span> <span class="st">"https://www.ensembl.org"</span>)</span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>ens_genes <span class="ot">&lt;-</span> <span class="fu">row.names</span>(tcga_filtered)</span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove the version of the Ensembl ID (the number after the .)</span></span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a>ens_genes <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, ens_genes)</span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a>annot <span class="ot">&lt;-</span> <span class="fu">getBM</span>(<span class="at">attributes=</span><span class="fu">c</span>(<span class="st">'ensembl_gene_id'</span>, <span class="st">'entrezgene_id'</span>, <span class="st">'hgnc_symbol'</span>), </span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">filters =</span> <span class="st">'ensembl_gene_id'</span>, </span>
<span id="cb133-7"><a href="#cb133-7" aria-hidden="true" tabindex="-1"></a>      <span class="at">values =</span> ens_genes, </span>
<span id="cb133-8"><a href="#cb133-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">mart =</span> ensembl)</span>
<span id="cb133-9"><a href="#cb133-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(annot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  ensembl_gene_id entrezgene_id hgnc_symbol
1 ENSG00000000457         57147       SCYL3
2 ENSG00000000460         55732       FIRRM
3 ENSG00000000938          2268         FGR
4 ENSG00000000971          3075         CFH
5 ENSG00000001460         90529       STPG1
6 ENSG00000001461         57185      NIPAL3</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>up_females <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_sex)[<span class="fu">which</span>(res_sex<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">2</span> <span class="sc">&amp;</span> res_sex<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)]</span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a>up_females <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, up_females) <span class="co"># This will remove the version in the ensemble ID</span></span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a>up_females_ID <span class="ot">&lt;-</span> annot[<span class="fu">match</span>(up_females, annot<span class="sc">$</span>ensembl_gene_id), <span class="st">"entrezgene_id"</span>]</span>
<span id="cb135-4"><a href="#cb135-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-5"><a href="#cb135-5" aria-hidden="true" tabindex="-1"></a>up_males <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res)[<span class="fu">which</span>(res_sex<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">2</span> <span class="sc">&amp;</span> res_sex<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)]</span>
<span id="cb135-6"><a href="#cb135-6" aria-hidden="true" tabindex="-1"></a>up_males <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, up_males)</span>
<span id="cb135-7"><a href="#cb135-7" aria-hidden="true" tabindex="-1"></a>up_males_ID <span class="ot">&lt;-</span> annot[<span class="fu">match</span>(up_males, annot<span class="sc">$</span>ensembl_gene_id), <span class="st">"entrezgene_id"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform over-representation analyses for Gene Ontology terms using the limma package</span></span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>go_females <span class="ot">&lt;-</span> <span class="fu">goana</span>(up_females_ID, <span class="at">species=</span><span class="st">"Hs"</span>)</span>
<span id="cb136-3"><a href="#cb136-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the list of top GO terms enriched for genes highly expressed in females</span></span>
<span id="cb136-4"><a href="#cb136-4" aria-hidden="true" tabindex="-1"></a><span class="fu">topGO</span>(go_females, <span class="at">ontology =</span> <span class="st">"BP"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                    Term Ont
GO:2000627                positive regulation of miRNA catabolic process  BP
GO:0000512           lncRNA-mediated post-transcriptional gene silencing  BP
GO:2000625                         regulation of miRNA catabolic process  BP
GO:0060816                           random inactivation of X chromosome  BP
GO:0010587                                       miRNA catabolic process  BP
GO:0009048           dosage compensation by inactivation of X chromosome  BP
GO:0007549                            sex-chromosome dosage compensation  BP
GO:0031048           regulatory ncRNA-mediated heterochromatin formation  BP
GO:0140719                        constitutive heterochromatin formation  BP
GO:2000630                positive regulation of miRNA metabolic process  BP
GO:2000628                         regulation of miRNA metabolic process  BP
GO:0010586                                       miRNA metabolic process  BP
GO:0031507                                     heterochromatin formation  BP
GO:0045814            negative regulation of gene expression, epigenetic  BP
GO:0040029                      epigenetic regulation of gene expression  BP
GO:0006401                                         RNA catabolic process  BP
GO:0141188                                nucleic acid catabolic process  BP
GO:0034655              nucleobase-containing compound catabolic process  BP
GO:0009896                      positive regulation of catabolic process  BP
GO:0035194 regulatory ncRNA-mediated post-transcriptional gene silencing  BP
             N DE         P.DE
GO:2000627   9  1 0.0004321729
GO:0000512  11  1 0.0005282113
GO:2000625  11  1 0.0005282113
GO:0060816  14  1 0.0006722689
GO:0010587  20  1 0.0009603842
GO:0009048  21  1 0.0010084034
GO:0007549  23  1 0.0011044418
GO:0031048  26  1 0.0012484994
GO:0140719  62  1 0.0029771909
GO:2000630  67  1 0.0032172869
GO:2000628  92  1 0.0044177671
GO:0010586 110  1 0.0052821128
GO:0031507 161  1 0.0077310924
GO:0045814 192  1 0.0092196879
GO:0040029 295  1 0.0141656663
GO:0006401 412  1 0.0197839136
GO:0141188 441  1 0.0211764706
GO:0034655 632  1 0.0303481393
GO:0009896 648  1 0.0311164466
GO:0035194 648  1 0.0311164466</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform over-representation analyses for Gene Ontology terms using the limma package</span></span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>go_males <span class="ot">&lt;-</span> <span class="fu">goana</span>(up_males_ID, <span class="at">species=</span><span class="st">"Hs"</span>)</span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is the list of top GO terms enriched for genes highly expressed in males</span></span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a><span class="fu">topGO</span>(go_males, <span class="at">ontology =</span> <span class="st">"BP"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                                              Term
GO:0071266                             'de novo' L-methionine biosynthetic process
GO:0070060                                     'de novo' actin filament nucleation
GO:1901737                                 (R)-mevalonic acid biosynthetic process
GO:1901735                                    (R)-mevalonic acid metabolic process
GO:1903100       1-phosphatidyl-1D-myo-inositol 3,5-bisphosphate metabolic process
GO:0046360                                      2-oxobutyrate biosynthetic process
GO:0019606                                         2-oxobutyrate catabolic process
GO:0006666                                    3-keto-sphinganine metabolic process
GO:0018960                                         4-nitrophenol metabolic process
GO:0140484                     5-aminolevulinic acid import across plasma membrane
GO:0034463                                                90S preribosome assembly
GO:0086072 AV node cell-bundle of His cell adhesion involved in cell communication
GO:0002368                                              B cell cytokine production
GO:0032597                            B cell receptor transport into membrane raft
GO:0032595                          B cell receptor transport within lipid bilayer
GO:0002336                                           B-1 B cell lineage commitment
GO:0038118                           C-C chemokine receptor CCR7 signaling pathway
GO:0140633                                     CARD8 inflammasome complex assembly
GO:0140708                                                             CAT tailing
GO:0035724                                               CD24 biosynthetic process
           Ont N DE P.DE
GO:0071266  BP 1  0    1
GO:0070060  BP 1  0    1
GO:1901737  BP 1  0    1
GO:1901735  BP 1  0    1
GO:1903100  BP 1  0    1
GO:0046360  BP 1  0    1
GO:0019606  BP 1  0    1
GO:0006666  BP 1  0    1
GO:0018960  BP 1  0    1
GO:0140484  BP 1  0    1
GO:0034463  BP 1  0    1
GO:0086072  BP 1  0    1
GO:0002368  BP 1  0    1
GO:0032597  BP 1  0    1
GO:0032595  BP 1  0    1
GO:0002336  BP 1  0    1
GO:0038118  BP 1  0    1
GO:0140633  BP 1  0    1
GO:0140708  BP 1  0    1
GO:0035724  BP 1  0    1</code></pre>
</div>
</div>
</section>
<section id="gene-network-analysis" class="level3">
<h3 class="anchored" data-anchor-id="gene-network-analysis">Gene Network Analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Filtering based on PANDA requirements</span></span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a>males <span class="ot">&lt;-</span> tcga_filtered[,<span class="fu">colData</span>(tcga_filtered)<span class="sc">$</span>tcga.gdc_cases.demographic.gender<span class="sc">==</span><span class="st">'male'</span>]</span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a>females <span class="ot">&lt;-</span> tcga_filtered[,<span class="fu">colData</span>(tcga_filtered)<span class="sc">$</span>tcga.gdc_cases.demographic.gender<span class="sc">==</span><span class="st">'female'</span>]</span>
<span id="cb140-4"><a href="#cb140-4" aria-hidden="true" tabindex="-1"></a>zero_na_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assay</span>(males, <span class="st">"logtpm"</span>), <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="fu">length</span>(x[(<span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x<span class="sc">!=</span><span class="dv">0</span>) ]))</span>
<span id="cb140-5"><a href="#cb140-5" aria-hidden="true" tabindex="-1"></a>expr_males <span class="ot">&lt;-</span> <span class="fu">assay</span>(males, <span class="st">"logtpm"</span>)[zero_na_counts <span class="sc">&gt;</span> <span class="dv">20</span>,]</span>
<span id="cb140-6"><a href="#cb140-6" aria-hidden="true" tabindex="-1"></a>zero_na_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assay</span>(females, <span class="st">"logtpm"</span>), <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="fu">length</span>(x[(<span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x<span class="sc">!=</span><span class="dv">0</span>) ]))</span>
<span id="cb140-7"><a href="#cb140-7" aria-hidden="true" tabindex="-1"></a>expr_females <span class="ot">&lt;-</span> <span class="fu">assay</span>(females, <span class="st">"logtpm"</span>)[zero_na_counts <span class="sc">&gt;</span> <span class="dv">20</span>,]</span>
<span id="cb140-8"><a href="#cb140-8" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(expr_males) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">row.names</span>(expr_males))</span>
<span id="cb140-9"><a href="#cb140-9" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(expr_females) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">row.names</span>(expr_females))</span>
<span id="cb140-10"><a href="#cb140-10" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(expr_males, file = "males.txt", sep = "\t", quote = FALSE, row.names = TRUE)</span></span>
<span id="cb140-11"><a href="#cb140-11" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(expr_females, file = "females.txt", sep = "\t", quote = FALSE, row.names = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Load PANDA results</span></span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a>male_network_full <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"output_males.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb141-3"><a href="#cb141-3" aria-hidden="true" tabindex="-1"></a>male_network <span class="ot">&lt;-</span> male_network_full[<span class="fu">order</span>(male_network_full<span class="sc">$</span>force, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb141-4"><a href="#cb141-4" aria-hidden="true" tabindex="-1"></a>male_network <span class="ot">&lt;-</span> male_network[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>,]</span>
<span id="cb141-5"><a href="#cb141-5" aria-hidden="true" tabindex="-1"></a>male_network <span class="ot">&lt;-</span> <span class="fu">subset</span>(male_network, <span class="at">select=</span><span class="sc">-</span>motif)</span>
<span id="cb141-6"><a href="#cb141-6" aria-hidden="true" tabindex="-1"></a>male_network<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(male_network<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,male_network<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb141-7"><a href="#cb141-7" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>hgnc_symbol[<span class="fu">match</span>(male_network<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb141-8"><a href="#cb141-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb141-9"><a href="#cb141-9" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(male_network,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              tf            gene    force
20665626  ZNF341 ENSG00000268836 20.56114
21236918 ZNF385D ENSG00000272829 17.89920
21843094 ZNF385D ENSG00000277368 17.41910
20589865 ZNF385D ENSG00000268129 15.82165
16564976 ZNF385D      ZNF687-AS1 15.53342
14243960 ZNF385D         TMEM240 15.09874
9779394  ZNF385D          SPACA9 15.08535
5260990  ZNF385D           HOXD1 15.03299
10454363 ZNF385D           RSPO1 14.84496
18255116   CEBPB     ALDH1L1-AS1 14.84339</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb143"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb143-1"><a href="#cb143-1" aria-hidden="true" tabindex="-1"></a>female_network_full <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"output_females.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb143-2"><a href="#cb143-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-3"><a href="#cb143-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Some genes were filtered differently in the two groups, so this makes the set of genes same now</span></span>
<span id="cb143-4"><a href="#cb143-4" aria-hidden="true" tabindex="-1"></a>female_network_full <span class="ot">&lt;-</span> female_network_full[female_network_full<span class="sc">$</span>gene <span class="sc">%in%</span> male_network_full<span class="sc">$</span>gene,]</span>
<span id="cb143-5"><a href="#cb143-5" aria-hidden="true" tabindex="-1"></a>female_network_full <span class="ot">&lt;-</span> female_network_full[female_network_full<span class="sc">$</span>tf <span class="sc">%in%</span> male_network_full<span class="sc">$</span>tf,]</span>
<span id="cb143-6"><a href="#cb143-6" aria-hidden="true" tabindex="-1"></a>male_network_full <span class="ot">&lt;-</span> male_network_full[male_network_full<span class="sc">$</span>gene <span class="sc">%in%</span> female_network_full<span class="sc">$</span>gene,]</span>
<span id="cb143-7"><a href="#cb143-7" aria-hidden="true" tabindex="-1"></a>male_network_full <span class="ot">&lt;-</span> male_network_full[male_network_full<span class="sc">$</span>tf <span class="sc">%in%</span> female_network_full<span class="sc">$</span>tf,]</span>
<span id="cb143-8"><a href="#cb143-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb143-9"><a href="#cb143-9" aria-hidden="true" tabindex="-1"></a>diffnet <span class="ot">&lt;-</span> male_network_full</span>
<span id="cb143-10"><a href="#cb143-10" aria-hidden="true" tabindex="-1"></a>diffnet<span class="sc">$</span>force <span class="ot">&lt;-</span> male_network_full<span class="sc">$</span>force<span class="sc">-</span>female_network_full<span class="sc">$</span>force</span>
<span id="cb143-11"><a href="#cb143-11" aria-hidden="true" tabindex="-1"></a>diffnet <span class="ot">&lt;-</span> <span class="fu">subset</span>(diffnet, <span class="at">select=</span><span class="sc">-</span>motif)</span>
<span id="cb143-12"><a href="#cb143-12" aria-hidden="true" tabindex="-1"></a>diffnet<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(diffnet<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,diffnet<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb143-13"><a href="#cb143-13" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>hgnc_symbol[<span class="fu">match</span>(diffnet<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb143-14"><a href="#cb143-14" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb143-15"><a href="#cb143-15" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(diffnet,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       tf   gene       force
1    ALX1 TSPAN6  0.72289784
2    ALX3 TSPAN6  0.01036401
3    ALX4 TSPAN6  0.61023013
4      AR TSPAN6 -0.28111050
5   ARGFX TSPAN6  0.40545259
6  ARID3A TSPAN6  0.90807872
7  ARID5B TSPAN6  0.75849500
8    ARNT TSPAN6 -0.16604893
9   ARNT2 TSPAN6 -0.16400622
10  ARNTL TSPAN6 -0.18442869</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> diffnet</span>
<span id="cb145-2"><a href="#cb145-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(edges) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"from"</span>, <span class="st">"to"</span>, <span class="st">"value"</span>)</span>
<span id="cb145-3"><a href="#cb145-3" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>arrows <span class="ot">=</span> <span class="st">"to"</span>   </span>
<span id="cb145-4"><a href="#cb145-4" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>color  <span class="ot">=</span> <span class="fu">ifelse</span>(edges<span class="sc">$</span>value <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"green"</span>, <span class="st">"red"</span>)</span>
<span id="cb145-5"><a href="#cb145-5" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>value  <span class="ot">=</span> <span class="fu">abs</span>(edges<span class="sc">$</span>value)</span>
<span id="cb145-6"><a href="#cb145-6" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges[<span class="fu">order</span>(edges<span class="sc">$</span>value, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb145-7"><a href="#cb145-7" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>,]</span>
<span id="cb145-8"><a href="#cb145-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(edges,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           from              to    value arrows color
15864031 ZNF235 ENSG00000227077 7.520183     to   red
15864053 ZNF287 ENSG00000227077 7.434822     to   red
15863766  PRDM6 ENSG00000227077 7.288868     to   red
15864122 ZNF487 ENSG00000227077 7.149582     to   red
4103419  ZNF235            MYCL 7.115679     to   red
5384564  ZNF235         CRACR2A 7.078856     to   red
15864028 ZNF225 ENSG00000227077 7.050082     to   red
15864094 ZNF418 ENSG00000227077 7.021414     to   red
17753346 ZNF235         UBA52P3 6.990990     to   red
4103441  ZNF287            MYCL 6.936712     to   red</code></pre>
</div>
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Green is UP in males, red is UP in females</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb148"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb148-1"><a href="#cb148-1" aria-hidden="true" tabindex="-1"></a>nodes       <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))), </span>
<span id="cb148-2"><a href="#cb148-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">label=</span><span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))))</span>
<span id="cb148-3"><a href="#cb148-3" aria-hidden="true" tabindex="-1"></a>nodes<span class="sc">$</span>group <span class="ot">=</span> <span class="fu">ifelse</span>(nodes<span class="sc">$</span>id <span class="sc">%in%</span> edges<span class="sc">$</span>from, <span class="st">"TF"</span>, <span class="st">"gene"</span>)</span>
<span id="cb148-4"><a href="#cb148-4" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visNetwork</span>(nodes, edges, <span class="at">width =</span> <span class="st">"100%"</span>)</span>
<span id="cb148-5"><a href="#cb148-5" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"TF"</span>, <span class="at">shape =</span> <span class="st">"triangle"</span>,</span>
<span id="cb148-6"><a href="#cb148-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"purple"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb148-7"><a href="#cb148-7" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"gene"</span>, <span class="at">shape =</span> <span class="st">"dot"</span>,       </span>
<span id="cb148-8"><a href="#cb148-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"teal"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb148-9"><a href="#cb148-9" aria-hidden="true" tabindex="-1"></a><span class="fu">visLegend</span>(net, <span class="at">main=</span><span class="st">"Legend"</span>, <span class="at">position=</span><span class="st">"right"</span>, <span class="at">ncol=</span><span class="dv">1</span>) <span class="co">#GREEN is upregulated in cancer</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="htmlwidget-24a3285a9193ac5126c2" style="width:100%;height:480px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-24a3285a9193ac5126c2">{"x":{"nodes":{"id":["ZNF235","ZNF287","PRDM6","ZNF487","ZNF225","ZNF418","ZFP28","ZNF879","ZNF98","ZNF182","ZNF570","ZNF613","ZNF492","IRF1","ZNF567","ENSG00000227077","MYCL","CRACR2A","UBA52P3","RPSAP19","RPSAP29","THEM5","SPINK4","RPL21P45","TSIX","RPL23AP57","MYCL-AS1","ATP5MGP2",null,"ENSG00000272884","NFE2","SLC2A11","SERPINB11","ENSG00000272088","GATA4","ENSG00000228399","RPL7","PLA2G4F","C12orf75","RPL39P15","LINC02481","RPL7P26","ITLN1","ENSG00000228360","FAM162A","RPL31P31","ENSG00000265936","UBA52P8","ENSG00000226439","LINC02320","MB","COX5BP7"],"label":["ZNF235","ZNF287","PRDM6","ZNF487","ZNF225","ZNF418","ZFP28","ZNF879","ZNF98","ZNF182","ZNF570","ZNF613","ZNF492","IRF1","ZNF567","ENSG00000227077","MYCL","CRACR2A","UBA52P3","RPSAP19","RPSAP29","THEM5","SPINK4","RPL21P45","TSIX","RPL23AP57","MYCL-AS1","ATP5MGP2",null,"ENSG00000272884","NFE2","SLC2A11","SERPINB11","ENSG00000272088","GATA4","ENSG00000228399","RPL7","PLA2G4F","C12orf75","RPL39P15","LINC02481","RPL7P26","ITLN1","ENSG00000228360","FAM162A","RPL31P31","ENSG00000265936","UBA52P8","ENSG00000226439","LINC02320","MB","COX5BP7"],"group":["TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene"]},"edges":{"from":["ZNF235","ZNF287","PRDM6","ZNF487","ZNF235","ZNF235","ZNF225","ZNF418","ZNF235","ZNF287","ZNF287","ZNF287","ZNF418","PRDM6","ZFP28","ZNF418","ZNF418","PRDM6","ZNF235","ZNF225","PRDM6","ZNF879","ZNF225","ZNF235","ZNF487","ZNF287","ZNF879","ZNF225","ZNF235","ZNF487","ZNF235","ZNF235","ZNF287","ZNF98","ZNF235","ZNF879","ZNF235","ZNF879","ZNF235","ZNF487","ZNF418","ZFP28","ZNF418","ZFP28","ZNF418","ZNF235","ZNF287","ZNF287","ZNF235","PRDM6","ZNF287","ZNF287","ZNF418","ZNF182","ZNF287","ZNF235","ZNF287","ZFP28","ZNF418","ZNF235","ZNF418","ZNF287","PRDM6","ZNF235","ZNF418","PRDM6","ZNF570","ZNF487","ZNF225","ZNF235","PRDM6","ZNF487","ZNF418","PRDM6","ZNF879","ZNF287","ZNF418","PRDM6","ZNF225","ZNF235","PRDM6","ZNF418","PRDM6","ZNF613","ZNF879","PRDM6","ZNF287","ZNF225","ZNF225","ZNF287","ZNF487","ZNF225","ZNF235","ZNF492","ZNF287","IRF1","ZNF879","ZNF570","ZNF879","ZNF225","ZNF879","ZNF879","ZNF235","ZNF225","ZNF287","ZNF487","ZNF235","ZNF287","ZNF487","ZNF487","ZNF235","ZFP28","PRDM6","ZNF98","ZNF879","ZNF235","ZFP28","ZNF418","ZNF879","ZNF487","ZNF570","ZNF487","ZNF287","ZNF235","PRDM6","PRDM6","ZNF98","ZNF487","ZNF418","ZNF235","PRDM6","ZNF487","ZNF225","ZFP28","ZNF98","ZNF225","ZNF235","ZNF287","ZNF235","ZNF225","ZNF287","ZNF287","ZNF418","ZNF879","ZNF487","ZNF879","ZNF570","ZNF487","ZNF287","ZNF235","ZNF235","ZNF225","ZNF879","PRDM6","ZFP28","ZNF225","ZNF418","ZFP28","ZNF418","ZNF487","ZNF287","ZNF567","ZNF235","ZNF225","ZNF235","ZFP28","ZNF418","ZFP28","ZNF418","PRDM6","ZNF418","ZNF235","ZNF287","ZNF487","ZFP28","ZNF235","ZNF235","ZNF418","ZNF225","ZNF225","ZNF879","PRDM6","PRDM6","ZFP28","ZNF287","ZNF235","PRDM6","ZNF287","ZNF287","ZNF235","IRF1","ZNF287","ZFP28","ZNF235","ZNF225","ZFP28","PRDM6","ZNF287","PRDM6","ZNF287"],"to":["ENSG00000227077","ENSG00000227077","ENSG00000227077","ENSG00000227077","MYCL","CRACR2A","ENSG00000227077","ENSG00000227077","UBA52P3","MYCL","CRACR2A","UBA52P3","MYCL","MYCL","ENSG00000227077","CRACR2A","UBA52P3","CRACR2A","RPSAP19","MYCL","UBA52P3","ENSG00000227077","CRACR2A","RPSAP29","MYCL","RPSAP19","MYCL","UBA52P3","THEM5","UBA52P3","SPINK4","RPL21P45","TSIX","ENSG00000227077","TSIX","CRACR2A","RPL23AP57","UBA52P3","MYCL-AS1","CRACR2A","THEM5","MYCL","SPINK4","CRACR2A","RPSAP19","ATP5MGP2","RPL21P45","RPL23AP57",null,"RPSAP19","SPINK4","RPSAP29","RPSAP29","ENSG00000227077","MYCL-AS1","ENSG00000272884","ATP5MGP2","UBA52P3","RPL21P45","NFE2","RPL23AP57","THEM5","TSIX","SLC2A11","ATP5MGP2","RPL21P45","ENSG00000227077","TSIX","RPSAP19","SERPINB11","SPINK4","RPSAP19","MYCL-AS1","RPSAP29","THEM5","ENSG00000272884",null,"MYCL-AS1","RPSAP29","ENSG00000272088","THEM5","NFE2","ATP5MGP2","ENSG00000227077","SPINK4","RPL23AP57","SLC2A11","THEM5","SPINK4",null,"RPL21P45","MYCL-AS1","GATA4","ENSG00000227077","SERPINB11","ENSG00000227077","RPL23AP57","MYCL","RPL21P45","TSIX","ATP5MGP2","RPSAP19","ENSG00000228399","RPL21P45","ENSG00000272088","RPSAP29","RPL7","ENSG00000228399","THEM5","ATP5MGP2",null,"THEM5","ENSG00000272884","MYCL","RPSAP29","PLA2G4F","SPINK4","ENSG00000272884","NFE2","ENSG00000272884","CRACR2A","SPINK4","RPL7","C12orf75","SLC2A11","SERPINB11","UBA52P3","RPL23AP57","SLC2A11","RPL39P15",null,"MYCL-AS1","ATP5MGP2","RPSAP19","CRACR2A","RPL23AP57","LINC02481","RPL39P15","RPL7P26",null,null,"GATA4","TSIX","MYCL-AS1","SLC2A11",null,"UBA52P3","SERPINB11","NFE2","ITLN1","ENSG00000228360","SERPINB11","ENSG00000272884","ENSG00000272088","RPL21P45","ENSG00000272884","SERPINB11","ATP5MGP2","LINC02481",null,"PLA2G4F","ENSG00000227077","FAM162A","SLC2A11","RPL31P31","RPSAP29","GATA4","MYCL-AS1","ENSG00000265936","ENSG00000228399","C12orf75","UBA52P8","ENSG00000228360","ENSG00000272088","RPL23AP57","ENSG00000226439","LINC02320","ENSG00000272088","NFE2","ENSG00000272088","SLC2A11","NFE2",null,"ENSG00000272884","ITLN1","MB","RPL7","LINC02481","RPL7P26","ENSG00000265936","MYCL","ENSG00000226439","TSIX","COX5BP7","GATA4",null,"GATA4","RPL31P31","RPL39P15","FAM162A"],"value":[7.52018313800579,7.434821809710352,7.288868361757808,7.149582255637593,7.115678803796824,7.078855751274467,7.050081805448369,7.021414238382603,6.99099006178265,6.936712348474872,6.923009289140722,6.8665743354835,6.837965329623523,6.783722129548583,6.774674451880799,6.771852575332164,6.744253675333579,6.708216009621262,6.695012563314709,6.694777015354416,6.674677341412961,6.670187750211879,6.642477940630052,6.596041698576148,6.556491308559092,6.55162673974408,6.551037648040984,6.539969628211804,6.5364350351943,6.533062440684533,6.51624044600624,6.512003735747198,6.506222828635033,6.505487485128561,6.489358332369429,6.484624906667403,6.464615734408301,6.459108201252322,6.458125833680332,6.456741624759076,6.452581143165627,6.445186483977319,6.430363898237229,6.418311807256744,6.364729257313197,6.362358020177489,6.357247997044544,6.33363478082415,6.333160491027715,6.331963307777769,6.320040024056224,6.319556555937915,6.308891648874767,6.305847379286738,6.301110433759999,6.292420298306373,6.288933559712865,6.284009946250606,6.282727679078507,6.263999534020501,6.261406215195596,6.260617550486998,6.259133668816283,6.249669368740071,6.249100016323656,6.243457132895384,6.238678514340309,6.238080389238553,6.215638283366098,6.205306458995256,6.200642958605401,6.176783333620326,6.173056220701257,6.171170852515266,6.167122436673822,6.162886787616408,6.161234890909326,6.154079385493634,6.150852273380428,6.144754453651548,6.14078749507642,6.1375917188724,6.131142561676752,6.120225745656567,6.120071431735289,6.11362083334871,6.107051645203944,6.102177189982656,6.100562488944289,6.095271794475684,6.091281138250787,6.064934655399153,6.063196989929803,6.062132484134322,6.0597833543391,6.041393028175088,6.037288918904654,6.037134144352889,6.036908617802581,6.031448763254561,6.027187431006251,6.024580216286822,6.02140565504389,6.01898173666753,6.018608463138618,6.01557019023633,6.011613059692502,6.01092172586579,6.00699243747016,6.006683979273937,6.004614557029265,6.001622097614734,5.985616612160613,5.981559297271143,5.980779714168804,5.978785201251744,5.97805986131269,5.977272843160299,5.976425999514932,5.975185087560106,5.969540685328361,5.968420999238153,5.966339325952573,5.965605514462197,5.962711191590518,5.962149136882791,5.960362529962799,5.960264923441268,5.954038598316661,5.951547234547494,5.950284535264348,5.94864345620404,5.943115734457172,5.942780347089283,5.934440245069483,5.931538474222628,5.931002878331925,5.927341552097339,5.92647974704323,5.922706410368641,5.922144761916668,5.921096600637917,5.92028064270111,5.916618757700385,5.916512172356869,5.909754955819899,5.90894290720545,5.905569078462573,5.89989410017461,5.896300977917509,5.894676854263669,5.892718046913714,5.881844576195436,5.875179004478589,5.871986480792968,5.860459722522153,5.859047611485853,5.855776874903579,5.854849104425811,5.851149438268703,5.847363200772146,5.846323846212346,5.844626332437418,5.841199466186367,5.839227229581124,5.837292810344108,5.833203393881671,5.832937727591082,5.816140567693804,5.814966053016919,5.811567500727992,5.809862179031912,5.808636746736539,5.804495063437207,5.803556445013707,5.798478996442594,5.786740286071333,5.772976249601229,5.772247276676371,5.768777049100372,5.768056765573485,5.764145161978833,5.761239596363943,5.760423961253982,5.757818967118739,5.753700368924022,5.750349463954663,5.74961323711852,5.744538410588749,5.743873992865201,5.743740886274175,5.742731293497487,5.74039644067776,5.740385826127611,5.740113571861276,5.726871034379884,5.721464853968005,5.719017104049801,5.719010200098273,5.71239710463041],"arrows":["to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to"],"color":["red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","green","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","green","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false},"groups":{"TF":{"shape":"triangle","color":{"background":"purple","border":"black"}},"useDefaultGroups":true,"gene":{"shape":"dot","color":{"background":"teal","border":"black"}}}},"groups":["TF","gene"],"width":"100%","height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","legend":{"width":0.2,"useGroups":true,"position":"right","ncol":1,"stepX":100,"stepY":100,"zoom":true,"main":{"text":"Legend","style":"font-family:Georgia, Times New Roman, Times, serif;font-weight:bold;font-size:14px;text-align:center;"}}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a>in_degree_male<span class="ot">&lt;-</span> male_network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(gene) <span class="sc">|&gt;</span></span>
<span id="cb149-3"><a href="#cb149-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">in_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb149-4"><a href="#cb149-4" aria-hidden="true" tabindex="-1"></a>in_degree_female<span class="ot">&lt;-</span> female_network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(gene) <span class="sc">|&gt;</span></span>
<span id="cb149-5"><a href="#cb149-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">in_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb149-6"><a href="#cb149-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-7"><a href="#cb149-7" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_degree_male</span>
<span id="cb149-8"><a href="#cb149-8" aria-hidden="true" tabindex="-1"></a>in_difference<span class="sc">$</span>in_degree <span class="ot">&lt;-</span> in_degree_male<span class="sc">$</span>in_degree <span class="sc">-</span> in_degree_female<span class="sc">$</span>in_degree</span>
<span id="cb149-9"><a href="#cb149-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-10"><a href="#cb149-10" aria-hidden="true" tabindex="-1"></a>pathways <span class="ot">&lt;-</span> <span class="fu">gmtPathways</span>(<span class="st">"c2.cp.kegg_legacy.v2024.1.Hs.entrez.gmt"</span>)</span>
<span id="cb149-11"><a href="#cb149-11" aria-hidden="true" tabindex="-1"></a>in_difference<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(in_difference<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,in_difference<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb149-12"><a href="#cb149-12" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>entrezgene_id[<span class="fu">match</span>(in_difference<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb149-13"><a href="#cb149-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb149-14"><a href="#cb149-14" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_difference[<span class="sc">!</span><span class="fu">is.na</span>(in_difference<span class="sc">$</span>gene),]</span>
<span id="cb149-15"><a href="#cb149-15" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_difference[<span class="sc">!</span><span class="fu">duplicated</span>(in_difference<span class="sc">$</span>gene), ]</span>
<span id="cb149-16"><a href="#cb149-16" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> <span class="fu">setNames</span>(in_difference<span class="sc">$</span>in_degree, in_difference<span class="sc">$</span>gene)</span>
<span id="cb149-17"><a href="#cb149-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-18"><a href="#cb149-18" aria-hidden="true" tabindex="-1"></a>fgseaRes <span class="ot">&lt;-</span> <span class="fu">fgsea</span>(pathways, in_difference, <span class="at">minSize=</span><span class="dv">15</span>, <span class="at">maxSize=</span><span class="dv">500</span>,<span class="at">nPermSimple =</span> <span class="dv">10000</span>)</span>
<span id="cb149-19"><a href="#cb149-19" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span> fgseaRes[fgseaRes<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb149-20"><a href="#cb149-20" aria-hidden="true" tabindex="-1"></a>sig[<span class="fu">order</span>(sig<span class="sc">$</span>NES)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],] <span class="co">#Lower targeting in females, higher in males</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                              pathway         pval         padj
                                               &lt;char&gt;        &lt;num&gt;        &lt;num&gt;
 1: KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450 2.967564e-06 0.0005044858
 2:     KEGG_PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS 5.738848e-04 0.0139372015
 3:            KEGG_ASCORBATE_AND_ALDARATE_METABOLISM 9.880326e-04 0.0209956928
 4:                KEGG_STARCH_AND_SUCROSE_METABOLISM 3.924427e-04 0.0133430521
 5:                 KEGG_STEROID_HORMONE_BIOSYNTHESIS 5.348785e-04 0.0139372015
 6:                           KEGG_RETINOL_METABOLISM 2.501606e-04 0.0133430521
 7:              KEGG_DRUG_METABOLISM_CYTOCHROME_P450 1.706027e-04 0.0133430521
 8:                                  KEGG_SPLICEOSOME 1.900000e-03 0.0358888889
 9:                                   KEGG_PROTEASOME 3.378781e-04 0.0133430521
10:                                              &lt;NA&gt;           NA           NA
      log2err         ES       NES  size  leadingEdge
        &lt;num&gt;      &lt;num&gt;     &lt;num&gt; &lt;int&gt;       &lt;list&gt;
 1: 0.6272567 -0.4742504 -2.186725    52 54600, 5....
 2: 0.4772708 -0.5627321 -2.080729    21 54600, 5....
 3: 0.4550599 -0.5864141 -2.062658    18 54600, 5....
 4: 0.4984931 -0.4620650 -2.005675    39 54600, 5....
 5: 0.4772708 -0.4720310 -1.960873    33 54600, 5....
 6: 0.4984931 -0.4392137 -1.915102    40 54600, 5....
 7: 0.5188481 -0.4051360 -1.857277    53 2330, 54....
 8: 0.3350686  0.4840610  1.397247   126 84844, 3....
 9: 0.4984931  0.6281855  1.674398    41 5705, 56....
10:        NA         NA        NA    NA             </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>sig[<span class="fu">order</span>(<span class="sc">-</span>sig<span class="sc">$</span>NES)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],] <span class="co">#Lower targeting in males, higher in females</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                              pathway         pval         padj
                                               &lt;char&gt;        &lt;num&gt;        &lt;num&gt;
 1:                                   KEGG_PROTEASOME 3.378781e-04 0.0133430521
 2:                                  KEGG_SPLICEOSOME 1.900000e-03 0.0358888889
 3:              KEGG_DRUG_METABOLISM_CYTOCHROME_P450 1.706027e-04 0.0133430521
 4:                           KEGG_RETINOL_METABOLISM 2.501606e-04 0.0133430521
 5:                 KEGG_STEROID_HORMONE_BIOSYNTHESIS 5.348785e-04 0.0139372015
 6:                KEGG_STARCH_AND_SUCROSE_METABOLISM 3.924427e-04 0.0133430521
 7:            KEGG_ASCORBATE_AND_ALDARATE_METABOLISM 9.880326e-04 0.0209956928
 8:     KEGG_PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS 5.738848e-04 0.0139372015
 9: KEGG_METABOLISM_OF_XENOBIOTICS_BY_CYTOCHROME_P450 2.967564e-06 0.0005044858
10:                                              &lt;NA&gt;           NA           NA
      log2err         ES       NES  size  leadingEdge
        &lt;num&gt;      &lt;num&gt;     &lt;num&gt; &lt;int&gt;       &lt;list&gt;
 1: 0.4984931  0.6281855  1.674398    41 5705, 56....
 2: 0.3350686  0.4840610  1.397247   126 84844, 3....
 3: 0.5188481 -0.4051360 -1.857277    53 2330, 54....
 4: 0.4984931 -0.4392137 -1.915102    40 54600, 5....
 5: 0.4772708 -0.4720310 -1.960873    33 54600, 5....
 6: 0.4984931 -0.4620650 -2.005675    39 54600, 5....
 7: 0.4550599 -0.5864141 -2.062658    18 54600, 5....
 8: 0.4772708 -0.5627321 -2.080729    21 54600, 5....
 9: 0.6272567 -0.4742504 -2.186725    52 54600, 5....
10:        NA         NA        NA    NA             </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(fgseaRes)</span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Settings</span></span>
<span id="cb153-3"><a href="#cb153-3" aria-hidden="true" tabindex="-1"></a>fdrcut <span class="ot">&lt;-</span> <span class="fl">0.05</span> <span class="co"># FDR cut-off to use as output for significant signatures</span></span>
<span id="cb153-4"><a href="#cb153-4" aria-hidden="true" tabindex="-1"></a>dencol_neg <span class="ot">&lt;-</span> <span class="st">"blue"</span> <span class="co"># bubble plot color for negative ES</span></span>
<span id="cb153-5"><a href="#cb153-5" aria-hidden="true" tabindex="-1"></a>dencol_pos <span class="ot">&lt;-</span> <span class="st">"red"</span> <span class="co"># bubble plot color for positive ES</span></span>
<span id="cb153-6"><a href="#cb153-6" aria-hidden="true" tabindex="-1"></a>signnamelength <span class="ot">&lt;-</span> <span class="dv">4</span> <span class="co"># set to remove prefix from signature names (2 for "GO", 4 for "KEGG", 8 for "REACTOME")</span></span>
<span id="cb153-7"><a href="#cb153-7" aria-hidden="true" tabindex="-1"></a>asp <span class="ot">&lt;-</span> <span class="dv">3</span> <span class="co"># aspect ratio of bubble plot</span></span>
<span id="cb153-8"><a href="#cb153-8" aria-hidden="true" tabindex="-1"></a>charcut <span class="ot">&lt;-</span> <span class="dv">100</span> <span class="co"># cut signature name in heatmap to this nr of characters</span></span>
<span id="cb153-9"><a href="#cb153-9" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">as.character</span>(dat<span class="sc">$</span>pathway) <span class="co"># 'a' is a great variable name to substitute row names with something more readable</span></span>
<span id="cb153-10"><a href="#cb153-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(a)){</span>
<span id="cb153-11"><a href="#cb153-11" aria-hidden="true" tabindex="-1"></a>  a[j] <span class="ot">&lt;-</span> <span class="fu">substr</span>(a[j], signnamelength<span class="sc">+</span><span class="dv">2</span>, <span class="fu">nchar</span>(a[j]))</span>
<span id="cb153-12"><a href="#cb153-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb153-13"><a href="#cb153-13" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">tolower</span>(a) <span class="co"># convert to lower case (you may want to comment this out, it really depends on what signatures you are looking at, c6 signatures contain gene names, and converting those to lower case may be confusing)</span></span>
<span id="cb153-14"><a href="#cb153-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(a)){</span>
<span id="cb153-15"><a href="#cb153-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nchar</span>(a[j])<span class="sc">&gt;</span>charcut) { a[j] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">substr</span>(a[j], <span class="dv">1</span>, charcut), <span class="st">"..."</span>, <span class="at">sep=</span><span class="st">" "</span>)}</span>
<span id="cb153-16"><a href="#cb153-16" aria-hidden="true" tabindex="-1"></a>} <span class="co"># cut signature names that have more characters than charcut, and add "..."</span></span>
<span id="cb153-17"><a href="#cb153-17" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, a)</span>
<span id="cb153-18"><a href="#cb153-18" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>NAME <span class="ot">&lt;-</span> a</span>
<span id="cb153-19"><a href="#cb153-19" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat[dat[,<span class="st">"padj"</span>]<span class="sc">&lt;</span>fdrcut,]</span>
<span id="cb153-20"><a href="#cb153-20" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat2[<span class="fu">order</span>(dat2[,<span class="st">"padj"</span>]),] </span>
<span id="cb153-21"><a href="#cb153-21" aria-hidden="true" tabindex="-1"></a>dat2<span class="sc">$</span>signature <span class="ot">&lt;-</span> <span class="fu">factor</span>(dat2<span class="sc">$</span>NAME, <span class="fu">rev</span>(<span class="fu">as.character</span>(dat2<span class="sc">$</span>NAME)))</span>
<span id="cb153-22"><a href="#cb153-22" aria-hidden="true" tabindex="-1"></a>sign_neg <span class="ot">&lt;-</span> <span class="fu">which</span>(dat2[,<span class="st">"NES"</span>]<span class="sc">&lt;</span><span class="dv">0</span>) </span>
<span id="cb153-23"><a href="#cb153-23" aria-hidden="true" tabindex="-1"></a>sign_pos <span class="ot">&lt;-</span> <span class="fu">which</span>(dat2[,<span class="st">"NES"</span>]<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb153-24"><a href="#cb153-24" aria-hidden="true" tabindex="-1"></a>signcol <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(dat2<span class="sc">$</span>signature))</span>
<span id="cb153-25"><a href="#cb153-25" aria-hidden="true" tabindex="-1"></a>signcol[sign_neg] <span class="ot">&lt;-</span> dencol_neg <span class="co"># text color of negative signatures</span></span>
<span id="cb153-26"><a href="#cb153-26" aria-hidden="true" tabindex="-1"></a>signcol[sign_pos] <span class="ot">&lt;-</span> dencol_pos <span class="co"># text color of positive signatures</span></span>
<span id="cb153-27"><a href="#cb153-27" aria-hidden="true" tabindex="-1"></a>signcol <span class="ot">&lt;-</span> <span class="fu">rev</span>(signcol) <span class="co"># need to revert vector of colors, because ggplot starts plotting these from below</span></span>
<span id="cb153-28"><a href="#cb153-28" aria-hidden="true" tabindex="-1"></a>g<span class="ot">&lt;-</span><span class="fu">ggplot</span>(dat2, <span class="fu">aes</span>(<span class="at">x=</span>padj,<span class="at">y=</span>signature,<span class="at">size=</span>size))</span>
<span id="cb153-29"><a href="#cb153-29" aria-hidden="true" tabindex="-1"></a>g<span class="sc">+</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill=</span>NES), <span class="at">shape=</span><span class="dv">21</span>, <span class="at">colour=</span><span class="st">"white"</span>)<span class="sc">+</span></span>
<span id="cb153-30"><a href="#cb153-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span> <span class="co"># white background, needs to be placed before the "signcol" line</span></span>
<span id="cb153-31"><a href="#cb153-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>,fdrcut)<span class="sc">+</span></span>
<span id="cb153-32"><a href="#cb153-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_area</span>(<span class="at">max_size=</span><span class="dv">10</span>,<span class="at">guide=</span><span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb153-33"><a href="#cb153-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low=</span>dencol_neg, <span class="at">high=</span>dencol_pos)<span class="sc">+</span></span>
<span id="cb153-34"><a href="#cb153-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">colour=</span>signcol))<span class="sc">+</span></span>
<span id="cb153-35"><a href="#cb153-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">aspect.ratio=</span>asp, <span class="at">axis.title.y=</span><span class="fu">element_blank</span>()) <span class="co"># test aspect.ratio</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Vectorized input to `element_text()` is not officially supported.
ℹ Results may be unexpected or may change in future versions of ggplot2.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BLUE is upregulated in females</span></span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a><span class="do">##RED is upgregulated in males</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="analysis-by-diabetes-status" class="level2">
<h2 class="anchored" data-anchor-id="analysis-by-diabetes-status">Analysis by Diabetes status</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colData</span>(tcga)<span class="sc">@</span>listData<span class="sc">$</span>tcga.xml_history_of_diabetes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 NO YES 
106  35 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb158"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb158-1"><a href="#cb158-1" aria-hidden="true" tabindex="-1"></a>non_na_samples <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">colData</span>(tcga)<span class="sc">$</span>tcga.xml_history_of_diabetes)</span>
<span id="cb158-2"><a href="#cb158-2" aria-hidden="true" tabindex="-1"></a>tcga_exprs_filtered_diab <span class="ot">&lt;-</span> <span class="fu">assay</span>(tcga_filtered, <span class="st">"raw_counts"</span>)[, non_na_samples]</span>
<span id="cb158-3"><a href="#cb158-3" aria-hidden="true" tabindex="-1"></a>tcga_colData_filtered_diab <span class="ot">&lt;-</span> <span class="fu">colData</span>(tcga_filtered)[non_na_samples, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb159"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a>condition_diabetes <span class="ot">&lt;-</span> <span class="fu">factor</span>(tcga_colData_filtered_diab<span class="sc">$</span>tcga.xml_history_of_diabetes, </span>
<span id="cb159-2"><a href="#cb159-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"NO"</span>, <span class="st">"YES"</span>))</span>
<span id="cb159-3"><a href="#cb159-3" aria-hidden="true" tabindex="-1"></a>design_diabetes <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> condition_diabetes)</span>
<span id="cb159-4"><a href="#cb159-4" aria-hidden="true" tabindex="-1"></a>voomOutput_diabetes <span class="ot">&lt;-</span> <span class="fu">voom</span>(tcga_exprs_filtered_diab, design_diabetes, <span class="at">plot=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb160"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb160-1"><a href="#cb160-1" aria-hidden="true" tabindex="-1"></a>fit_diabetes <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(voomOutput_diabetes, design_diabetes)</span>
<span id="cb160-2"><a href="#cb160-2" aria-hidden="true" tabindex="-1"></a>fit_diabetes <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit_diabetes)</span>
<span id="cb160-3"><a href="#cb160-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSA</span>(fit_diabetes, <span class="at">main=</span><span class="st">"Final model mean-variance trend"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-47-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb161"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="co"># DEGs</span></span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a>res_diabetes <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit_diabetes, <span class="at">coef=</span><span class="st">"condition_diabetesYES"</span>, <span class="at">number=</span><span class="cn">Inf</span>)</span>
<span id="cb161-3"><a href="#cb161-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_diabetes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                        logFC  AveExpr         t      P.Value adj.P.Val
ENSG00000226715.2   0.4741746 4.814916  3.540581 0.0005400814 0.9389277
ENSG00000143742.12 -0.2747228 6.660782 -3.360077 0.0010005453 0.9389277
ENSG00000226137.5  -0.3910933 4.434697 -3.481544 0.0006624521 0.9389277
ENSG00000094975.13 -0.3325307 5.526553 -3.281618 0.0012985194 0.9389277
ENSG00000145022.4  -0.3553842 4.991389 -3.322223 0.0011352749 0.9389277
ENSG00000211772.11  1.0091612 4.250333  3.331550 0.0011005873 0.9389277
                           B
ENSG00000226715.2  -2.090732
ENSG00000143742.12 -2.290789
ENSG00000226137.5  -2.344184
ENSG00000094975.13 -2.433390
ENSG00000145022.4  -2.442673
ENSG00000211772.11 -2.456270</code></pre>
</div>
<div class="sourceCode cell-code" id="cb163"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb163-1"><a href="#cb163-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of genes differentially expressed</span></span>
<span id="cb163-2"><a href="#cb163-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_diabetes<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(res_diabetes<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb165"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb165-1"><a href="#cb165-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of genes differentially expressed</span></span>
<span id="cb165-2"><a href="#cb165-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_diabetes<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(res_diabetes<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb167"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb167-1"><a href="#cb167-1" aria-hidden="true" tabindex="-1"></a><span class="co"># overexpressed in TCGA</span></span>
<span id="cb167-2"><a href="#cb167-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_diabetes<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> res_diabetes<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="co"># overexpressed in GTEX</span></span>
<span id="cb169-2"><a href="#cb169-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_diabetes<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> res_diabetes<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
<section id="gene-network-analysis-1" class="level3">
<h3 class="anchored" data-anchor-id="gene-network-analysis-1">Gene Network Analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Filtering based on PANDA requirements</span></span>
<span id="cb171-2"><a href="#cb171-2" aria-hidden="true" tabindex="-1"></a>tcga_filtered_nas <span class="ot">&lt;-</span> tcga_filtered[,<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">colData</span>(tcga)<span class="sc">$</span>tcga.xml_history_of_diabetes)]</span>
<span id="cb171-3"><a href="#cb171-3" aria-hidden="true" tabindex="-1"></a>no_diabetes <span class="ot">&lt;-</span> tcga_filtered_nas[,tcga_filtered_nas<span class="sc">$</span>tcga.xml_history_of_diabetes<span class="sc">==</span><span class="st">'NO'</span>]</span>
<span id="cb171-4"><a href="#cb171-4" aria-hidden="true" tabindex="-1"></a>diabetes <span class="ot">&lt;-</span> tcga_filtered_nas[,tcga_filtered_nas<span class="sc">$</span>tcga.xml_history_of_diabetes<span class="sc">==</span><span class="st">'YES'</span>]</span>
<span id="cb171-5"><a href="#cb171-5" aria-hidden="true" tabindex="-1"></a>zero_na_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assay</span>(no_diabetes, <span class="st">"logtpm"</span>), <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="fu">length</span>(x[(<span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x<span class="sc">!=</span><span class="dv">0</span>) ]))</span>
<span id="cb171-6"><a href="#cb171-6" aria-hidden="true" tabindex="-1"></a>expr_no_diabetes <span class="ot">&lt;-</span> <span class="fu">assay</span>(no_diabetes, <span class="st">"logtpm"</span>)[zero_na_counts <span class="sc">&gt;</span> <span class="dv">20</span>,]</span>
<span id="cb171-7"><a href="#cb171-7" aria-hidden="true" tabindex="-1"></a>zero_na_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assay</span>(diabetes, <span class="st">"logtpm"</span>), <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="fu">length</span>(x[(<span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x<span class="sc">!=</span><span class="dv">0</span>) ]))</span>
<span id="cb171-8"><a href="#cb171-8" aria-hidden="true" tabindex="-1"></a>expr_diabetes <span class="ot">&lt;-</span> <span class="fu">assay</span>(diabetes, <span class="st">"logtpm"</span>)[zero_na_counts <span class="sc">&gt;</span> <span class="dv">20</span>,]</span>
<span id="cb171-9"><a href="#cb171-9" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(expr_no_diabetes) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">row.names</span>(expr_no_diabetes))</span>
<span id="cb171-10"><a href="#cb171-10" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(expr_diabetes) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">row.names</span>(expr_diabetes))</span>
<span id="cb171-11"><a href="#cb171-11" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(expr_no_diabetes, file = "no_diabetes.txt", sep = "\t", quote = FALSE, row.names = TRUE)</span></span>
<span id="cb171-12"><a href="#cb171-12" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(expr_diabetes, file = "diabetes.txt", sep = "\t", quote = FALSE, row.names = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb172"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb172-1"><a href="#cb172-1" aria-hidden="true" tabindex="-1"></a>network_full <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"output_diabetes.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb172-2"><a href="#cb172-2" aria-hidden="true" tabindex="-1"></a>network <span class="ot">&lt;-</span> network_full[<span class="fu">order</span>(network_full<span class="sc">$</span>force, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb172-3"><a href="#cb172-3" aria-hidden="true" tabindex="-1"></a>network <span class="ot">&lt;-</span> network[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>,]</span>
<span id="cb172-4"><a href="#cb172-4" aria-hidden="true" tabindex="-1"></a>network <span class="ot">&lt;-</span> <span class="fu">subset</span>(network, <span class="at">select=</span><span class="sc">-</span>motif)</span>
<span id="cb172-5"><a href="#cb172-5" aria-hidden="true" tabindex="-1"></a>network<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(network<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,network<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb172-6"><a href="#cb172-6" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>hgnc_symbol[<span class="fu">match</span>(network<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb172-7"><a href="#cb172-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb172-8"><a href="#cb172-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(network,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              tf            gene    force
20630731  ZNF341 ENSG00000268836 20.18954
21201026 ZNF385D ENSG00000272829 17.59784
21802217 ZNF385D ENSG00000277368 16.92181
20554970 ZNF385D ENSG00000268129 15.75676
16542045 ZNF385D      ZNF687-AS1 15.44385
10453366 ZNF385D           RSPO1 15.34556
5260990  ZNF385D           HOXD1 14.99427
18230191   CEBPB     ALDH1L1-AS1 14.63922
11360636 ZNF385D           DDIT3 14.62481
14237978 ZNF385D         TMEM240 14.61684</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb174"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb174-1"><a href="#cb174-1" aria-hidden="true" tabindex="-1"></a>normal_network_full <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"output_no_diabetes.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb174-2"><a href="#cb174-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-3"><a href="#cb174-3" aria-hidden="true" tabindex="-1"></a>normal_network_full <span class="ot">&lt;-</span> normal_network_full[normal_network_full<span class="sc">$</span>gene <span class="sc">%in%</span> network_full<span class="sc">$</span>gene,]</span>
<span id="cb174-4"><a href="#cb174-4" aria-hidden="true" tabindex="-1"></a>normal_network_full <span class="ot">&lt;-</span> normal_network_full[normal_network_full<span class="sc">$</span>tf <span class="sc">%in%</span> network_full<span class="sc">$</span>tf,]</span>
<span id="cb174-5"><a href="#cb174-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb174-6"><a href="#cb174-6" aria-hidden="true" tabindex="-1"></a>diffnet <span class="ot">&lt;-</span> network_full</span>
<span id="cb174-7"><a href="#cb174-7" aria-hidden="true" tabindex="-1"></a>diffnet<span class="sc">$</span>force <span class="ot">&lt;-</span> network_full<span class="sc">$</span>force<span class="sc">-</span>normal_network_full<span class="sc">$</span>force</span>
<span id="cb174-8"><a href="#cb174-8" aria-hidden="true" tabindex="-1"></a>diffnet <span class="ot">&lt;-</span> <span class="fu">subset</span>(diffnet, <span class="at">select=</span><span class="sc">-</span>motif)</span>
<span id="cb174-9"><a href="#cb174-9" aria-hidden="true" tabindex="-1"></a>diffnet<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(diffnet<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,diffnet<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb174-10"><a href="#cb174-10" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>hgnc_symbol[<span class="fu">match</span>(diffnet<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb174-11"><a href="#cb174-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb174-12"><a href="#cb174-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(diffnet,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       tf   gene      force
1    ALX1 TSPAN6  0.7483091
2    ALX3 TSPAN6  0.1993255
3    ALX4 TSPAN6  0.7199155
4      AR TSPAN6 -0.3288039
5   ARGFX TSPAN6  0.5722896
6  ARID3A TSPAN6  0.9593037
7  ARID5B TSPAN6  0.8563283
8    ARNT TSPAN6  0.1521822
9   ARNT2 TSPAN6  0.1550007
10  ARNTL TSPAN6  0.1357747</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> diffnet</span>
<span id="cb176-2"><a href="#cb176-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(edges) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"from"</span>, <span class="st">"to"</span>, <span class="st">"value"</span>)</span>
<span id="cb176-3"><a href="#cb176-3" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>arrows <span class="ot">=</span> <span class="st">"to"</span>   </span>
<span id="cb176-4"><a href="#cb176-4" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>color  <span class="ot">=</span> <span class="fu">ifelse</span>(edges<span class="sc">$</span>value <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"green"</span>, <span class="st">"red"</span>)</span>
<span id="cb176-5"><a href="#cb176-5" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>value  <span class="ot">=</span> <span class="fu">abs</span>(edges<span class="sc">$</span>value)</span>
<span id="cb176-6"><a href="#cb176-6" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges[<span class="fu">order</span>(edges<span class="sc">$</span>value, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb176-7"><a href="#cb176-7" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>,]</span>
<span id="cb176-8"><a href="#cb176-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(edges,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           from       to    value arrows color
17082365 ZNF235 MYCL-AS1 8.758660     to   red
17082387 ZNF287 MYCL-AS1 8.603691     to   red
18589829 ZNF235    PINX1 8.414052     to green
17082100  PRDM6 MYCL-AS1 8.374263     to   red
1306834  ZNF235    MKNK1 8.254749     to   red
1306856  ZNF287    MKNK1 8.240481     to   red
17082362 ZNF225 MYCL-AS1 8.228499     to   red
18589851 ZNF287    PINX1 8.226254     to green
17082456 ZNF487 MYCL-AS1 8.202573     to   red
18589564  PRDM6    PINX1 8.196016     to green</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb178"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb178-1"><a href="#cb178-1" aria-hidden="true" tabindex="-1"></a>nodes     <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))), </span>
<span id="cb178-2"><a href="#cb178-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">label=</span><span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))))</span>
<span id="cb178-3"><a href="#cb178-3" aria-hidden="true" tabindex="-1"></a>nodes<span class="sc">$</span>group <span class="ot">=</span> <span class="fu">ifelse</span>(nodes<span class="sc">$</span>id <span class="sc">%in%</span> edges<span class="sc">$</span>from, <span class="st">"TF"</span>, <span class="st">"gene"</span>)</span>
<span id="cb178-4"><a href="#cb178-4" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visNetwork</span>(nodes, edges, <span class="at">width =</span> <span class="st">"100%"</span>)</span>
<span id="cb178-5"><a href="#cb178-5" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"TF"</span>, <span class="at">shape =</span> <span class="st">"triangle"</span>,</span>
<span id="cb178-6"><a href="#cb178-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"purple"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb178-7"><a href="#cb178-7" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"gene"</span>, <span class="at">shape =</span> <span class="st">"dot"</span>,       </span>
<span id="cb178-8"><a href="#cb178-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"teal"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb178-9"><a href="#cb178-9" aria-hidden="true" tabindex="-1"></a><span class="fu">visLegend</span>(net, <span class="at">main=</span><span class="st">"Legend"</span>, <span class="at">position=</span><span class="st">"right"</span>, <span class="at">ncol=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="htmlwidget-715ab92cff8b55589ba5" style="width:100%;height:480px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-715ab92cff8b55589ba5">{"x":{"nodes":{"id":["ZNF235","ZNF287","PRDM6","ZNF225","ZNF487","ZNF418","ZFP28","ZNF879","ZNF98","ZNF571","ZNF182","ZNF570","ZNF492","ZNF613","IRF1","ZNF567","ZNF432","ZNF467","MYCL-AS1","PINX1","MKNK1","STC2","NEIL2","MKNK1-AS1","COMMD6","SESN2","MYCL","ENSG00000276185","RRBP1","ENSG00000228843","ARL16","TMEM176B","ENSG00000236266","TIMP1","WDR72","TMEM176A","CHIT1","MIR7705","RPS26P31","AGT","ENSG00000243403","C19orf53","USP27X","RPS26P28","MID1IP1-AS1","ENSG00000196656","RPL23"],"label":["ZNF235","ZNF287","PRDM6","ZNF225","ZNF487","ZNF418","ZFP28","ZNF879","ZNF98","ZNF571","ZNF182","ZNF570","ZNF492","ZNF613","IRF1","ZNF567","ZNF432","ZNF467","MYCL-AS1","PINX1","MKNK1","STC2","NEIL2","MKNK1-AS1","COMMD6","SESN2","MYCL","ENSG00000276185","RRBP1","ENSG00000228843","ARL16","TMEM176B","ENSG00000236266","TIMP1","WDR72","TMEM176A","CHIT1","MIR7705","RPS26P31","AGT","ENSG00000243403","C19orf53","USP27X","RPS26P28","MID1IP1-AS1","ENSG00000196656","RPL23"],"group":["TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene"]},"edges":{"from":["ZNF235","ZNF287","ZNF235","PRDM6","ZNF235","ZNF287","ZNF225","ZNF287","ZNF487","PRDM6","ZNF235","ZNF487","ZNF418","ZNF225","ZNF235","ZNF235","PRDM6","ZNF487","ZNF287","ZNF287","ZNF287","ZNF418","ZNF225","ZFP28","ZFP28","PRDM6","PRDM6","PRDM6","ZNF487","ZNF225","ZNF235","ZNF879","ZNF487","ZNF235","ZNF487","ZNF418","ZNF287","ZNF225","ZNF879","ZNF225","ZNF418","ZNF418","ZNF98","ZNF235","ZNF571","ZNF287","PRDM6","ZFP28","ZNF98","PRDM6","ZNF287","ZNF182","ZFP28","ZNF487","ZNF235","ZNF418","ZNF235","ZNF418","ZNF225","ZNF570","ZNF879","ZNF879","ZNF487","ZNF879","ZFP28","ZNF287","ZFP28","ZNF287","ZNF98","ZNF492","ZNF225","ZNF235","ZNF613","ZNF570","ZNF235","ZNF418","ZNF235","PRDM6","ZNF182","ZNF287","ZNF235","ZNF571","ZNF182","IRF1","IRF1","PRDM6","ZNF287","ZNF98","ZNF235","PRDM6","ZNF98","ZNF879","ZNF879","ZNF567","ZNF235","ZNF418","ZNF235","ZNF287","ZNF487","ZFP28","ZNF287","ZNF182","ZNF98","ZNF225","ZNF225","ZNF487","ZNF287","ZFP28","ZNF235","ZNF287","ZNF182","ZNF225","ZNF418","ZNF487","ZNF235","ZNF225","PRDM6","ZNF570","ZNF418","ZNF487","ZNF287","ZNF879","ZNF287","ZNF287","ZNF570","ZNF492","PRDM6","ZNF235","ZNF225","ZNF492","ZNF571","PRDM6","ZNF571","ZNF287","ZNF613","ZNF879","ZNF235","ZNF418","ZNF613","ZNF287","ZNF225","PRDM6","ZNF570","ZNF98","ZNF487","PRDM6","IRF1","ZNF235","ZNF225","ZNF418","ZNF235","ZNF567","PRDM6","ZNF487","ZNF432","ZNF487","ZNF487","ZNF98","ZNF235","ZNF613","ZNF235","PRDM6","ZNF570","ZNF225","PRDM6","PRDM6","ZNF487","IRF1","ZNF418","ZFP28","ZNF235","ZNF487","ZNF287","ZNF487","ZNF418","ZNF287","ZNF492","ZNF182","ZNF235","ZNF487","ZNF492","ZNF613","PRDM6","ZFP28","ZNF287","ZNF225","ZNF225","ZNF879","ZNF235","ZNF182","ZNF225","ZNF418","ZNF287","ZNF182","ZNF879","ZNF287","IRF1","ZNF225","ZFP28","ZNF467"],"to":["MYCL-AS1","MYCL-AS1","PINX1","MYCL-AS1","MKNK1","MKNK1","MYCL-AS1","PINX1","MYCL-AS1","PINX1","STC2","PINX1","MYCL-AS1","PINX1","NEIL2","MKNK1-AS1","MKNK1","MKNK1","MKNK1-AS1","NEIL2","STC2","PINX1","MKNK1","MYCL-AS1","PINX1","NEIL2","MKNK1-AS1","STC2","MKNK1-AS1","STC2","COMMD6","MYCL-AS1","STC2","SESN2","NEIL2","STC2","COMMD6","NEIL2","PINX1","MKNK1-AS1","NEIL2","MKNK1","PINX1","MYCL","PINX1","SESN2","COMMD6","MKNK1","MYCL-AS1","SESN2","MYCL","PINX1","NEIL2","COMMD6","ENSG00000276185","SESN2","RRBP1","MKNK1-AS1","SESN2","PINX1","NEIL2","STC2","SESN2","MKNK1","MKNK1-AS1","RRBP1","STC2","ENSG00000276185","MKNK1","PINX1","COMMD6","ENSG00000228843","PINX1","MYCL-AS1","ARL16","COMMD6","TMEM176B","MYCL","MKNK1","TMEM176B","ENSG00000236266","TIMP1","MYCL-AS1","MYCL-AS1","PINX1","RRBP1","ENSG00000228843","STC2","WDR72","ENSG00000276185","MKNK1-AS1","MKNK1-AS1","SESN2","PINX1","TMEM176A","ENSG00000228843","CHIT1","ARL16","MYCL","COMMD6","TMEM176A","MKNK1-AS1","NEIL2","RRBP1","ENSG00000276185","RRBP1","ENSG00000236266","SESN2","MIR7705","WDR72","STC2","MYCL","MYCL","ENSG00000276185","TIMP1","ARL16","TMEM176B","MKNK1","ENSG00000276185","TMEM176B","MIR7705","COMMD6","TIMP1","CHIT1","NEIL2","MYCL-AS1","ENSG00000228843","RPS26P31","ENSG00000228843","MKNK1","MKNK1-AS1","WDR72","MKNK1","RPS26P31","MYCL-AS1","ENSG00000228843","AGT","CHIT1","MKNK1","AGT","TMEM176B","ENSG00000236266","STC2","SESN2","ENSG00000236266","ARL16","NEIL2","ENSG00000243403","ENSG00000236266","TMEM176B","C19orf53","MYCL-AS1","TMEM176A","WDR72","PINX1","ENSG00000228843","TMEM176A","COMMD6","USP27X","STC2","RPS26P28","MIR7705","MKNK1-AS1","WDR72","TIMP1","RPS26P31","TIMP1","MKNK1","WDR72","ENSG00000276185","MID1IP1-AS1","ARL16","C19orf53","RPS26P31","ARL16","USP27X","MKNK1-AS1","SESN2","ENSG00000196656","MIR7705","STC2","MKNK1-AS1","CHIT1","RRBP1","MID1IP1-AS1","TMEM176A","CHIT1","ENSG00000276185","RPL23","COMMD6","TIMP1","RRBP1","ENSG00000243403","NEIL2","WDR72","RPS26P28","MKNK1-AS1","MIR7705","MYCL","PINX1"],"value":[8.758660038065784,8.603690592090482,8.414052415455224,8.374263457960158,8.254748890832008,8.240480674096801,8.228498500045232,8.226254397206176,8.202573287160469,8.196016166093687,8.089082851122816,8.086717621826674,8.075531084428517,8.017137873719063,8.015826103215499,7.995896734588772,7.973951197746869,7.97384277303486,7.965385489519962,7.923511499735897,7.909457860266777,7.889547492260707,7.806243470747497,7.777245822746486,7.761964368320801,7.744781777236682,7.739387794255764,7.734723934749173,7.718787392896306,7.691872015236061,7.666986717024432,7.658752703300707,7.648785275587841,7.627378511937904,7.601712999293023,7.600796640349019,7.582143708397355,7.571345838246076,7.570239337915152,7.556040728351808,7.544853699048069,7.527567167082429,7.499976429930338,7.474325924769378,7.415994549829435,7.414582945948109,7.413126565568859,7.397930658994964,7.383127031335565,7.372941030640306,7.370591092061069,7.367321207578783,7.353574477581971,7.339017716716231,7.328517833816089,7.3267748697104,7.323445891833325,7.308543089324387,7.260795853110058,7.258782668120492,7.249363373741833,7.240313309467267,7.233262714433876,7.230896674406556,7.214415339973979,7.214401611923832,7.214324112256278,7.197259444427651,7.19229109214271,7.184334271920068,7.183578482947682,7.170484443961984,7.168230548456216,7.157421483499425,7.151993091614527,7.144965222265408,7.124321916694122,7.115434417274606,7.099673447237096,7.083333363852592,7.070340331725221,7.06962026780773,7.066460893616861,7.043544906250181,7.035310211061598,7.029682035049966,7.017790187219001,7.014747831551581,7.00797348995798,7.004650816776021,7.002868775502442,6.999518141175372,6.98590072728858,6.984414854090659,6.984370316538085,6.979001560212494,6.969226992576898,6.967025072890236,6.956448510269045,6.9541690620299,6.952304738332325,6.94783018732493,6.942328361887627,6.942003319359607,6.938956316342075,6.928377486497564,6.921773459199766,6.920852964726404,6.913658275190675,6.912480125120551,6.908705470196459,6.901663476804028,6.89526051506467,6.889517356051949,6.884621444523036,6.853212591001023,6.850999528088721,6.850856854130832,6.846857063879832,6.838153244033559,6.832045026962312,6.831997808227054,6.830818366556573,6.8302410719789,6.828362478892696,6.826630890782752,6.825969012858842,6.810566697181597,6.795829040471388,6.794259728810169,6.792950471999795,6.792593823483025,6.790485113966108,6.778148908405134,6.772582782828599,6.769453996408103,6.765397160672716,6.763413831284292,6.760529045947252,6.758935941580139,6.751846928185094,6.7478194882895,6.742198813290125,6.737823892148217,6.736871688420528,6.730057357714832,6.726644753037387,6.722908967441171,6.722477764897833,6.716896121620793,6.714623503545351,6.713524734630771,6.713261384746174,6.705748616868194,6.705232527923285,6.701292016644168,6.698896983056139,6.697899811752404,6.695719062223029,6.693069924520961,6.691621623086734,6.688148067464894,6.686045688641149,6.685779641622942,6.681398146763163,6.678836153376986,6.677200570665294,6.676697711377193,6.676205756122084,6.667793217265604,6.665093476412669,6.664523124783525,6.663450402359483,6.650377400579991,6.642722728493761,6.640285648979366,6.639059840299629,6.628142340089448,6.626632596288474,6.620301380255364,6.618747081487257,6.612836297157063,6.611527044949727,6.606409448066426,6.604683125806874,6.603328700605246,6.597730138466158,6.595170408662725,6.593539321062797,6.592446977737001,6.590918768727272,6.590480164694853,6.578600705477578,6.57572934163113,6.565228704146422,6.563094527493035,6.546401473669583,6.543401114976672,6.541720649994724,6.54063768355266],"arrows":["to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to"],"color":["red","red","green","red","red","red","red","green","red","green","red","green","red","green","green","red","red","red","red","green","red","green","red","red","green","green","red","red","red","red","green","red","red","red","green","red","green","green","green","red","green","red","green","red","green","red","green","red","red","red","red","green","green","green","green","red","red","red","red","green","green","red","red","red","red","red","red","green","red","green","green","green","green","red","green","green","green","red","red","green","red","green","red","red","green","red","green","red","red","green","red","red","red","green","green","green","red","green","red","green","green","red","green","red","green","red","red","red","green","red","red","red","red","green","green","green","green","red","green","green","green","green","green","red","green","red","green","green","green","red","red","red","red","green","red","green","green","red","red","green","green","red","red","red","red","green","green","green","red","green","green","red","green","red","green","green","green","green","green","red","green","green","red","red","green","green","green","red","red","green","red","green","green","green","green","green","red","red","green","green","red","red","red","red","red","green","red","green","green","green","green","red","green","green","red","green","red","green","red","green"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false},"groups":{"TF":{"shape":"triangle","color":{"background":"purple","border":"black"}},"useDefaultGroups":true,"gene":{"shape":"dot","color":{"background":"teal","border":"black"}}}},"groups":["TF","gene"],"width":"100%","height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","legend":{"width":0.2,"useGroups":true,"position":"right","ncol":1,"stepX":100,"stepY":100,"zoom":true,"main":{"text":"Legend","style":"font-family:Georgia, Times New Roman, Times, serif;font-weight:bold;font-size:14px;text-align:center;"}}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb179"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb179-1"><a href="#cb179-1" aria-hidden="true" tabindex="-1"></a><span class="co">#GREEN is upregulated in diabetes</span></span>
<span id="cb179-2"><a href="#cb179-2" aria-hidden="true" tabindex="-1"></a><span class="co">#RED is downregulated in diabetes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb180"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb180-1"><a href="#cb180-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb180-2"><a href="#cb180-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb180-3"><a href="#cb180-3" aria-hidden="true" tabindex="-1"></a>in_degree_normal<span class="ot">&lt;-</span> normal_network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(gene) <span class="sc">|&gt;</span></span>
<span id="cb180-4"><a href="#cb180-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">in_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb180-5"><a href="#cb180-5" aria-hidden="true" tabindex="-1"></a>in_degree<span class="ot">&lt;-</span> network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(gene) <span class="sc">|&gt;</span></span>
<span id="cb180-6"><a href="#cb180-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">in_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb180-7"><a href="#cb180-7" aria-hidden="true" tabindex="-1"></a>in_degree <span class="ot">&lt;-</span> in_degree[in_degree<span class="sc">$</span>gene <span class="sc">%in%</span> in_degree_normal<span class="sc">$</span>gene,]</span>
<span id="cb180-8"><a href="#cb180-8" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_degree_normal</span>
<span id="cb180-9"><a href="#cb180-9" aria-hidden="true" tabindex="-1"></a>in_difference<span class="sc">$</span>in_degree <span class="ot">&lt;-</span> in_degree_normal<span class="sc">$</span>in_degree <span class="sc">-</span> in_degree<span class="sc">$</span>in_degree</span>
<span id="cb180-10"><a href="#cb180-10" aria-hidden="true" tabindex="-1"></a>pathways <span class="ot">&lt;-</span> <span class="fu">gmtPathways</span>(<span class="st">"c2.cp.kegg_legacy.v2024.1.Hs.entrez.gmt"</span>)</span>
<span id="cb180-11"><a href="#cb180-11" aria-hidden="true" tabindex="-1"></a>in_difference<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(in_difference<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,in_difference<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb180-12"><a href="#cb180-12" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>entrezgene_id[<span class="fu">match</span>(in_difference<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb180-13"><a href="#cb180-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb180-14"><a href="#cb180-14" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_difference[<span class="sc">!</span><span class="fu">is.na</span>(in_difference<span class="sc">$</span>gene),]</span>
<span id="cb180-15"><a href="#cb180-15" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_difference[<span class="sc">!</span><span class="fu">duplicated</span>(in_difference<span class="sc">$</span>gene), ]</span>
<span id="cb180-16"><a href="#cb180-16" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> <span class="fu">setNames</span>(in_difference<span class="sc">$</span>in_degree, in_difference<span class="sc">$</span>gene)</span>
<span id="cb180-17"><a href="#cb180-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb180-18"><a href="#cb180-18" aria-hidden="true" tabindex="-1"></a>fgseaRes <span class="ot">&lt;-</span> <span class="fu">fgsea</span>(pathways, in_difference, <span class="at">minSize=</span><span class="dv">15</span>, <span class="at">maxSize=</span><span class="dv">500</span>,<span class="at">nPermSimple =</span> <span class="dv">10000</span>)</span>
<span id="cb180-19"><a href="#cb180-19" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span> fgseaRes[fgseaRes<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb180-20"><a href="#cb180-20" aria-hidden="true" tabindex="-1"></a>sig[<span class="fu">order</span>(sig<span class="sc">$</span>NES)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],] <span class="co">#Lower targeting in diabetes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                           pathway         pval         padj
                                            &lt;char&gt;        &lt;num&gt;        &lt;num&gt;
 1:       KEGG_COMPLEMENT_AND_COAGULATION_CASCADES 3.407282e-04 0.0072404734
 2:                                  KEGG_RIBOSOME 2.596529e-04 0.0063058573
 3:          KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS 2.027465e-03 0.0313335544
 4: KEGG_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY 1.103281e-03 0.0187557814
 5:                               KEGG_SPLICEOSOME 1.312272e-04 0.0055771574
 6:                KEGG_NUCLEOTIDE_EXCISION_REPAIR 2.330731e-03 0.0330186847
 7:                                KEGG_CELL_CYCLE 8.911637e-05 0.0050499277
 8:               KEGG_CHEMOKINE_SIGNALING_PATHWAY 3.926964e-06 0.0006675839
 9:                    KEGG_ACUTE_MYELOID_LEUKEMIA 2.384820e-04 0.0063058573
10: KEGG_SNARE_INTERACTIONS_IN_VESICULAR_TRANSPORT 6.975302e-04 0.0131755708
      log2err         ES       NES  size  leadingEdge
        &lt;num&gt;      &lt;num&gt;     &lt;num&gt; &lt;int&gt;       &lt;list&gt;
 1: 0.4984931 -0.4478346 -1.863136    56 629, 526....
 2: 0.4984931 -0.3806425 -1.709647    85 6122, 61....
 3: 0.4317077  0.4716664  1.580418    93 653361, ....
 4: 0.4550599  0.4708507  1.583557    97 5777, 31....
 5: 0.5188481  0.4717739  1.621554   126 10450, 8....
 6: 0.3263516  0.5479585  1.659690    43 3978, 20....
 7: 0.5384341  0.4873647  1.666510   119 896, 103....
 8: 0.6105269  0.4757109  1.670327   169 6361, 12....
 9: 0.5188481  0.5512012  1.734604    56 369, 529....
10: 0.4772708  0.5885213  1.742164    38 6843, 51....</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a>sig[<span class="fu">order</span>(<span class="sc">-</span>sig<span class="sc">$</span>NES)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],] <span class="co">#Higher targeting in diabetes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                           pathway         pval         padj
                                            &lt;char&gt;        &lt;num&gt;        &lt;num&gt;
 1:                  KEGG_PRIMARY_IMMUNODEFICIENCY 2.454718e-04 0.0063058573
 2:         KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY 2.148532e-05 0.0018262526
 3: KEGG_SNARE_INTERACTIONS_IN_VESICULAR_TRANSPORT 6.975302e-04 0.0131755708
 4:                    KEGG_ACUTE_MYELOID_LEUKEMIA 2.384820e-04 0.0063058573
 5:               KEGG_CHEMOKINE_SIGNALING_PATHWAY 3.926964e-06 0.0006675839
 6:                                KEGG_CELL_CYCLE 8.911637e-05 0.0050499277
 7:                KEGG_NUCLEOTIDE_EXCISION_REPAIR 2.330731e-03 0.0330186847
 8:                               KEGG_SPLICEOSOME 1.312272e-04 0.0055771574
 9: KEGG_NATURAL_KILLER_CELL_MEDIATED_CYTOTOXICITY 1.103281e-03 0.0187557814
10:          KEGG_FC_GAMMA_R_MEDIATED_PHAGOCYTOSIS 2.027465e-03 0.0313335544
      log2err        ES      NES  size  leadingEdge
        &lt;num&gt;     &lt;num&gt;    &lt;num&gt; &lt;int&gt;       &lt;list&gt;
 1: 0.4984931 0.6474564 1.827810    29 916, 393....
 2: 0.5756103 0.5260268 1.771546   100 5777, 91....
 3: 0.4772708 0.5885213 1.742164    38 6843, 51....
 4: 0.5188481 0.5512012 1.734604    56 369, 529....
 5: 0.6105269 0.4757109 1.670327   169 6361, 12....
 6: 0.5384341 0.4873647 1.666510   119 896, 103....
 7: 0.3263516 0.5479585 1.659690    43 3978, 20....
 8: 0.5188481 0.4717739 1.621554   126 10450, 8....
 9: 0.4550599 0.4708507 1.583557    97 5777, 31....
10: 0.4317077 0.4716664 1.580418    93 653361, ....</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(fgseaRes)</span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Settings</span></span>
<span id="cb184-3"><a href="#cb184-3" aria-hidden="true" tabindex="-1"></a>fdrcut <span class="ot">&lt;-</span> <span class="fl">0.05</span> <span class="co"># FDR cut-off to use as output for significant signatures</span></span>
<span id="cb184-4"><a href="#cb184-4" aria-hidden="true" tabindex="-1"></a>dencol_neg <span class="ot">&lt;-</span> <span class="st">"blue"</span> <span class="co"># bubble plot color for negative ES</span></span>
<span id="cb184-5"><a href="#cb184-5" aria-hidden="true" tabindex="-1"></a>dencol_pos <span class="ot">&lt;-</span> <span class="st">"red"</span> <span class="co"># bubble plot color for positive ES</span></span>
<span id="cb184-6"><a href="#cb184-6" aria-hidden="true" tabindex="-1"></a>signnamelength <span class="ot">&lt;-</span> <span class="dv">4</span> <span class="co"># set to remove prefix from signature names (2 for "GO", 4 for "KEGG", 8 for "REACTOME")</span></span>
<span id="cb184-7"><a href="#cb184-7" aria-hidden="true" tabindex="-1"></a>asp <span class="ot">&lt;-</span> <span class="dv">3</span> <span class="co"># aspect ratio of bubble plot</span></span>
<span id="cb184-8"><a href="#cb184-8" aria-hidden="true" tabindex="-1"></a>charcut <span class="ot">&lt;-</span> <span class="dv">100</span> <span class="co"># cut signature name in heatmap to this nr of characters</span></span>
<span id="cb184-9"><a href="#cb184-9" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">as.character</span>(dat<span class="sc">$</span>pathway) <span class="co"># 'a' is a great variable name to substitute row names with something more readable</span></span>
<span id="cb184-10"><a href="#cb184-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(a)){</span>
<span id="cb184-11"><a href="#cb184-11" aria-hidden="true" tabindex="-1"></a>  a[j] <span class="ot">&lt;-</span> <span class="fu">substr</span>(a[j], signnamelength<span class="sc">+</span><span class="dv">2</span>, <span class="fu">nchar</span>(a[j]))</span>
<span id="cb184-12"><a href="#cb184-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb184-13"><a href="#cb184-13" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">tolower</span>(a) <span class="co"># convert to lower case (you may want to comment this out, it really depends on what signatures you are looking at, c6 signatures contain gene names, and converting those to lower case may be confusing)</span></span>
<span id="cb184-14"><a href="#cb184-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(a)){</span>
<span id="cb184-15"><a href="#cb184-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nchar</span>(a[j])<span class="sc">&gt;</span>charcut) { a[j] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">substr</span>(a[j], <span class="dv">1</span>, charcut), <span class="st">"..."</span>, <span class="at">sep=</span><span class="st">" "</span>)}</span>
<span id="cb184-16"><a href="#cb184-16" aria-hidden="true" tabindex="-1"></a>} <span class="co"># cut signature names that have more characters than charcut, and add "..."</span></span>
<span id="cb184-17"><a href="#cb184-17" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, a)</span>
<span id="cb184-18"><a href="#cb184-18" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>NAME <span class="ot">&lt;-</span> a</span>
<span id="cb184-19"><a href="#cb184-19" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat[dat[,<span class="st">"padj"</span>]<span class="sc">&lt;</span>fdrcut,]</span>
<span id="cb184-20"><a href="#cb184-20" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat2[<span class="fu">order</span>(dat2[,<span class="st">"padj"</span>]),] </span>
<span id="cb184-21"><a href="#cb184-21" aria-hidden="true" tabindex="-1"></a>dat2<span class="sc">$</span>signature <span class="ot">&lt;-</span> <span class="fu">factor</span>(dat2<span class="sc">$</span>NAME, <span class="fu">rev</span>(<span class="fu">as.character</span>(dat2<span class="sc">$</span>NAME)))</span>
<span id="cb184-22"><a href="#cb184-22" aria-hidden="true" tabindex="-1"></a>sign_neg <span class="ot">&lt;-</span> <span class="fu">which</span>(dat2[,<span class="st">"NES"</span>]<span class="sc">&lt;</span><span class="dv">0</span>) <span class="co">#Up regulated in cancer</span></span>
<span id="cb184-23"><a href="#cb184-23" aria-hidden="true" tabindex="-1"></a>sign_pos <span class="ot">&lt;-</span> <span class="fu">which</span>(dat2[,<span class="st">"NES"</span>]<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb184-24"><a href="#cb184-24" aria-hidden="true" tabindex="-1"></a>signcol <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(dat2<span class="sc">$</span>signature))</span>
<span id="cb184-25"><a href="#cb184-25" aria-hidden="true" tabindex="-1"></a>signcol[sign_neg] <span class="ot">&lt;-</span> dencol_neg <span class="co"># text color of negative signatures</span></span>
<span id="cb184-26"><a href="#cb184-26" aria-hidden="true" tabindex="-1"></a>signcol[sign_pos] <span class="ot">&lt;-</span> dencol_pos <span class="co"># text color of positive signatures</span></span>
<span id="cb184-27"><a href="#cb184-27" aria-hidden="true" tabindex="-1"></a>signcol <span class="ot">&lt;-</span> <span class="fu">rev</span>(signcol) <span class="co"># need to revert vector of colors, because ggplot starts plotting these from below</span></span>
<span id="cb184-28"><a href="#cb184-28" aria-hidden="true" tabindex="-1"></a>g<span class="ot">&lt;-</span><span class="fu">ggplot</span>(dat2, <span class="fu">aes</span>(<span class="at">x=</span>padj,<span class="at">y=</span>signature,<span class="at">size=</span>size))</span>
<span id="cb184-29"><a href="#cb184-29" aria-hidden="true" tabindex="-1"></a>g<span class="sc">+</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill=</span>NES), <span class="at">shape=</span><span class="dv">21</span>, <span class="at">colour=</span><span class="st">"white"</span>)<span class="sc">+</span></span>
<span id="cb184-30"><a href="#cb184-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span> <span class="co"># white background, needs to be placed before the "signcol" line</span></span>
<span id="cb184-31"><a href="#cb184-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>,fdrcut)<span class="sc">+</span></span>
<span id="cb184-32"><a href="#cb184-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_area</span>(<span class="at">max_size=</span><span class="dv">10</span>,<span class="at">guide=</span><span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb184-33"><a href="#cb184-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low=</span>dencol_neg, <span class="at">high=</span>dencol_pos)<span class="sc">+</span></span>
<span id="cb184-34"><a href="#cb184-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">colour=</span>signcol))<span class="sc">+</span></span>
<span id="cb184-35"><a href="#cb184-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">aspect.ratio=</span>asp, <span class="at">axis.title.y=</span><span class="fu">element_blank</span>()) <span class="co"># test aspect.ratio</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Vectorized input to `element_text()` is not officially supported.
ℹ Results may be unexpected or may change in future versions of ggplot2.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb186"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb186-1"><a href="#cb186-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BLUE is upregulated in diabetes, red is downregulated in diabetes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="analysis-by-alcohol-consumption" class="level2">
<h2 class="anchored" data-anchor-id="analysis-by-alcohol-consumption">Analysis by alcohol consumption</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">colData</span>(tcga)<span class="sc">@</span>listData<span class="sc">$</span>tcga.gdc_cases.exposures.alcohol_history)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 no yes 
 61  96 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb189"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a>non_na_samples <span class="ot">&lt;-</span> <span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">colData</span>(tcga)<span class="sc">$</span>tcga.gdc_cases.exposures.alcohol_history)</span>
<span id="cb189-2"><a href="#cb189-2" aria-hidden="true" tabindex="-1"></a>tcga_exprs_filtered_alcohol <span class="ot">&lt;-</span> <span class="fu">assay</span>(tcga_filtered, <span class="st">"raw_counts"</span>)[, non_na_samples]</span>
<span id="cb189-3"><a href="#cb189-3" aria-hidden="true" tabindex="-1"></a>tcga_colData_filtered_alcohol <span class="ot">&lt;-</span> <span class="fu">colData</span>(tcga_filtered)[non_na_samples, ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb190"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb190-1"><a href="#cb190-1" aria-hidden="true" tabindex="-1"></a>condition_alcohol <span class="ot">&lt;-</span> <span class="fu">factor</span>(tcga_colData_filtered_alcohol<span class="sc">$</span>tcga.gdc_cases.exposures.alcohol_history, </span>
<span id="cb190-2"><a href="#cb190-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"no"</span>, <span class="st">"yes"</span>))</span>
<span id="cb190-3"><a href="#cb190-3" aria-hidden="true" tabindex="-1"></a>design_alcohol <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> condition_alcohol)</span>
<span id="cb190-4"><a href="#cb190-4" aria-hidden="true" tabindex="-1"></a>voomOutput_alcohol <span class="ot">&lt;-</span> <span class="fu">voom</span>(tcga_exprs_filtered_alcohol, design_alcohol, <span class="at">plot=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-58-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb191"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb191-1"><a href="#cb191-1" aria-hidden="true" tabindex="-1"></a>fit_alcohol <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(voomOutput_alcohol, design_alcohol)</span>
<span id="cb191-2"><a href="#cb191-2" aria-hidden="true" tabindex="-1"></a>fit_alcohol <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(fit_alcohol)</span>
<span id="cb191-3"><a href="#cb191-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSA</span>(fit_alcohol, <span class="at">main=</span><span class="st">"Final model mean-variance trend"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-58-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb192"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb192-1"><a href="#cb192-1" aria-hidden="true" tabindex="-1"></a>res_alcohol <span class="ot">&lt;-</span> <span class="fu">topTable</span>(fit_alcohol, <span class="at">coef=</span><span class="st">"condition_alcoholyes"</span>, <span class="at">number=</span><span class="cn">Inf</span>)</span>
<span id="cb192-2"><a href="#cb192-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_alcohol)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       logFC  AveExpr        t      P.Value adj.P.Val
ENSG00000161513.11 0.4327979 3.290735 4.013902 9.197379e-05 0.5676376
ENSG00000156521.13 0.3304523 4.297531 3.743975 2.531527e-04 0.5676376
ENSG00000239857.6  0.3380079 5.645865 3.490405 6.250782e-04 0.5676376
ENSG00000168061.14 0.3951702 2.686615 3.677814 3.219206e-04 0.5676376
ENSG00000115274.14 0.2972405 4.242625 3.485647 6.354849e-04 0.5676376
ENSG00000162066.14 0.3763268 4.407840 3.454939 7.066740e-04 0.5676376
                            B
ENSG00000161513.11  0.1740560
ENSG00000156521.13 -0.2310389
ENSG00000239857.6  -0.7396564
ENSG00000168061.14 -0.8142611
ENSG00000115274.14 -0.8838893
ENSG00000162066.14 -0.9361971</code></pre>
</div>
<div class="sourceCode cell-code" id="cb194"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb194-1"><a href="#cb194-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of genes differentially expressed</span></span>
<span id="cb194-2"><a href="#cb194-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_diabetes<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(res_diabetes<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb196"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb196-1"><a href="#cb196-1" aria-hidden="true" tabindex="-1"></a><span class="co"># number of genes differentially expressed</span></span>
<span id="cb196-2"><a href="#cb196-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_diabetes<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">abs</span>(res_diabetes<span class="sc">$</span>logFC) <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co"># overexpressed in TCGA</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_diabetes<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> res_diabetes<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
<div class="sourceCode cell-code" id="cb200"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb200-1"><a href="#cb200-1" aria-hidden="true" tabindex="-1"></a><span class="co"># overexpressed in GTEX</span></span>
<span id="cb200-2"><a href="#cb200-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(res_diabetes<span class="sc">$</span>adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> res_diabetes<span class="sc">$</span>logFC <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0</code></pre>
</div>
</div>
<section id="gene-network-analysis-2" class="level3">
<h3 class="anchored" data-anchor-id="gene-network-analysis-2">Gene Network Analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb202"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a>tcga_filtered_nas <span class="ot">&lt;-</span> tcga_filtered[,<span class="sc">!</span><span class="fu">is.na</span>(tcga_filtered<span class="sc">$</span>tcga.gdc_cases.exposures.alcohol_history)]</span>
<span id="cb202-2"><a href="#cb202-2" aria-hidden="true" tabindex="-1"></a>no_alcohol <span class="ot">&lt;-</span> tcga_filtered_nas[,tcga_filtered_nas<span class="sc">$</span>tcga.gdc_cases.exposures.alcohol_history<span class="sc">==</span><span class="st">'no'</span>]</span>
<span id="cb202-3"><a href="#cb202-3" aria-hidden="true" tabindex="-1"></a>alcohol <span class="ot">&lt;-</span> tcga_filtered_nas[,tcga_filtered_nas<span class="sc">$</span>tcga.gdc_cases.exposures.alcohol_history<span class="sc">==</span><span class="st">'yes'</span>]</span>
<span id="cb202-4"><a href="#cb202-4" aria-hidden="true" tabindex="-1"></a>zero_na_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assay</span>(no_alcohol, <span class="st">"logtpm"</span>), <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="fu">length</span>(x[(<span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x<span class="sc">!=</span><span class="dv">0</span>) ]))</span>
<span id="cb202-5"><a href="#cb202-5" aria-hidden="true" tabindex="-1"></a>expr_no_alcohol <span class="ot">&lt;-</span> <span class="fu">assay</span>(no_alcohol, <span class="st">"logtpm"</span>)[zero_na_counts <span class="sc">&gt;</span> <span class="dv">20</span>,]</span>
<span id="cb202-6"><a href="#cb202-6" aria-hidden="true" tabindex="-1"></a>zero_na_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(<span class="fu">assay</span>(alcohol, <span class="st">"logtpm"</span>), <span class="at">MARGIN =</span> <span class="dv">1</span>, <span class="at">FUN =</span> <span class="cf">function</span>(x) <span class="fu">length</span>(x[(<span class="sc">!</span><span class="fu">is.na</span>(x) <span class="sc">&amp;</span> x<span class="sc">!=</span><span class="dv">0</span>) ]))</span>
<span id="cb202-7"><a href="#cb202-7" aria-hidden="true" tabindex="-1"></a>expr_alcohol <span class="ot">&lt;-</span> <span class="fu">assay</span>(alcohol, <span class="st">"logtpm"</span>)[zero_na_counts <span class="sc">&gt;</span> <span class="dv">20</span>,]</span>
<span id="cb202-8"><a href="#cb202-8" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(expr_no_alcohol) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">row.names</span>(expr_no_alcohol))</span>
<span id="cb202-9"><a href="#cb202-9" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(expr_alcohol) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">..*"</span>, <span class="st">""</span>, <span class="fu">row.names</span>(expr_alcohol))</span>
<span id="cb202-10"><a href="#cb202-10" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(expr_no_alcohol, file = "no_alcohol.txt", sep = "\t", quote = FALSE, row.names = TRUE)</span></span>
<span id="cb202-11"><a href="#cb202-11" aria-hidden="true" tabindex="-1"></a><span class="co">#write.table(expr_alcohol, file = "alcohol.txt", sep = "\t", quote = FALSE, row.names = TRUE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a>network_full <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"output_alcohol.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a>network <span class="ot">&lt;-</span> network_full[<span class="fu">order</span>(network_full<span class="sc">$</span>force, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a>network <span class="ot">&lt;-</span> network[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>,]</span>
<span id="cb203-4"><a href="#cb203-4" aria-hidden="true" tabindex="-1"></a>network <span class="ot">&lt;-</span> <span class="fu">subset</span>(network, <span class="at">select=</span><span class="sc">-</span>motif)</span>
<span id="cb203-5"><a href="#cb203-5" aria-hidden="true" tabindex="-1"></a>network<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(network<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,network<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb203-6"><a href="#cb203-6" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>hgnc_symbol[<span class="fu">match</span>(network<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb203-7"><a href="#cb203-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb203-8"><a href="#cb203-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(network,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>              tf            gene    force
20665626  ZNF341 ENSG00000268836 20.99288
21236918 ZNF385D ENSG00000272829 17.81017
21844091 ZNF385D ENSG00000277368 17.18748
20589865 ZNF385D ENSG00000268129 15.45473
16564976 ZNF385D      ZNF687-AS1 15.45402
5260990  ZNF385D           HOXD1 15.43861
9779394  ZNF385D          SPACA9 15.24638
10454363 ZNF385D           RSPO1 15.22768
14243960 ZNF385D         TMEM240 15.02135
18255116   CEBPB     ALDH1L1-AS1 14.47294</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb205"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb205-1"><a href="#cb205-1" aria-hidden="true" tabindex="-1"></a>normal_network_full <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"output_no_alcohol.txt"</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">stringsAsFactors=</span><span class="cn">FALSE</span>)</span>
<span id="cb205-2"><a href="#cb205-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-3"><a href="#cb205-3" aria-hidden="true" tabindex="-1"></a>network_full <span class="ot">&lt;-</span> network_full[network_full<span class="sc">$</span>gene <span class="sc">%in%</span> normal_network_full<span class="sc">$</span>gene,]</span>
<span id="cb205-4"><a href="#cb205-4" aria-hidden="true" tabindex="-1"></a>network_full <span class="ot">&lt;-</span> network_full[network_full<span class="sc">$</span>tf <span class="sc">%in%</span> normal_network_full<span class="sc">$</span>tf,]</span>
<span id="cb205-5"><a href="#cb205-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb205-6"><a href="#cb205-6" aria-hidden="true" tabindex="-1"></a>diffnet <span class="ot">&lt;-</span> network_full</span>
<span id="cb205-7"><a href="#cb205-7" aria-hidden="true" tabindex="-1"></a>diffnet<span class="sc">$</span>force <span class="ot">&lt;-</span> network_full<span class="sc">$</span>force<span class="sc">-</span>normal_network_full<span class="sc">$</span>force</span>
<span id="cb205-8"><a href="#cb205-8" aria-hidden="true" tabindex="-1"></a>diffnet <span class="ot">&lt;-</span> <span class="fu">subset</span>(diffnet, <span class="at">select=</span><span class="sc">-</span>motif)</span>
<span id="cb205-9"><a href="#cb205-9" aria-hidden="true" tabindex="-1"></a>diffnet<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(diffnet<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,diffnet<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb205-10"><a href="#cb205-10" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>hgnc_symbol[<span class="fu">match</span>(diffnet<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb205-11"><a href="#cb205-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb205-12"><a href="#cb205-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(diffnet,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       tf   gene       force
1    ALX1 TSPAN6 -0.28368464
2    ALX3 TSPAN6  0.02208510
3    ALX4 TSPAN6 -0.21773207
4      AR TSPAN6  0.05960558
5   ARGFX TSPAN6 -0.06444600
6  ARID3A TSPAN6 -0.29166012
7  ARID5B TSPAN6 -0.37497464
8    ARNT TSPAN6  0.25309625
9   ARNT2 TSPAN6  0.25141121
10  ARNTL TSPAN6  0.26784691</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb207"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb207-1"><a href="#cb207-1" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> diffnet</span>
<span id="cb207-2"><a href="#cb207-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(edges) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"from"</span>, <span class="st">"to"</span>, <span class="st">"value"</span>)</span>
<span id="cb207-3"><a href="#cb207-3" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>arrows <span class="ot">=</span> <span class="st">"to"</span>   </span>
<span id="cb207-4"><a href="#cb207-4" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>color  <span class="ot">=</span> <span class="fu">ifelse</span>(edges<span class="sc">$</span>value <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"green"</span>, <span class="st">"red"</span>)</span>
<span id="cb207-5"><a href="#cb207-5" aria-hidden="true" tabindex="-1"></a>edges<span class="sc">$</span>value  <span class="ot">=</span> <span class="fu">abs</span>(edges<span class="sc">$</span>value)</span>
<span id="cb207-6"><a href="#cb207-6" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges[<span class="fu">order</span>(edges<span class="sc">$</span>value, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb207-7"><a href="#cb207-7" aria-hidden="true" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>,]</span>
<span id="cb207-8"><a href="#cb207-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(edges,<span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           from    to    value arrows color
18616748 ZNF235 PINX1 7.950825     to green
18616770 ZNF287 PINX1 7.905846     to green
18616745 ZNF225 PINX1 7.688056     to green
18616483  PRDM6 PINX1 7.684752     to green
18616839 ZNF487 PINX1 7.672290     to green
8301786  ZNF235 NEIL2 7.621407     to green
8301808  ZNF287 NEIL2 7.573878     to green
18616811 ZNF418 PINX1 7.435805     to green
8301521   PRDM6 NEIL2 7.368233     to green
8301877  ZNF487 NEIL2 7.283991     to green</code></pre>
</div>
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Green is upregulated in alcoholics, red is downregulated in alcoholics</span></span>
<span id="cb209-2"><a href="#cb209-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Note that the difference here is not too much. e.g, for sex it was ~17-18. Here its 8</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a>nodes     <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">id =</span> <span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))), </span>
<span id="cb210-2"><a href="#cb210-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">label=</span><span class="fu">unique</span>(<span class="fu">as.vector</span>(<span class="fu">as.matrix</span>(edges[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>)]))))</span>
<span id="cb210-3"><a href="#cb210-3" aria-hidden="true" tabindex="-1"></a>nodes<span class="sc">$</span>group <span class="ot">=</span> <span class="fu">ifelse</span>(nodes<span class="sc">$</span>id <span class="sc">%in%</span> edges<span class="sc">$</span>from, <span class="st">"TF"</span>, <span class="st">"gene"</span>)</span>
<span id="cb210-4"><a href="#cb210-4" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visNetwork</span>(nodes, edges, <span class="at">width =</span> <span class="st">"100%"</span>)</span>
<span id="cb210-5"><a href="#cb210-5" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"TF"</span>, <span class="at">shape =</span> <span class="st">"triangle"</span>,</span>
<span id="cb210-6"><a href="#cb210-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"purple"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb210-7"><a href="#cb210-7" aria-hidden="true" tabindex="-1"></a>net <span class="ot">&lt;-</span> <span class="fu">visGroups</span>(net, <span class="at">groupname =</span> <span class="st">"gene"</span>, <span class="at">shape =</span> <span class="st">"dot"</span>,       </span>
<span id="cb210-8"><a href="#cb210-8" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="fu">list</span>(<span class="at">background =</span> <span class="st">"teal"</span>, <span class="at">border=</span><span class="st">"black"</span>))</span>
<span id="cb210-9"><a href="#cb210-9" aria-hidden="true" tabindex="-1"></a><span class="fu">visLegend</span>(net, <span class="at">main=</span><span class="st">"Legend"</span>, <span class="at">position=</span><span class="st">"right"</span>, <span class="at">ncol=</span><span class="dv">1</span>) <span class="co">#GREEN is upregulated in alcohol people</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div id="htmlwidget-1871ab522cf9784ab3a3" style="width:100%;height:480px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-1871ab522cf9784ab3a3">{"x":{"nodes":{"id":["ZNF235","ZNF287","ZNF225","PRDM6","ZNF487","ZNF418","ZNF879","ZFP28","ZNF98","ZNF182","ZNF570","ZNF613","ZNF492","IRF1","ZNF567","ZNF79","ZNF571","ZNF22","ZNF334","ZNF443","PINX1","NEIL2","RPS4XP22","ELOBP2","ENSG00000254317","LYNX1-SLURP2","RAET1G","ENSG00000277702","CLYBL","SNORC","ENSG00000274904","ENSG00000280195","ENSG00000272220","ENSG00000227619","SNORA20","CCZ1P1",null,"GNG3","TPT1P15","ITM2C","RNVU1-1","RDM1P2","IL1RL2","CST4","MTCO2P2","PCDHB8","HDGFP1","ENSG00000267924","ENSG00000244137","CRAT","TM4SF18-AS1","NKD1"],"label":["ZNF235","ZNF287","ZNF225","PRDM6","ZNF487","ZNF418","ZNF879","ZFP28","ZNF98","ZNF182","ZNF570","ZNF613","ZNF492","IRF1","ZNF567","ZNF79","ZNF571","ZNF22","ZNF334","ZNF443","PINX1","NEIL2","RPS4XP22","ELOBP2","ENSG00000254317","LYNX1-SLURP2","RAET1G","ENSG00000277702","CLYBL","SNORC","ENSG00000274904","ENSG00000280195","ENSG00000272220","ENSG00000227619","SNORA20","CCZ1P1",null,"GNG3","TPT1P15","ITM2C","RNVU1-1","RDM1P2","IL1RL2","CST4","MTCO2P2","PCDHB8","HDGFP1","ENSG00000267924","ENSG00000244137","CRAT","TM4SF18-AS1","NKD1"],"group":["TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","TF","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene","gene"]},"edges":{"from":["ZNF235","ZNF287","ZNF225","PRDM6","ZNF487","ZNF235","ZNF287","ZNF418","PRDM6","ZNF487","ZNF225","ZNF879","ZFP28","ZNF418","ZFP28","ZNF98","ZNF879","ZNF182","ZNF570","ZNF613","ZNF235","ZNF98","ZNF570","ZNF235","ZNF492","ZNF418","ZNF287","ZNF235","ZNF287","IRF1","ZNF235","PRDM6","ZNF879","ZNF418","ZNF487","ZNF235","ZNF235","ZNF418","ZNF567","IRF1","ZNF235","ZNF182","ZNF287","ZNF287","PRDM6","ZNF225","ZNF418","ZNF487","PRDM6","ZNF613","ZNF235","PRDM6","ZNF879","ZNF235","ZNF287","ZNF287","ZNF225","ZNF225","ZNF487","ZFP28","ZNF492","ZNF418","ZNF418","ZNF287","ZNF879","ZNF287","ZNF567","PRDM6","ZNF225","ZNF487","ZNF418","ZNF235","PRDM6","ZNF418","ZNF235","ZNF235","ZNF79","PRDM6","ZNF879","PRDM6","ZNF487","ZNF418","ZNF571","ZNF487","ZNF235","ZNF879","ZNF225","ZNF225","ZNF418","ZNF235","ZNF570","ZNF487","ZNF879","ZNF79","ZFP28","ZNF418","ZNF235","ZNF879","ZNF287","ZNF287","ZNF287","ZNF487","ZFP28","ZNF235","PRDM6","ZFP28","PRDM6","ZNF418","ZFP28","ZNF98","ZNF235","ZNF235","ZFP28","ZFP28","ZNF879","ZNF225","PRDM6","ZNF225","ZNF487","ZNF287","ZNF418","ZNF22","ZFP28","ZNF235","ZNF879","ZNF879","ZNF287","ZNF287","ZNF418","ZNF225","PRDM6","ZNF79","ZNF287","ZNF287","ZNF98","ZNF225","ZNF487","ZNF235","ZNF235","ZNF879","ZNF487","ZNF235","ZNF235","ZNF287","ZNF287","PRDM6","ZNF182","ZNF613","ZNF418","ZNF879","ZNF418","ZNF225","ZNF879","ZNF235","ZNF418","ZNF570","ZNF287","ZNF487","ZNF487","ZNF287","ZNF225","ZNF235","ZNF287","ZNF225","PRDM6","ZNF418","PRDM6","PRDM6","ZNF98","ZFP28","ZNF418","ZNF570","PRDM6","ZNF98","ZNF334","ZNF570","ZNF334","ZNF225","ZNF567","ZNF235","ZNF287","ZNF418","IRF1","ZNF287","PRDM6","ZNF570","ZNF418","ZNF487","ZNF571","ZNF287","ZNF443","PRDM6","ZFP28","ZNF235","ZNF98","ZNF879","ZNF235","PRDM6","ZNF235","ZNF487"],"to":["PINX1","PINX1","PINX1","PINX1","PINX1","NEIL2","NEIL2","PINX1","NEIL2","NEIL2","NEIL2","PINX1","PINX1","NEIL2","NEIL2","PINX1","NEIL2","PINX1","PINX1","PINX1","RPS4XP22","NEIL2","NEIL2","ELOBP2","PINX1","RPS4XP22","RPS4XP22","ENSG00000254317","ENSG00000254317","PINX1","LYNX1-SLURP2","RPS4XP22","RPS4XP22","LYNX1-SLURP2","RPS4XP22","RAET1G","ENSG00000277702","ELOBP2","PINX1","NEIL2","CLYBL","NEIL2","RAET1G","ELOBP2","ENSG00000254317","RPS4XP22","CLYBL","ENSG00000254317","RAET1G","NEIL2","SNORC","ELOBP2","LYNX1-SLURP2","ENSG00000274904","CLYBL","ENSG00000277702","ELOBP2","ENSG00000254317","RAET1G","RPS4XP22","NEIL2","ENSG00000277702","ENSG00000254317","SNORC","ELOBP2","LYNX1-SLURP2","NEIL2","LYNX1-SLURP2","LYNX1-SLURP2","ELOBP2","SNORC","ENSG00000280195","ENSG00000277702","ENSG00000274904","ENSG00000272220","ENSG00000227619","PINX1","SNORC","CLYBL","CLYBL","ENSG00000277702","ENSG00000227619","PINX1","SNORC","SNORA20","SNORC","ENSG00000277702","RAET1G","ENSG00000280195","CCZ1P1","RPS4XP22","LYNX1-SLURP2","ENSG00000277702","NEIL2","ELOBP2","RAET1G",null,"ENSG00000254317","SNORA20","ENSG00000274904","ENSG00000280195","CLYBL","LYNX1-SLURP2","GNG3","SNORA20","ENSG00000254317","ENSG00000280195","CCZ1P1","ENSG00000277702","RPS4XP22","TPT1P15","ITM2C","CLYBL","RAET1G","ENSG00000274904","SNORC","ENSG00000274904","CLYBL","SNORA20","ENSG00000272220","GNG3","PINX1","SNORC","RNVU1-1","ENSG00000227619","ENSG00000280195","ENSG00000227619","ITM2C","ENSG00000272220","ENSG00000274904","ENSG00000227619","RPS4XP22","CCZ1P1","GNG3","ENSG00000254317","SNORA20","ENSG00000280195","RDM1P2","IL1RL2","RAET1G","ENSG00000274904","CST4","MTCO2P2","IL1RL2",null,"ENSG00000272220","RPS4XP22","RPS4XP22","TPT1P15","CCZ1P1",null,"ENSG00000280195","GNG3","PCDHB8","ITM2C","LYNX1-SLURP2","RNVU1-1","ENSG00000227619","ENSG00000272220","TPT1P15","ENSG00000227619","HDGFP1","ENSG00000267924","ENSG00000272220","GNG3","PCDHB8","CCZ1P1","ITM2C","RAET1G","ENSG00000274904","HDGFP1","ELOBP2",null,"ELOBP2","NEIL2","CLYBL","RPS4XP22","CCZ1P1","RPS4XP22","ENSG00000244137","CST4","CST4","RPS4XP22","RDM1P2","TPT1P15","ENSG00000254317","SNORA20","CCZ1P1","SNORA20","CRAT","PINX1","RNVU1-1","ENSG00000280195","TM4SF18-AS1","LYNX1-SLURP2",null,"ENSG00000267924","ENSG00000267924","NKD1","GNG3"],"value":[7.950825169354037,7.905846160857484,7.688056475186784,7.6847520403698,7.672289980714389,7.621406954760944,7.573878393531702,7.4358051217512,7.368232622025178,7.283990801216545,7.25605888362179,7.200621341748677,7.200383021015407,7.178287041952375,6.988740737930831,6.988053145526714,6.982059238793251,6.873433202026597,6.830895740527373,6.763317928589828,6.716257773334211,6.690605016322033,6.634305586647458,6.620181871907414,6.619896792038467,6.602608385091314,6.592769582011213,6.590084271080302,6.585309203470368,6.520520378224276,6.491610736618261,6.485061014925841,6.477601839589827,6.445130840629937,6.434908461129612,6.408395665848104,6.389730998597205,6.384552622139549,6.375525380948478,6.367076809861902,6.363863397171501,6.356305054227463,6.342870349634634,6.327919022972669,6.305987615443946,6.2944949706113,6.271056432860227,6.258537699373183,6.231680374128853,6.229192892094426,6.220733235802246,6.216553712513626,6.205508006782018,6.199949105803642,6.195690168893664,6.194510348891839,6.186642582861317,6.186052826384427,6.179019828895244,6.176520807989178,6.158165513375572,6.155211867821823,6.137242512612943,6.136972960576996,6.125942757637452,6.116139086882973,6.114714944047489,6.111713300582705,6.096219460114058,6.08972235121713,6.087584824924557,6.087223549051386,6.085004618921944,6.078642468687759,6.071675658440752,6.067218767980615,6.062827187530188,6.030410054757953,6.02795744853099,6.025133891304545,6.021583547941951,6.013548210242298,6.008251677437009,5.992053945598501,5.991632211723885,5.989911908732644,5.985131961769287,5.981937405531796,5.980098499413247,5.978952340751752,5.973531511622205,5.968931004213933,5.962104586867063,5.961472095799412,5.9549623469207,5.950953108868399,5.949674548663266,5.940065603338265,5.939059615518255,5.935225839900992,5.93220362362942,5.924412700547466,5.920700227680686,5.915882078757538,5.914942930608997,5.914793134486553,5.90125815572324,5.90001404036863,5.899666814269938,5.898378040668398,5.896066248279176,5.887929500520761,5.886262764476047,5.86879028316668,5.864541437068541,5.859130995583462,5.858228222851364,5.856643577670628,5.84302507407631,5.840607088333162,5.828186926505888,5.82387266147148,5.820134370618534,5.819338224085186,5.816907271603408,5.808386497883735,5.799292874747129,5.795520379415047,5.790794412900492,5.786356380151897,5.771452118265558,5.76974762612349,5.768510322821298,5.757699678678121,5.75756850732443,5.755897518742942,5.755779452670384,5.754079653604716,5.751110184140384,5.749939814064714,5.749329817643171,5.744893024964032,5.739766336977921,5.739600522887653,5.73409975278601,5.719040651456178,5.716640012785192,5.715837942482292,5.710263259808373,5.708031519039352,5.706983509978334,5.705217279295095,5.70453429572683,5.70117108590788,5.697624732024032,5.691571832357986,5.691075069400827,5.688496910511939,5.68831499815312,5.683213623447483,5.681967082575949,5.674453091173858,5.673902830391528,5.670698274772274,5.669146052581365,5.667522459505946,5.667248733565703,5.663652527647391,5.663473753037984,5.660540459438108,5.657790249558698,5.656699695143125,5.65356187265838,5.641533540113731,5.640313408544546,5.631752441579812,5.63021527877393,5.62489618746696,5.618286760883637,5.616747259113458,5.616161078875824,5.610472145735288,5.60843429658356,5.605402598379135,5.605087750733985,5.604893154769086,5.604637301719881,5.602519831848589,5.600346921981525,5.600140975541203,5.59882897875887,5.592913296448224,5.58793426243893,5.586856777902881,5.586628242926609,5.584283061534488,5.582408190357318,5.580715937539282,5.579264643471429,5.578623779931687],"arrows":["to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to","to"],"color":["green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","green","red","green","green","green","green","green","green","red","green","green","red","green","red","green","red","green","green","red","green","red","red","green","green","red","green","red","green","red","red","red","green","red","green","red","green","red","green","green","green","green","red","red","red","green","red","red","red","red","green","green","green","green","red","green","red","red","red","green","red","green","red","green","red","green","red","green","green","green","red","green","green","red","red","green","green","green","green","green","red","red","green","green","green","green","green","green","green","green","green","red","red","green","red","green","red","green","green","green","green","red","green","red","green","red","green","green","green","red","green","green","green","green","green","green","green","red","red","green","red","red","red","green","green","green","green","green","green","green","green","green","green","green","red","green","red","green","green","red","green","green","green","green","green","green","green","red","green","green","red","green","red","green","red","green","green","green","red","red","red","green","green","green","green","green","green","green","red","green","green","green","green","red","green","green","green","red","green"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot"},"manipulation":{"enabled":false},"groups":{"TF":{"shape":"triangle","color":{"background":"purple","border":"black"}},"useDefaultGroups":true,"gene":{"shape":"dot","color":{"background":"teal","border":"black"}}}},"groups":["TF","gene"],"width":"100%","height":null,"idselection":{"enabled":false},"byselection":{"enabled":false},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","legend":{"width":0.2,"useGroups":true,"position":"right","ncol":1,"stepX":100,"stepY":100,"zoom":true,"main":{"text":"Legend","style":"font-family:Georgia, Times New Roman, Times, serif;font-weight:bold;font-size:14px;text-align:center;"}}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb211"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="do">##PINX1 seems kinda interesting</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb212-2"><a href="#cb212-2" aria-hidden="true" tabindex="-1"></a>out_degree_normal<span class="ot">&lt;-</span> normal_network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(tf) <span class="sc">|&gt;</span></span>
<span id="cb212-3"><a href="#cb212-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">out_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb212-4"><a href="#cb212-4" aria-hidden="true" tabindex="-1"></a>out_degree<span class="ot">&lt;-</span> network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(tf) <span class="sc">|&gt;</span></span>
<span id="cb212-5"><a href="#cb212-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">out_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb212-6"><a href="#cb212-6" aria-hidden="true" tabindex="-1"></a>in_degree_normal<span class="ot">&lt;-</span> normal_network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(gene) <span class="sc">|&gt;</span></span>
<span id="cb212-7"><a href="#cb212-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">in_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb212-8"><a href="#cb212-8" aria-hidden="true" tabindex="-1"></a>in_degree<span class="ot">&lt;-</span> network_full <span class="sc">|&gt;</span> <span class="fu">group_by</span>(gene) <span class="sc">|&gt;</span></span>
<span id="cb212-9"><a href="#cb212-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">in_degree =</span> <span class="fu">sum</span>(force))</span>
<span id="cb212-10"><a href="#cb212-10" aria-hidden="true" tabindex="-1"></a>in_degree <span class="ot">&lt;-</span> in_degree[in_degree<span class="sc">$</span>gene <span class="sc">%in%</span> in_degree_normal<span class="sc">$</span>gene,]</span>
<span id="cb212-11"><a href="#cb212-11" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_degree_normal</span>
<span id="cb212-12"><a href="#cb212-12" aria-hidden="true" tabindex="-1"></a>in_difference<span class="sc">$</span>in_degree <span class="ot">&lt;-</span> in_degree_normal<span class="sc">$</span>in_degree <span class="sc">-</span> in_degree<span class="sc">$</span>in_degree</span>
<span id="cb212-13"><a href="#cb212-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-14"><a href="#cb212-14" aria-hidden="true" tabindex="-1"></a>pathways <span class="ot">&lt;-</span> <span class="fu">gmtPathways</span>(<span class="st">"c2.cp.kegg_legacy.v2024.1.Hs.entrez.gmt"</span>)</span>
<span id="cb212-15"><a href="#cb212-15" aria-hidden="true" tabindex="-1"></a>in_difference<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(annot[<span class="fu">match</span>(in_difference<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id),<span class="st">"hgnc_symbol"</span>]<span class="sc">==</span><span class="st">""</span>,in_difference<span class="sc">$</span>gene,  <span class="co"># If no match, keep Ensembl ID</span></span>
<span id="cb212-16"><a href="#cb212-16" aria-hidden="true" tabindex="-1"></a>  annot<span class="sc">$</span>entrezgene_id[<span class="fu">match</span>(in_difference<span class="sc">$</span>gene, annot<span class="sc">$</span>ensembl_gene_id)]  <span class="co"># Else, use HGNC symbol</span></span>
<span id="cb212-17"><a href="#cb212-17" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb212-18"><a href="#cb212-18" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_difference[<span class="sc">!</span><span class="fu">is.na</span>(in_difference<span class="sc">$</span>gene),]</span>
<span id="cb212-19"><a href="#cb212-19" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> in_difference[<span class="sc">!</span><span class="fu">duplicated</span>(in_difference<span class="sc">$</span>gene), ]</span>
<span id="cb212-20"><a href="#cb212-20" aria-hidden="true" tabindex="-1"></a>in_difference <span class="ot">&lt;-</span> <span class="fu">setNames</span>(in_difference<span class="sc">$</span>in_degree, in_difference<span class="sc">$</span>gene)</span>
<span id="cb212-21"><a href="#cb212-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-22"><a href="#cb212-22" aria-hidden="true" tabindex="-1"></a>fgseaRes <span class="ot">&lt;-</span> <span class="fu">fgsea</span>(pathways, in_difference, <span class="at">minSize=</span><span class="dv">15</span>, <span class="at">maxSize=</span><span class="dv">500</span>,<span class="at">nPermSimple =</span> <span class="dv">10000</span>)</span>
<span id="cb212-23"><a href="#cb212-23" aria-hidden="true" tabindex="-1"></a>sig <span class="ot">&lt;-</span> fgseaRes[fgseaRes<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb212-24"><a href="#cb212-24" aria-hidden="true" tabindex="-1"></a>sig[<span class="fu">order</span>(sig<span class="sc">$</span>NES)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],] <span class="co">#Lower targeting in alcoholics</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                                                      pathway         pval
                                                       &lt;char&gt;        &lt;num&gt;
 1:                             KEGG_ECM_RECEPTOR_INTERACTION 1.538491e-07
 2:                                       KEGG_FOCAL_ADHESION 5.262354e-06
 3: KEGG_ARRHYTHMOGENIC_RIGHT_VENTRICULAR_CARDIOMYOPATHY_ARVC 3.427588e-03
 4:                                           KEGG_PEROXISOME 2.522068e-03
 5:                                KEGG_FATTY_ACID_METABOLISM 2.413297e-03
 6:                                       KEGG_RNA_POLYMERASE 2.269495e-03
 7:                                      KEGG_DNA_REPLICATION 2.182276e-04
 8:                                KEGG_PYRIMIDINE_METABOLISM 5.363336e-06
 9:                                             KEGG_RIBOSOME 1.568215e-05
10:                                          KEGG_SPLICEOSOME 9.070182e-08
            padj   log2err         ES       NES  size  leadingEdge
           &lt;num&gt;     &lt;num&gt;      &lt;num&gt;     &lt;num&gt; &lt;int&gt;       &lt;list&gt;
 1: 4.359057e-06 0.6901325 -0.5578700 -2.219538    72 3381, 13....
 2: 1.013075e-04 0.6105269 -0.3805702 -1.744332   179 3381, 13....
 3: 3.884599e-02 0.4317077 -0.4319016 -1.647300    58 93589, 5....
 4: 3.062511e-02 0.3445213  0.4773879  1.662502    70 8310, 57....
 5: 3.062511e-02 0.4317077  0.5636585  1.749706    37 8310, 18....
 6: 3.062511e-02 0.4317077  0.6051518  1.769854    28 5435, 95....
 7: 3.372608e-03 0.5188481  0.6242161  1.914050    35 5424, 10....
 8: 1.013075e-04 0.6105269  0.5402905  1.963115    92 5435, 95....
 9: 2.665966e-04 0.5756103  0.5482781  1.965520    85 6133, 20....
10: 3.083862e-06 0.7049757  0.5314565  2.012344   126 6636, 66....</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb214"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb214-1"><a href="#cb214-1" aria-hidden="true" tabindex="-1"></a>sig[<span class="fu">order</span>(<span class="sc">-</span>sig<span class="sc">$</span>NES)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>],] <span class="co">#higher  targeting in alcoholics</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                           pathway         pval         padj   log2err
                            &lt;char&gt;        &lt;num&gt;        &lt;num&gt;     &lt;num&gt;
 1: KEGG_OXIDATIVE_PHOSPHORYLATION 9.785503e-13 1.003809e-10 0.9101197
 2:       KEGG_HUNTINGTONS_DISEASE 1.180952e-12 1.003809e-10 0.9101197
 3:        KEGG_PARKINSONS_DISEASE 4.014613e-09 1.706210e-07 0.7614608
 4:                KEGG_PROTEASOME 1.763458e-06 4.282683e-05 0.6435518
 5:        KEGG_ALZHEIMERS_DISEASE 3.372928e-09 1.706210e-07 0.7749390
 6:               KEGG_SPLICEOSOME 9.070182e-08 3.083862e-06 0.7049757
 7:                  KEGG_RIBOSOME 1.568215e-05 2.665966e-04 0.5756103
 8:     KEGG_PYRIMIDINE_METABOLISM 5.363336e-06 1.013075e-04 0.6105269
 9:           KEGG_DNA_REPLICATION 2.182276e-04 3.372608e-03 0.5188481
10:            KEGG_RNA_POLYMERASE 2.269495e-03 3.062511e-02 0.4317077
           ES      NES  size  leadingEdge
        &lt;num&gt;    &lt;num&gt; &lt;int&gt;       &lt;list&gt;
 1: 0.6164091 2.323239   121 4700, 13....
 2: 0.5705614 2.227533   162 4700, 13....
 3: 0.5699586 2.144767   119 4700, 13....
 4: 0.6721863 2.129278    41 5691, 56....
 5: 0.5393797 2.082267   145 4700, 13....
 6: 0.5314565 2.012344   126 6636, 66....
 7: 0.5482781 1.965520    85 6133, 20....
 8: 0.5402905 1.963115    92 5435, 95....
 9: 0.6242161 1.914050    35 5424, 10....
10: 0.6051518 1.769854    28 5435, 95....</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(fgseaRes)</span>
<span id="cb216-2"><a href="#cb216-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Settings</span></span>
<span id="cb216-3"><a href="#cb216-3" aria-hidden="true" tabindex="-1"></a>fdrcut <span class="ot">&lt;-</span> <span class="fl">0.05</span> <span class="co"># FDR cut-off to use as output for significant signatures</span></span>
<span id="cb216-4"><a href="#cb216-4" aria-hidden="true" tabindex="-1"></a>dencol_neg <span class="ot">&lt;-</span> <span class="st">"blue"</span> <span class="co"># bubble plot color for negative ES</span></span>
<span id="cb216-5"><a href="#cb216-5" aria-hidden="true" tabindex="-1"></a>dencol_pos <span class="ot">&lt;-</span> <span class="st">"red"</span> <span class="co"># bubble plot color for positive ES</span></span>
<span id="cb216-6"><a href="#cb216-6" aria-hidden="true" tabindex="-1"></a>signnamelength <span class="ot">&lt;-</span> <span class="dv">4</span> <span class="co"># set to remove prefix from signature names (2 for "GO", 4 for "KEGG", 8 for "REACTOME")</span></span>
<span id="cb216-7"><a href="#cb216-7" aria-hidden="true" tabindex="-1"></a>asp <span class="ot">&lt;-</span> <span class="dv">3</span> <span class="co"># aspect ratio of bubble plot</span></span>
<span id="cb216-8"><a href="#cb216-8" aria-hidden="true" tabindex="-1"></a>charcut <span class="ot">&lt;-</span> <span class="dv">100</span> <span class="co"># cut signature name in heatmap to this nr of characters</span></span>
<span id="cb216-9"><a href="#cb216-9" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">as.character</span>(dat<span class="sc">$</span>pathway) <span class="co"># 'a' is a great variable name to substitute row names with something more readable</span></span>
<span id="cb216-10"><a href="#cb216-10" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(a)){</span>
<span id="cb216-11"><a href="#cb216-11" aria-hidden="true" tabindex="-1"></a>  a[j] <span class="ot">&lt;-</span> <span class="fu">substr</span>(a[j], signnamelength<span class="sc">+</span><span class="dv">2</span>, <span class="fu">nchar</span>(a[j]))</span>
<span id="cb216-12"><a href="#cb216-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb216-13"><a href="#cb216-13" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">tolower</span>(a) <span class="co"># convert to lower case (you may want to comment this out, it really depends on what signatures you are looking at, c6 signatures contain gene names, and converting those to lower case may be confusing)</span></span>
<span id="cb216-14"><a href="#cb216-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(a)){</span>
<span id="cb216-15"><a href="#cb216-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">nchar</span>(a[j])<span class="sc">&gt;</span>charcut) { a[j] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">substr</span>(a[j], <span class="dv">1</span>, charcut), <span class="st">"..."</span>, <span class="at">sep=</span><span class="st">" "</span>)}</span>
<span id="cb216-16"><a href="#cb216-16" aria-hidden="true" tabindex="-1"></a>} <span class="co"># cut signature names that have more characters than charcut, and add "..."</span></span>
<span id="cb216-17"><a href="#cb216-17" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, a)</span>
<span id="cb216-18"><a href="#cb216-18" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>NAME <span class="ot">&lt;-</span> a</span>
<span id="cb216-19"><a href="#cb216-19" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat[dat[,<span class="st">"padj"</span>]<span class="sc">&lt;</span>fdrcut,]</span>
<span id="cb216-20"><a href="#cb216-20" aria-hidden="true" tabindex="-1"></a>dat2 <span class="ot">&lt;-</span> dat2[<span class="fu">order</span>(dat2[,<span class="st">"padj"</span>]),] </span>
<span id="cb216-21"><a href="#cb216-21" aria-hidden="true" tabindex="-1"></a>dat2<span class="sc">$</span>signature <span class="ot">&lt;-</span> <span class="fu">factor</span>(dat2<span class="sc">$</span>NAME, <span class="fu">rev</span>(<span class="fu">as.character</span>(dat2<span class="sc">$</span>NAME)))</span>
<span id="cb216-22"><a href="#cb216-22" aria-hidden="true" tabindex="-1"></a>sign_neg <span class="ot">&lt;-</span> <span class="fu">which</span>(dat2[,<span class="st">"NES"</span>]<span class="sc">&lt;</span><span class="dv">0</span>) <span class="co">#Up regulated in cancer</span></span>
<span id="cb216-23"><a href="#cb216-23" aria-hidden="true" tabindex="-1"></a>sign_pos <span class="ot">&lt;-</span> <span class="fu">which</span>(dat2[,<span class="st">"NES"</span>]<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb216-24"><a href="#cb216-24" aria-hidden="true" tabindex="-1"></a>signcol <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="cn">NA</span>, <span class="fu">length</span>(dat2<span class="sc">$</span>signature))</span>
<span id="cb216-25"><a href="#cb216-25" aria-hidden="true" tabindex="-1"></a>signcol[sign_neg] <span class="ot">&lt;-</span> dencol_neg <span class="co"># text color of negative signatures</span></span>
<span id="cb216-26"><a href="#cb216-26" aria-hidden="true" tabindex="-1"></a>signcol[sign_pos] <span class="ot">&lt;-</span> dencol_pos <span class="co"># text color of positive signatures</span></span>
<span id="cb216-27"><a href="#cb216-27" aria-hidden="true" tabindex="-1"></a>signcol <span class="ot">&lt;-</span> <span class="fu">rev</span>(signcol) <span class="co"># need to revert vector of colors, because ggplot starts plotting these from below</span></span>
<span id="cb216-28"><a href="#cb216-28" aria-hidden="true" tabindex="-1"></a>g<span class="ot">&lt;-</span><span class="fu">ggplot</span>(dat2, <span class="fu">aes</span>(<span class="at">x=</span>padj,<span class="at">y=</span>signature,<span class="at">size=</span>size))</span>
<span id="cb216-29"><a href="#cb216-29" aria-hidden="true" tabindex="-1"></a>g<span class="sc">+</span><span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill=</span>NES), <span class="at">shape=</span><span class="dv">21</span>, <span class="at">colour=</span><span class="st">"white"</span>)<span class="sc">+</span></span>
<span id="cb216-30"><a href="#cb216-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span> <span class="co"># white background, needs to be placed before the "signcol" line</span></span>
<span id="cb216-31"><a href="#cb216-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>,fdrcut)<span class="sc">+</span></span>
<span id="cb216-32"><a href="#cb216-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size_area</span>(<span class="at">max_size=</span><span class="dv">10</span>,<span class="at">guide=</span><span class="st">"none"</span>)<span class="sc">+</span></span>
<span id="cb216-33"><a href="#cb216-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low=</span>dencol_neg, <span class="at">high=</span>dencol_pos)<span class="sc">+</span></span>
<span id="cb216-34"><a href="#cb216-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">colour=</span>signcol))<span class="sc">+</span></span>
<span id="cb216-35"><a href="#cb216-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">aspect.ratio=</span>asp, <span class="at">axis.title.y=</span><span class="fu">element_blank</span>()) <span class="co"># test aspect.ratio</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Vectorized input to `element_text()` is not officially supported.
ℹ Results may be unexpected or may change in future versions of ggplot2.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="bst280_files/figure-html/unnamed-chunk-67-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb218"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb218-1"><a href="#cb218-1" aria-hidden="true" tabindex="-1"></a><span class="co">#BLUE is upregulated in alcohol people</span></span>
<span id="cb218-2"><a href="#cb218-2" aria-hidden="true" tabindex="-1"></a><span class="co">#red is downregulated</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>